// @daytona/sdk/cjs/Daytona.js
"use strict";
/*
 * Copyright Daytona Platforms Inc.
 * SPDX-License-Identifier: Apache-2.0
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Daytona = exports.CodeLanguage = exports.CODE_TOOLBOX_LANGUAGE_LABEL = void 0;
const tslib_1 = require("tslib");
const api_client_1 = require("@daytona/api-client");
const axios_1 = tslib_1.__importStar(require("axios"));
const DaytonaError_1 = require("./errors/DaytonaError");
const Image_1 = require("./Image");
const Sandbox_1 = require("./Sandbox");
const Secret_1 = require("./Secret");
const Snapshot_1 = require("./Snapshot");
const Volume_1 = require("./Volume");
const Import_1 = require("./utils/Import");
const packageJson = (0, Import_1.getPackageInfo)();
const Stream_1 = require("./utils/Stream");
const Runtime_1 = require("./utils/Runtime");
const otel_decorator_1 = require("./utils/otel.decorator");
const api_1 = require("@opentelemetry/api");
exports.CODE_TOOLBOX_LANGUAGE_LABEL = 'code-toolbox-language';
/**
 * Supported programming languages for code execution
 *
 * Python is used as the default sandbox language when no language is explicitly specified.
 */
var CodeLanguage;
(function (CodeLanguage) {
    CodeLanguage["PYTHON"] = "python";
    CodeLanguage["TYPESCRIPT"] = "typescript";
    CodeLanguage["JAVASCRIPT"] = "javascript";
})(CodeLanguage || (exports.CodeLanguage = CodeLanguage = {}));
/**
 * Main class for interacting with the Daytona API.
 * Provides methods for creating, managing, and interacting with Daytona Sandboxes.
 * Can be initialized either with explicit configuration or using environment variables.
 *
 * @property {VolumeService} volume - Service for managing Daytona Volumes
 * @property {SnapshotService} snapshot - Service for managing Daytona Snapshots
 * @property {SecretService} secret - Service for managing Daytona Secrets
 *
 * @example
 * // Using environment variables
 * // Uses DAYTONA_API_KEY, DAYTONA_API_URL, DAYTONA_TARGET
 * const daytona = new Daytona();
 * const sandbox = await daytona.create();
 *
 * @example
 * // Using explicit configuration
 * const config: DaytonaConfig = {
 *     apiKey: "your-api-key",
 *     apiUrl: "https://your-api.com",
 *     target: "us"
 * };
 * const daytona = new Daytona(config);
 *
 * @example
 * // Disposes daytona and flushes traces when done
 * await using daytona = new Daytona({
 *   otelEnabled: true,
 * });
 * @class
 */
let Daytona = (() => {
    let _instanceExtraInitializers = [];
    let _create_decorators;
    let _get_decorators;
    let _start_decorators;
    let _stop_decorators;
    let __experimental_fork_decorators;
    let _delete_decorators;
    return class Daytona {
        static {
            const _metadata = typeof Symbol === "function" && Symbol.metadata ? Object.create(null) : void 0;
            _create_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            _get_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            _start_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            _stop_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            __experimental_fork_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            _delete_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            tslib_1.__esDecorate(this, null, _create_decorators, { kind: "method", name: "create", static: false, private: false, access: { has: obj => "create" in obj, get: obj => obj.create }, metadata: _metadata }, null, _instanceExtraInitializers);
            tslib_1.__esDecorate(this, null, _get_decorators, { kind: "method", name: "get", static: false, private: false, access: { has: obj => "get" in obj, get: obj => obj.get }, metadata: _metadata }, null, _instanceExtraInitializers);
            tslib_1.__esDecorate(this, null, _start_decorators, { kind: "method", name: "start", static: false, private: false, access: { has: obj => "start" in obj, get: obj => obj.start }, metadata: _metadata }, null, _instanceExtraInitializers);
            tslib_1.__esDecorate(this, null, _stop_decorators, { kind: "method", name: "stop", static: false, private: false, access: { has: obj => "stop" in obj, get: obj => obj.stop }, metadata: _metadata }, null, _instanceExtraInitializers);
            tslib_1.__esDecorate(this, null, __experimental_fork_decorators, { kind: "method", name: "_experimental_fork", static: false, private: false, access: { has: obj => "_experimental_fork" in obj, get: obj => obj._experimental_fork }, metadata: _metadata }, null, _instanceExtraInitializers);
            tslib_1.__esDecorate(this, null, _delete_decorators, { kind: "method", name: "delete", static: false, private: false, access: { has: obj => "delete" in obj, get: obj => obj.delete }, metadata: _metadata }, null, _instanceExtraInitializers);
            if (_metadata) Object.defineProperty(this, Symbol.metadata, { enumerable: true, configurable: true, writable: true, value: _metadata });
        }
        clientConfig = tslib_1.__runInitializers(this, _instanceExtraInitializers);
        sandboxApi;
        objectStorageApi;
        configApi;
        target;
        apiKey;
        jwtToken;
        organizationId;
        apiUrl;
        otelSdk;
        volume;
        snapshot;
        secret;
        /**
         * Creates a new Daytona client instance.
         *
         * @param {DaytonaConfig} [config] - Configuration options
         * @throws {DaytonaAuthenticationError} When no credentials are provided (neither API key nor JWT token)
         * @throws {DaytonaAuthenticationError} When JWT token is provided without an organization ID
         */
        constructor(config) {
            let apiUrl;
            if (config) {
                this.apiKey = !config?.apiKey && config?.jwtToken ? undefined : config?.apiKey;
                this.jwtToken = config?.jwtToken;
                this.organizationId = config?.organizationId;
                apiUrl = config?.apiUrl || config?.serverUrl;
                this.target = config?.target;
            }
            let _envReader;
            const envReader = () => {
                if (_envReader === undefined) {
                    _envReader = Runtime_1.RUNTIME !== Runtime_1.Runtime.BROWSER ? new Runtime_1.DaytonaEnvReader() : null;
                }
                return _envReader;
            };
            if (!config ||
                (!(this.apiKey && apiUrl && this.target) && !(this.jwtToken && this.organizationId && apiUrl && this.target))) {
                const reader = envReader();
                if (reader) {
                    this.apiKey = this.apiKey || (this.jwtToken ? undefined : reader.get('DAYTONA_API_KEY'));
                    this.jwtToken = this.jwtToken || reader.get('DAYTONA_JWT_TOKEN');
                    this.organizationId = this.organizationId || reader.get('DAYTONA_ORGANIZATION_ID');
                    apiUrl = apiUrl || reader.get('DAYTONA_API_URL') || reader.get('DAYTONA_SERVER_URL');
                    this.target = this.target || reader.get('DAYTONA_TARGET');
                    if (reader.get('DAYTONA_SERVER_URL') && !reader.get('DAYTONA_API_URL')) {
                        console.warn('[Deprecation Warning] Environment variable `DAYTONA_SERVER_URL` is deprecated and will be removed in future versions. Use `DAYTONA_API_URL` instead.');
                    }
                }
            }
            this.apiUrl = apiUrl || 'https://app.daytona.io/api';
            if (!this.apiKey && !this.jwtToken) {
                throw new DaytonaError_1.DaytonaAuthenticationError('Authentication credentials not found. Set DAYTONA_API_KEY, or both DAYTONA_JWT_TOKEN and DAYTONA_ORGANIZATION_ID.' +
                    ' These can also be provided via DaytonaConfig.');
            }
            const orgHeader = {};
            if (!this.apiKey) {
                if (!this.organizationId) {
                    throw new DaytonaError_1.DaytonaAuthenticationError('DAYTONA_ORGANIZATION_ID is required when authenticating with DAYTONA_JWT_TOKEN.' +
                        ' It can also be provided via DaytonaConfig.');
                }
                orgHeader['X-Daytona-Organization-ID'] = this.organizationId;
            }
            const isLegacyPackage = packageJson.name === '@daytonaio/sdk';
            const sdkLabel = isLegacyPackage ? 'sdk-typescript-legacy' : 'sdk-typescript';
            const configuration = new api_client_1.Configuration({
                basePath: this.apiUrl,
                baseOptions: {
                    headers: {
                        Authorization: `Bearer ${this.apiKey || this.jwtToken}`,
                        'X-Daytona-Source': sdkLabel,
                        'X-Daytona-SDK-Version': packageJson.version,
                        'User-Agent': `${sdkLabel}/${packageJson.version}`,
                        ...orgHeader,
                    },
                },
            });
            const axiosInstance = Daytona.createAxiosInstance();
            this.sandboxApi = new api_client_1.SandboxApi(configuration, '', axiosInstance);
            this.objectStorageApi = new api_client_1.ObjectStorageApi(configuration, '', axiosInstance);
            this.configApi = new api_client_1.ConfigApi(configuration, '', axiosInstance);
            this.volume = new Volume_1.VolumeService(new api_client_1.VolumesApi(configuration, '', axiosInstance));
            this.secret = new Secret_1.SecretService(new api_client_1.SecretApi(configuration, '', axiosInstance));
            this.snapshot = new Snapshot_1.SnapshotService(configuration, new api_client_1.SnapshotsApi(configuration, '', axiosInstance), this.objectStorageApi, this.target);
            this.clientConfig = configuration;
            const env = envReader();
            const otelEnabled = config?.otelEnabled ||
                config?._experimental?.otelEnabled ||
                env?.get('DAYTONA_OTEL_ENABLED') === 'true' ||
                env?.get('DAYTONA_EXPERIMENTAL_OTEL_ENABLED') === 'true';
            if (!otelEnabled) {
                return;
            }
            const errPrefix = 'OpenTelemetry instrumentation is not supported: ';
            const { diag, DiagConsoleLogger, DiagLogLevel } = (0, Import_1.dynamicRequire)('@opentelemetry/api', errPrefix);
            const { NodeSDK } = (0, Import_1.dynamicRequire)('@opentelemetry/sdk-node', errPrefix);
            const { HttpInstrumentation } = (0, Import_1.dynamicRequire)('@opentelemetry/instrumentation-http', errPrefix);
            const { BatchSpanProcessor } = (0, Import_1.dynamicRequire)('@opentelemetry/sdk-trace-base', errPrefix);
            const { OTLPTraceExporter } = (0, Import_1.dynamicRequire)('@opentelemetry/exporter-trace-otlp-http', errPrefix);
            const { CompressionAlgorithm } = (0, Import_1.dynamicRequire)('@opentelemetry/otlp-exporter-base', errPrefix);
            const { ATTR_SERVICE_NAME, ATTR_SERVICE_VERSION } = (0, Import_1.dynamicRequire)('@opentelemetry/semantic-conventions', errPrefix);
            const { resourceFromAttributes } = (0, Import_1.dynamicRequire)('@opentelemetry/resources', errPrefix);
            diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.INFO);
            this.otelSdk = new NodeSDK({
                resource: resourceFromAttributes({
                    [ATTR_SERVICE_VERSION]: packageJson.version,
                    [ATTR_SERVICE_NAME]: 'daytona-typescript-sdk',
                }),
                instrumentations: [
                    new HttpInstrumentation({
                        requireParentforOutgoingSpans: false,
                    }),
                ],
                spanProcessors: [
                    new BatchSpanProcessor(new OTLPTraceExporter({
                        compression: CompressionAlgorithm.GZIP,
                    })),
                ],
            });
            this.otelSdk.start();
            // Flush and shutdown OTEL on process exit
            process.on('SIGTERM', async () => {
                await this.otelSdk?.shutdown();
            });
        }
        async [Symbol.asyncDispose]() {
            if (!this.otelSdk) {
                return;
            }
            await this.otelSdk.shutdown();
        }
        async create(params, options = { timeout: 60 }) {
            const startTime = Date.now();
            options = typeof options === 'number' ? { timeout: options } : { ...options };
            if (options.timeout == undefined || options.timeout == null) {
                options.timeout = 60;
            }
            if (params == null) {
                params = { language: 'python' };
            }
            if (!params.language) {
                params.language = 'python';
            }
            const labels = params.labels || {};
            if (params.language) {
                const validLanguages = Object.values(CodeLanguage);
                if (!validLanguages.includes(params.language)) {
                    throw new DaytonaError_1.DaytonaValidationError(`Invalid ${exports.CODE_TOOLBOX_LANGUAGE_LABEL}: ${params.language}. Supported languages: ${validLanguages.join(', ')}`);
                }
                labels[exports.CODE_TOOLBOX_LANGUAGE_LABEL] = params.language;
            }
            if (options.timeout < 0) {
                throw new DaytonaError_1.DaytonaValidationError('Timeout must be a non-negative number');
            }
            if (params.autoStopInterval !== undefined &&
                (!Number.isInteger(params.autoStopInterval) || params.autoStopInterval < 0)) {
                throw new DaytonaError_1.DaytonaValidationError('autoStopInterval must be a non-negative integer');
            }
            if (params.ephemeral) {
                if (params.autoDeleteInterval !== undefined && params.autoDeleteInterval !== 0) {
                    console.warn("'ephemeral' and 'autoDeleteInterval' cannot be used together. If ephemeral is true, autoDeleteInterval will be ignored and set to 0.");
                }
                params.autoDeleteInterval = 0;
            }
            if (params.autoArchiveInterval !== undefined &&
                (!Number.isInteger(params.autoArchiveInterval) || params.autoArchiveInterval < 0)) {
                throw new DaytonaError_1.DaytonaValidationError('autoArchiveInterval must be a non-negative integer');
            }
            try {
                let buildInfo;
                let snapshot;
                let resources;
                if ('snapshot' in params) {
                    snapshot = params.snapshot;
                }
                if ('image' in params) {
                    if (typeof params.image === 'string') {
                        buildInfo = {
                            dockerfileContent: Image_1.Image.base(params.image).dockerfile,
                        };
                    }
                    else if (params.image instanceof Image_1.Image) {
                        const contextHashes = await Snapshot_1.SnapshotService.processImageContext(this.objectStorageApi, params.image);
                        buildInfo = {
                            contextHashes,
                            dockerfileContent: params.image.dockerfile,
                        };
                    }
                }
                if ('resources' in params) {
                    resources = params.resources;
                }
                const response = await this.sandboxApi.createSandbox({
                    name: params.name,
                    snapshot: snapshot,
                    buildInfo,
                    user: params.user,
                    env: params.envVars || {},
                    labels: labels,
                    public: params.public,
                    target: this.target,
                    cpu: resources?.cpu,
                    gpu: resources?.gpu,
                    gpuType: resources?.gpuType === undefined
                        ? undefined
                        : Array.isArray(resources.gpuType)
                            ? resources.gpuType
                            : [resources.gpuType],
                    memory: resources?.memory,
                    disk: resources?.disk,
                    autoStopInterval: params.autoStopInterval,
                    autoArchiveInterval: params.autoArchiveInterval,
                    autoDeleteInterval: params.autoDeleteInterval,
                    volumes: params.volumes,
                    networkBlockAll: params.networkBlockAll,
                    networkAllowList: params.networkAllowList,
                    domainAllowList: params.domainAllowList,
                    linkedSandbox: params.linkedSandbox,
                    secrets: params.secrets
                        ? Object.entries(params.secrets).map(([envVar, secretName]) => ({ [envVar]: secretName }))
                        : undefined,
                }, undefined, {
                    timeout: options.timeout * 1000,
                });
                let sandboxInstance = response.data;
                if (sandboxInstance.state === api_client_1.SandboxState.PENDING_BUILD && options.onSnapshotCreateLogs) {
                    const terminalStates = [
                        api_client_1.SandboxState.STARTED,
                        api_client_1.SandboxState.STARTING,
                        api_client_1.SandboxState.ERROR,
                        api_client_1.SandboxState.BUILD_FAILED,
                    ];
                    while (sandboxInstance.state === api_client_1.SandboxState.PENDING_BUILD) {
                        if (options.timeout) {
                            const elapsed = (Date.now() - startTime) / 1000;
                            if (elapsed > options.timeout) {
                                throw new DaytonaError_1.DaytonaTimeoutError(`Sandbox build has been pending for more than ${options.timeout} seconds. Please check the sandbox state again later.`);
                            }
                        }
                        await new Promise((resolve) => setTimeout(resolve, 1000));
                        sandboxInstance = (await this.sandboxApi.getSandbox(sandboxInstance.id)).data;
                    }
                    const response = await this.sandboxApi.getBuildLogsUrl(sandboxInstance.id);
                    await (0, Stream_1.processStreamingResponse)(() => fetch(response.data.url + '?follow=true', {
                        method: 'GET',
                        headers: this.clientConfig.baseOptions.headers,
                    }), (chunk) => options.onSnapshotCreateLogs?.(chunk.trimEnd()), async () => {
                        sandboxInstance = (await this.sandboxApi.getSandbox(sandboxInstance.id)).data;
                        return sandboxInstance.state !== undefined && terminalStates.includes(sandboxInstance.state);
                    });
                }
                const sandbox = new Sandbox_1.Sandbox(sandboxInstance, new api_client_1.Configuration(structuredClone(this.clientConfig)), Daytona.createAxiosInstance(), this.sandboxApi);
                if (sandbox.state !== 'started') {
                    const timeElapsed = Date.now() - startTime;
                    await sandbox.waitUntilStarted(options.timeout ? Math.max(0.001, options.timeout - timeElapsed / 1000) : options.timeout);
                }
                return sandbox;
            }
            catch (error) {
                if (error instanceof DaytonaError_1.DaytonaTimeoutError) {
                    const errMsg = `Failed to create and start sandbox within ${options.timeout} seconds. Operation timed out.`;
                    throw new DaytonaError_1.DaytonaTimeoutError(errMsg, error.statusCode, error.headers, error.errorCode);
                }
                throw error;
            }
        }
        /**
         * Gets a Sandbox by its ID or name.
         *
         * @param {string} sandboxIdOrName - The ID or name of the Sandbox to retrieve
         * @returns {Promise<Sandbox>} The Sandbox
         *
         * @example
         * const sandbox = await daytona.get('my-sandbox-id-or-name');
         * console.log(`Sandbox state: ${sandbox.state}`);
         */
        async get(sandboxIdOrName) {
            const response = await this.sandboxApi.getSandbox(sandboxIdOrName);
            const sandboxInstance = response.data;
            return new Sandbox_1.Sandbox(sandboxInstance, structuredClone(this.clientConfig), Daytona.createAxiosInstance(), this.sandboxApi);
        }
        /**
         * Iterates over Sandboxes matching the given query.
         *
         * @param {ListSandboxesQuery} [query] - Optional filters, sorting, and per-page size.
         * @returns {AsyncIterableIterator<Sandbox>}
         *
         * @example
         * for await (const sandbox of daytona.list({ labels: { env: 'dev' } })) {
         *   console.log(sandbox.id)
         * }
         */
        list(query) {
            const { sandboxApi, clientConfig } = this;
            const tracer = api_1.trace.getTracer('');
            async function* generator() {
                let cursor = undefined;
                let firstPage = true;
                while (true) {
                    // Stop only when we've fetched at least one page and there's no further cursor.
                    if (!firstPage && !cursor) {
                        break;
                    }
                    const span = tracer.startSpan('Daytona.list.fetchPage', {
                        attributes: {
                            component: 'Daytona',
                            method: 'list',
                        },
                    }, api_1.context.active());
                    let response;
                    try {
                        response = await api_1.context.with(api_1.trace.setSpan(api_1.context.active(), span), () => sandboxApi.listSandboxes(undefined, cursor, query?.limit, query?.id, query?.name, query?.labels ? JSON.stringify(query.labels) : undefined, undefined, query?.states, query?.snapshots, query?.targets, undefined, query?.minCpu, query?.maxCpu, query?.minMemoryGib, query?.maxMemoryGib, query?.minDiskGib, query?.maxDiskGib, query?.isPublic, query?.isRecoverable, query?.createdAtAfter, query?.createdAtBefore, query?.lastActivityAfter, query?.lastActivityBefore, query?.sort, query?.order));
                        span.setStatus({ code: api_1.SpanStatusCode.OK });
                    }
                    catch (error) {
                        span.setStatus({
                            code: api_1.SpanStatusCode.ERROR,
                            message: error instanceof Error ? error.message : String(error),
                        });
                        span.recordException(error instanceof Error ? error : new Error(String(error)));
                        throw error;
                    }
                    finally {
                        span.end();
                    }
                    firstPage = false;
                    for (const sandbox of response.data.items) {
                        // Sandbox ctor mutates clientConfig.basePath — clone per item.
                        yield new Sandbox_1.Sandbox(sandbox, structuredClone(clientConfig), Daytona.createAxiosInstance(), sandboxApi);
                    }
                    cursor = response.data.nextCursor ?? undefined;
                    if (!cursor) {
                        break;
                    }
                }
            }
            return generator();
        }
        /**
         * Starts a Sandbox and waits for it to be ready.
         *
         * @param {Sandbox} sandbox - The Sandbox to start
         * @param {number} [timeout] - Optional timeout in seconds (0 means no timeout)
         * @returns {Promise<void>}
         *
         * @example
         * const sandbox = await daytona.get('my-sandbox-id');
         * // Wait up to 60 seconds for the sandbox to start
         * await daytona.start(sandbox, 60);
         */
        async start(sandbox, timeout) {
            await sandbox.start(timeout);
        }
        /**
         * Stops a Sandbox.
         *
         * @param {Sandbox} sandbox - The Sandbox to stop
         * @returns {Promise<void>}
         *
         * @example
         * const sandbox = await daytona.get('my-sandbox-id');
         * await daytona.stop(sandbox);
         */
        async stop(sandbox) {
            await sandbox.stop();
        }
        /**
         * Forks a Sandbox, creating a new Sandbox with an identical filesystem.
         *
         * @param {Sandbox} sandbox - The Sandbox to fork
         * @param {object} [params] - Fork parameters
         * @param {string} [params.name] - Optional name for the forked Sandbox
         * @param {number} [timeout] - Timeout in seconds (0 means no timeout, default is 60)
         * @returns {Promise<Sandbox>} The forked Sandbox
         *
         * @example
         * const sandbox = await daytona.get('my-sandbox-id');
         * const forked = await daytona._experimental_fork(sandbox, { name: 'my-fork' });
         * console.log(`Forked sandbox: ${forked.id}`);
         */
        async _experimental_fork(sandbox, params, timeout = 60) {
            return await sandbox._experimental_fork(params, timeout);
        }
        /**
         * Deletes a Sandbox.
         *
         * @param {Sandbox} sandbox - The Sandbox to delete
         * @param {number} timeout - Timeout in seconds (0 means no timeout, default is 60)
         * @returns {Promise<void>}
         *
         * @example
         * const sandbox = await daytona.get('my-sandbox-id');
         * await daytona.delete(sandbox);
         */
        async delete(sandbox, timeout = 60) {
            await sandbox.delete(timeout);
        }
        /**
         * @hidden
         */
        static createAxiosInstance() {
            const axiosInstance = axios_1.default.create({
                timeout: 24 * 60 * 60 * 1000, // 24 hours
            });
            // Request interceptor: Inject trace context into headers
            axiosInstance.interceptors.request.use((requestConfig) => {
                // Get the current active context (which may contain an active span)
                const currentContext = api_1.context.active();
                // Inject trace context into HTTP headers using W3C Trace Context propagation
                // This adds headers like 'traceparent' and 'tracestate'
                api_1.propagation.inject(currentContext, requestConfig.headers);
                requestConfig.metadata = { startTime: Date.now() };
                return requestConfig;
            }, (error) => {
                return Promise.reject(error);
            });
            axiosInstance.interceptors.response.use((response) => {
                return response;
            }, (error) => {
                if (error instanceof axios_1.AxiosError) {
                    throw (0, DaytonaError_1.createAxiosDaytonaError)(error);
                }
                throw new DaytonaError_1.DaytonaError(error instanceof Error ? error.message : String(error));
            });
            axiosInstance.interceptors.response.use((response) => {
                const startTime = response.config.metadata?.startTime;
                if (startTime) {
                    const duration = Date.now() - startTime;
                    // Get the active span to add attributes
                    const activeSpan = api_1.trace.getActiveSpan();
                    // Only modify the span if it's still recording (not ended)
                    if (activeSpan && activeSpan.isRecording()) {
                        // Add response metadata to the span
                        activeSpan.setAttributes({
                            'http.response.status_code': response.status,
                            'http.response.duration_ms': duration,
                            // 'http.response.size_bytes': JSON.stringify(response.data).length,
                        });
                    }
                }
                return response;
            }, (error) => {
                const startTime = error.config?.metadata?.startTime;
                if (startTime) {
                    const duration = Date.now() - startTime;
                    // Get the active span to record the error
                    const activeSpan = api_1.trace.getActiveSpan();
                    // Only modify the span if it's still recording (not ended)
                    if (activeSpan && activeSpan.isRecording()) {
                        activeSpan.setStatus({
                            code: api_1.SpanStatusCode.ERROR,
                            message: error.message,
                        });
                        activeSpan.setAttributes({
                            'http.response.duration_ms': duration,
                            'error.type': error.name,
                            'error.message': error.message,
                        });
                        if (error.response) {
                            activeSpan.setAttribute('http.response.status_code', error.response.status);
                        }
                        // Record the exception on the span
                        activeSpan.recordException(error);
                    }
                }
                return Promise.reject(error);
            });
            return axiosInstance;
        }
    };
})();
exports.Daytona = Daytona;
