// @daytona/sdk/cjs/Git.js
"use strict";
/*
 * Copyright Daytona Platforms Inc.
 * SPDX-License-Identifier: Apache-2.0
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Git = void 0;
const tslib_1 = require("tslib");
const otel_decorator_1 = require("./utils/otel.decorator");
/**
 * Provides Git operations within a Sandbox.
 *
 * @class
 */
let Git = (() => {
    let _instanceExtraInitializers = [];
    let _add_decorators;
    let _branches_decorators;
    let _createBranch_decorators;
    let _deleteBranch_decorators;
    let _checkoutBranch_decorators;
    let _clone_decorators;
    let _commit_decorators;
    let _push_decorators;
    let _pull_decorators;
    let _status_decorators;
    return class Git {
        static {
            const _metadata = typeof Symbol === "function" && Symbol.metadata ? Object.create(null) : void 0;
            _add_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            _branches_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            _createBranch_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            _deleteBranch_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            _checkoutBranch_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            _clone_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            _commit_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            _push_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            _pull_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            _status_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            tslib_1.__esDecorate(this, null, _add_decorators, { kind: "method", name: "add", static: false, private: false, access: { has: obj => "add" in obj, get: obj => obj.add }, metadata: _metadata }, null, _instanceExtraInitializers);
            tslib_1.__esDecorate(this, null, _branches_decorators, { kind: "method", name: "branches", static: false, private: false, access: { has: obj => "branches" in obj, get: obj => obj.branches }, metadata: _metadata }, null, _instanceExtraInitializers);
            tslib_1.__esDecorate(this, null, _createBranch_decorators, { kind: "method", name: "createBranch", static: false, private: false, access: { has: obj => "createBranch" in obj, get: obj => obj.createBranch }, metadata: _metadata }, null, _instanceExtraInitializers);
            tslib_1.__esDecorate(this, null, _deleteBranch_decorators, { kind: "method", name: "deleteBranch", static: false, private: false, access: { has: obj => "deleteBranch" in obj, get: obj => obj.deleteBranch }, metadata: _metadata }, null, _instanceExtraInitializers);
            tslib_1.__esDecorate(this, null, _checkoutBranch_decorators, { kind: "method", name: "checkoutBranch", static: false, private: false, access: { has: obj => "checkoutBranch" in obj, get: obj => obj.checkoutBranch }, metadata: _metadata }, null, _instanceExtraInitializers);
            tslib_1.__esDecorate(this, null, _clone_decorators, { kind: "method", name: "clone", static: false, private: false, access: { has: obj => "clone" in obj, get: obj => obj.clone }, metadata: _metadata }, null, _instanceExtraInitializers);
            tslib_1.__esDecorate(this, null, _commit_decorators, { kind: "method", name: "commit", static: false, private: false, access: { has: obj => "commit" in obj, get: obj => obj.commit }, metadata: _metadata }, null, _instanceExtraInitializers);
            tslib_1.__esDecorate(this, null, _push_decorators, { kind: "method", name: "push", static: false, private: false, access: { has: obj => "push" in obj, get: obj => obj.push }, metadata: _metadata }, null, _instanceExtraInitializers);
            tslib_1.__esDecorate(this, null, _pull_decorators, { kind: "method", name: "pull", static: false, private: false, access: { has: obj => "pull" in obj, get: obj => obj.pull }, metadata: _metadata }, null, _instanceExtraInitializers);
            tslib_1.__esDecorate(this, null, _status_decorators, { kind: "method", name: "status", static: false, private: false, access: { has: obj => "status" in obj, get: obj => obj.status }, metadata: _metadata }, null, _instanceExtraInitializers);
            if (_metadata) Object.defineProperty(this, Symbol.metadata, { enumerable: true, configurable: true, writable: true, value: _metadata });
        }
        apiClient = tslib_1.__runInitializers(this, _instanceExtraInitializers);
        constructor(apiClient) {
            this.apiClient = apiClient;
        }
        /**
         * Stages the specified files for the next commit, similar to
         * running 'git add' on the command line.
         *
         * @param {string} path - Path to the Git repository root. Relative paths are resolved based on the sandbox working directory.
         * @param {string[]} files - List of file paths or directories to stage, relative to the repository root
         * @returns {Promise<void>}
         *
         * @example
         * // Stage a single file
         * await git.add('workspace/repo', ['file.txt']);
         *
         * @example
         * // Stage whole repository
         * await git.add('workspace/repo', ['.']);
         */
        async add(path, files) {
            await this.apiClient.addFiles({
                path,
                files,
            });
        }
        /**
         * List branches in the repository.
         *
         * @param {string} path - Path to the Git repository root. Relative paths are resolved based on the sandbox working directory.
         * @returns {Promise<ListBranchResponse>} List of branches in the repository
         *
         * @example
         * const response = await git.branches('workspace/repo');
         * console.log(`Branches: ${response.branches}`);
         */
        async branches(path) {
            const response = await this.apiClient.listBranches(path);
            return response.data;
        }
        /**
         * Create branch in the repository.
         *
         * @param {string} path - Path to the Git repository root. Relative paths are resolved based on the sandbox working directory.
         * @param {string} name - Name of the new branch to create
         * @returns {Promise<void>}
         *
         * @example
         * await git.createBranch('workspace/repo', 'new-feature');
         */
        async createBranch(path, name) {
            await this.apiClient.createBranch({
                path,
                name,
            });
            return;
        }
        /**
         * Delete branche in the repository.
         *
         * @param {string} path - Path to the Git repository root. Relative paths are resolved based on the sandbox working directory.
         * @param {string} name - Name of the branch to delete
         * @returns {Promise<void>}
         *
         * @example
         * await git.deleteBranch('workspace/repo', 'new-feature');
         */
        async deleteBranch(path, name) {
            await this.apiClient.deleteBranch({
                path,
                name,
            });
            return;
        }
        /**
         * Checkout branche in the repository.
         *
         * @param {string} path - Path to the Git repository root. Relative paths are resolved based on the sandbox working directory.
         * @param {string} branch - Name of the branch to checkout
         * @returns {Promise<void>}
         *
         * @example
         * await git.checkoutBranch('workspace/repo', 'new-feature');
         */
        async checkoutBranch(path, branch) {
            await this.apiClient.checkoutBranch({
                path,
                branch,
            });
            return;
        }
        /**
         * Clones a Git repository into the specified path. It supports
         * cloning specific branches or commits, and can authenticate with the remote
         * repository if credentials are provided.
         *
         * @param {string} url - Repository URL to clone from
         * @param {string} path - Path where the repository should be cloned. Relative paths are resolved based on the sandbox working directory.
         * @param {string} [branch] - Specific branch to clone. If not specified, clones the default branch
         * @param {string} [commitId] - Specific commit to clone. If specified, the repository will be left in a detached HEAD state at this commit
         * @param {string} [username] - Git username for authentication
         * @param {string} [password] - Git password or token for authentication
         * @param {boolean} [insecureSkipTls] - Skip TLS certificate verification (insecure). Use only for trusted internal Git servers with self-signed or private-CA certs.
         * @returns {Promise<void>}
         *
         * @example
         * // Clone the default branch
         * await git.clone(
         *   'https://github.com/user/repo.git',
         *   'workspace/repo'
         * );
         *
         * @example
         * // Clone a specific branch with authentication
         * await git.clone(
         *   'https://github.com/user/private-repo.git',
         *   'workspace/private',
         *   branch='develop',
         *   username='user',
         *   password='token'
         * );
         *
         * @example
         * // Clone a specific commit
         * await git.clone(
         *   'https://github.com/user/repo.git',
         *   'workspace/repo-old',
         *   commitId='abc123'
         * );
         */
        async clone(url, path, branch, commitId, username, password, insecureSkipTls) {
            await this.apiClient.cloneRepository({
                url: url,
                branch: branch,
                path,
                username,
                password,
                commit_id: commitId,
                insecure_skip_tls: insecureSkipTls,
            });
        }
        /**
         * Commits staged changes.
         *
         * @param {string} path - Path to the Git repository root. Relative paths are resolved based on the sandbox working directory.
         * @param {string} message - Commit message describing the changes
         * @param {string} author - Name of the commit author
         * @param {string} email - Email address of the commit author
         * @param {boolean} [allowEmpty] - Allow creating an empty commit when no changes are staged
         * @returns {Promise<void>}
         *
         * @example
         * // Stage and commit changes
         * await git.add('workspace/repo', ['README.md']);
         * await git.commit(
         *   'workspace/repo',
         *   'Update documentation',
         *   'John Doe',
         *   'john@example.com',
         *   true
         * );
         *
         */
        async commit(path, message, author, email, allowEmpty) {
            const response = await this.apiClient.commitChanges({
                path,
                message,
                author,
                email,
                allow_empty: allowEmpty,
            });
            return {
                sha: response.data.hash,
            };
        }
        /**
         * Push local changes to the remote repository.
         *
         * @param {string} path - Path to the Git repository root. Relative paths are resolved based on the sandbox working directory.
         * @param {string} [username] - Git username for authentication
         * @param {string} [password] - Git password or token for authentication
         * @returns {Promise<void>}
         *
         * @example
         * // Push to a public repository
         * await git.push('workspace/repo');
         *
         * @example
         * // Push to a private repository
         * await git.push(
         *   'workspace/repo',
         *   'user',
         *   'token'
         * );
         */
        async push(path, username, password) {
            await this.apiClient.pushChanges({
                path,
                username,
                password,
            });
        }
        /**
         * Pulls changes from the remote repository.
         *
         * @param {string} path - Path to the Git repository root. Relative paths are resolved based on the sandbox working directory.
         * @param {string} [username] - Git username for authentication
         * @param {string} [password] - Git password or token for authentication
         * @returns {Promise<void>}
         *
         * @example
         * // Pull from a public repository
         * await git.pull('workspace/repo');
         *
         * @example
         * // Pull from a private repository
         * await git.pull(
         *   'workspace/repo',
         *   'user',
         *   'token'
         * );
         */
        async pull(path, username, password) {
            await this.apiClient.pullChanges({
                path,
                username,
                password,
            });
        }
        /**
         * Gets the current status of the Git repository.
         *
         * @param {string} path - Path to the Git repository root. Relative paths are resolved based on the sandbox working directory.
         * @returns {Promise<GitStatus>} Current repository status including:
         *                               - currentBranch: Name of the current branch
         *                               - ahead: Number of commits ahead of the remote branch
         *                               - behind: Number of commits behind the remote branch
         *                               - branchPublished: Whether the branch has been published to the remote repository
         *                               - fileStatus: List of file statuses
         *
         * @example
         * const status = await sandbox.git.status('workspace/repo');
         * console.log(`Current branch: ${status.currentBranch}`);
         * console.log(`Commits ahead: ${status.ahead}`);
         * console.log(`Commits behind: ${status.behind}`);
         */
        async status(path) {
            const response = await this.apiClient.getStatus(path);
            return response.data;
        }
    };
})();
exports.Git = Git;
