// @daytona/sdk/cjs/LspServer.js
"use strict";
/*
 * Copyright Daytona Platforms Inc.
 * SPDX-License-Identifier: Apache-2.0
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LspServer = exports.LspLanguageId = void 0;
const tslib_1 = require("tslib");
const DaytonaError_1 = require("./errors/DaytonaError");
const otel_decorator_1 = require("./utils/otel.decorator");
/**
 * Supported language server types.
 */
var LspLanguageId;
(function (LspLanguageId) {
    LspLanguageId["PYTHON"] = "python";
    LspLanguageId["TYPESCRIPT"] = "typescript";
    LspLanguageId["JAVASCRIPT"] = "javascript";
})(LspLanguageId || (exports.LspLanguageId = LspLanguageId = {}));
/**
 * Provides Language Server Protocol functionality for code intelligence to provide
 * IDE-like features such as code completion, symbol search, and more.
 *
 * @property {LspLanguageId} languageId - The language server type (e.g., "typescript")
 * @property {string} pathToProject - Path to the project root directory. Relative paths are resolved based on the sandbox working directory.
 * @property {LspApi} apiClient - API client for Sandbox lsp operations
 * @property {SandboxInstance} instance - The Sandbox instance this server belongs to
 *
 * @class
 */
let LspServer = (() => {
    let _instanceExtraInitializers = [];
    let _start_decorators;
    let _stop_decorators;
    let _didOpen_decorators;
    let _didClose_decorators;
    let _documentSymbols_decorators;
    let _workspaceSymbols_decorators;
    let _sandboxSymbols_decorators;
    let _completions_decorators;
    return class LspServer {
        static {
            const _metadata = typeof Symbol === "function" && Symbol.metadata ? Object.create(null) : void 0;
            _start_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            _stop_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            _didOpen_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            _didClose_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            _documentSymbols_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            _workspaceSymbols_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            _sandboxSymbols_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            _completions_decorators = [(0, otel_decorator_1.WithInstrumentation)()];
            tslib_1.__esDecorate(this, null, _start_decorators, { kind: "method", name: "start", static: false, private: false, access: { has: obj => "start" in obj, get: obj => obj.start }, metadata: _metadata }, null, _instanceExtraInitializers);
            tslib_1.__esDecorate(this, null, _stop_decorators, { kind: "method", name: "stop", static: false, private: false, access: { has: obj => "stop" in obj, get: obj => obj.stop }, metadata: _metadata }, null, _instanceExtraInitializers);
            tslib_1.__esDecorate(this, null, _didOpen_decorators, { kind: "method", name: "didOpen", static: false, private: false, access: { has: obj => "didOpen" in obj, get: obj => obj.didOpen }, metadata: _metadata }, null, _instanceExtraInitializers);
            tslib_1.__esDecorate(this, null, _didClose_decorators, { kind: "method", name: "didClose", static: false, private: false, access: { has: obj => "didClose" in obj, get: obj => obj.didClose }, metadata: _metadata }, null, _instanceExtraInitializers);
            tslib_1.__esDecorate(this, null, _documentSymbols_decorators, { kind: "method", name: "documentSymbols", static: false, private: false, access: { has: obj => "documentSymbols" in obj, get: obj => obj.documentSymbols }, metadata: _metadata }, null, _instanceExtraInitializers);
            tslib_1.__esDecorate(this, null, _workspaceSymbols_decorators, { kind: "method", name: "workspaceSymbols", static: false, private: false, access: { has: obj => "workspaceSymbols" in obj, get: obj => obj.workspaceSymbols }, metadata: _metadata }, null, _instanceExtraInitializers);
            tslib_1.__esDecorate(this, null, _sandboxSymbols_decorators, { kind: "method", name: "sandboxSymbols", static: false, private: false, access: { has: obj => "sandboxSymbols" in obj, get: obj => obj.sandboxSymbols }, metadata: _metadata }, null, _instanceExtraInitializers);
            tslib_1.__esDecorate(this, null, _completions_decorators, { kind: "method", name: "completions", static: false, private: false, access: { has: obj => "completions" in obj, get: obj => obj.completions }, metadata: _metadata }, null, _instanceExtraInitializers);
            if (_metadata) Object.defineProperty(this, Symbol.metadata, { enumerable: true, configurable: true, writable: true, value: _metadata });
        }
        languageId = tslib_1.__runInitializers(this, _instanceExtraInitializers);
        pathToProject;
        apiClient;
        constructor(languageId, pathToProject, apiClient) {
            this.languageId = languageId;
            this.pathToProject = pathToProject;
            this.apiClient = apiClient;
            if (!Object.values(LspLanguageId).includes(this.languageId)) {
                throw new DaytonaError_1.DaytonaValidationError(`Invalid languageId: ${this.languageId}. Supported values are: ${Object.values(LspLanguageId).join(', ')}`);
            }
        }
        /**
         * Starts the language server, must be called before using any other LSP functionality.
         * It initializes the language server for the specified language and project.
         *
         * @returns {Promise<void>}
         *
         * @example
         * const lsp = await sandbox.createLspServer('typescript', 'workspace/project');
         * await lsp.start();  // Initialize the server
         * // Now ready for LSP operations
         */
        async start() {
            await this.apiClient.start({
                languageId: this.languageId,
                pathToProject: this.pathToProject,
            });
        }
        /**
         * Stops the language server, should be called when the LSP server is no longer needed to
         * free up system resources.
         *
         * @returns {Promise<void>}
         *
         * @example
         * // When done with LSP features
         * await lsp.stop();  // Clean up resources
         */
        async stop() {
            await this.apiClient.stop({
                languageId: this.languageId,
                pathToProject: this.pathToProject,
            });
        }
        /**
         * Notifies the language server that a file has been opened, enabling
         * language features like diagnostics and completions for that file. The server
         * will begin tracking the file's contents and providing language features.
         *
         * @param {string} path - Path to the opened file. Relative paths are resolved based on the sandbox working directory.
         * @returns {Promise<void>}
         *
         * @example
         * // When opening a file for editing
         * await lsp.didOpen('workspace/project/src/index.ts');
         * // Now can get completions, symbols, etc. for this file
         */
        async didOpen(path) {
            await this.apiClient.didOpen({
                languageId: this.languageId,
                pathToProject: this.pathToProject,
                uri: 'file://' + path,
            });
        }
        /**
         * Notifies the language server that a file has been closed, should be called when a file is closed
         * in the editor to allow the language server to clean up any resources associated with that file.
         *
         * @param {string} path - Path to the closed file. Relative paths are resolved based on the project path
         * set in the LSP server constructor.
         * @returns {Promise<void>}
         *
         * @example
         * // When done editing a file
         * await lsp.didClose('workspace/project/src/index.ts');
         */
        async didClose(path) {
            await this.apiClient.didClose({
                languageId: this.languageId,
                pathToProject: this.pathToProject,
                uri: 'file://' + path,
            });
        }
        /**
         * Get symbol information (functions, classes, variables, etc.) from a document.
         *
         * @param {string} path - Path to the file to get symbols from. Relative paths are resolved based on the project path
         * set in the LSP server constructor.
         * @returns {Promise<LspSymbol[]>} List of symbols in the document. Each symbol includes:
         *                                 - name: The symbol's name
         *                                 - kind: The symbol's kind (function, class, variable, etc.)
         *                                 - location: The location of the symbol in the file
         *
         * @example
         * // Get all symbols in a file
         * const symbols = await lsp.documentSymbols('workspace/project/src/index.ts');
         * symbols.forEach(symbol => {
         *   console.log(`${symbol.kind} ${symbol.name}: ${symbol.location}`);
         * });
         */
        async documentSymbols(path) {
            const response = await this.apiClient.documentSymbols(this.languageId, this.pathToProject, 'file://' + path);
            return response.data;
        }
        /**
         * Searches for symbols matching the query string across the entire Sandbox.
         *
         * @param {string} query - Search query to match against symbol names
         * @returns {Promise<LspSymbol[]>} List of matching symbols from all files. Each symbol includes:
         *                                 - name: The symbol's name
         *                                 - kind: The symbol's kind (function, class, variable, etc.)
         *                                 - location: The location of the symbol in the file
         *
         * @deprecated Use `sandboxSymbols` instead. This method will be removed in a future version.
         */
        async workspaceSymbols(query) {
            return await this.sandboxSymbols(query);
        }
        /**
         * Searches for symbols matching the query string across the entire Sandbox.
         *
         * @param {string} query - Search query to match against symbol names
         * @returns {Promise<LspSymbol[]>} List of matching symbols from all files. Each symbol includes:
         *                                 - name: The symbol's name
         *                                 - kind: The symbol's kind (function, class, variable, etc.)
         *                                 - location: The location of the symbol in the file
         *
         * @example
         * // Search for all symbols containing "User"
         * const symbols = await lsp.sandboxSymbols('User');
         * symbols.forEach(symbol => {
         *   console.log(`${symbol.name} (${symbol.kind}) in ${symbol.location}`);
         * });
         */
        async sandboxSymbols(query) {
            const response = await this.apiClient.workspaceSymbols(query, this.languageId, this.pathToProject);
            return response.data;
        }
        /**
         * Gets completion suggestions at a position in a file.
         *
         * @param {string} path - Path to the file. Relative paths are resolved based on the project path
         * set in the LSP server constructor.
         * @param {Position} position - The position in the file where completion was requested
         * @returns {Promise<CompletionList>} List of completion suggestions. The list includes:
         *                                    - isIncomplete: Whether more items might be available
         *                                    - items: List of completion items, each containing:
         *                                      - label: The text to insert
         *                                      - kind: The kind of completion
         *                                      - detail: Additional details about the item
         *                                      - documentation: Documentation for the item
         *                                      - sortText: Text used to sort the item in the list
         *                                      - filterText: Text used to filter the item
         *                                      - insertText: The actual text to insert (if different from label)
         *
         * @example
         * // Get completions at a specific position
         * const completions = await lsp.completions('workspace/project/src/index.ts', {
         *   line: 10,
         *   character: 15
         * });
         * completions.items.forEach(item => {
         *   console.log(`${item.label} (${item.kind}): ${item.detail}`);
         * });
         */
        async completions(path, position) {
            const response = await this.apiClient.completions({
                languageId: this.languageId,
                pathToProject: this.pathToProject,
                uri: 'file://' + path,
                position: {
                    line: position.line,
                    character: position.character,
                },
            });
            return response.data;
        }
    };
})();
exports.LspServer = LspServer;
