// @daytona/sdk/cjs/types/Charts.js
"use strict";
/*
 * Copyright Daytona Platforms Inc.
 * SPDX-License-Identifier: Apache-2.0
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChartType = void 0;
exports.parseChart = parseChart;
var ChartType;
(function (ChartType) {
    ChartType["LINE"] = "line";
    ChartType["SCATTER"] = "scatter";
    ChartType["BAR"] = "bar";
    ChartType["PIE"] = "pie";
    ChartType["BOX_AND_WHISKER"] = "box_and_whisker";
    ChartType["COMPOSITE_CHART"] = "composite_chart";
    ChartType["UNKNOWN"] = "unknown";
})(ChartType || (exports.ChartType = ChartType = {}));
function parseChart(chart) {
    switch (chart.type) {
        case ChartType.LINE:
            return chart;
        case ChartType.SCATTER:
            return chart;
        case ChartType.BAR:
            return chart;
        case ChartType.PIE:
            return chart;
        case ChartType.BOX_AND_WHISKER:
            return chart;
        case ChartType.COMPOSITE_CHART:
            return chart;
        default:
            return chart;
    }
}
