// @daytona/toolbox-api-client/src/api/computer-use-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Daytona Toolbox API
 * Daytona Toolbox API. The base URL comes from the sandbox\'s `toolboxProxyUrl` field (returned in sandbox DTO by the main Daytona API) plus the sandbox ID: `{toolboxProxyUrl}/{sandboxId}/{endpoint}`. Default for Daytona Cloud: `https://proxy.app.daytona.io/toolbox/{sandboxId}`.
 *
 * The version of the OpenAPI document: v0.0.0-dev
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ComputerUseApi = exports.ComputerUseApiFactory = exports.ComputerUseApiFp = exports.ComputerUseApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * ComputerUseApi - axios parameter creator
 */
const ComputerUseApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Click the mouse button at the specified coordinates
         * @summary Click mouse button
         * @param {MouseClickRequest} request Mouse click request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        click: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('click', 'request', request);
            const localVarPath = `/computeruse/mouse/click`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Delete a recording file by ID
         * @summary Delete a recording
         * @param {string} id Recording ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteRecording: async (id, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('deleteRecording', 'id', id);
            const localVarPath = `/computeruse/recordings/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'DELETE', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = '*/*';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Download a recording by providing its ID
         * @summary Download a recording
         * @param {string} id Recording ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        downloadRecording: async (id, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('downloadRecording', 'id', id);
            const localVarPath = `/computeruse/recordings/{id}/download`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/octet-stream';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Drag the mouse from start to end coordinates
         * @summary Drag mouse
         * @param {MouseDragRequest} request Mouse drag request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        drag: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('drag', 'request', request);
            const localVarPath = `/computeruse/mouse/drag`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Search the AT-SPI tree for nodes matching a role/name/state filter and return a flat list.
         * @summary Find accessibility nodes
         * @param {FindAccessibilityNodesRequest} request Find request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        findAccessibilityNodes: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('findAccessibilityNodes', 'request', request);
            const localVarPath = `/computeruse/a11y/find`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Move keyboard focus to the AT-SPI node identified by id (bus-name:object-path).
         * @summary Focus an accessibility node
         * @param {AccessibilityNodeRequest} request Node focus request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        focusAccessibilityNode: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('focusAccessibilityNode', 'request', request);
            const localVarPath = `/computeruse/a11y/node/focus`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Fetch the AT-SPI accessibility tree for the focused application, a specific PID, or all registered applications.
         * @summary Get accessibility tree
         * @param {string} [scope] Scope: focused | pid | all (default: focused)
         * @param {number} [pid] Process ID when scope&#x3D;pid
         * @param {number} [maxDepth] Max tree depth (-1 unbounded, 0 root only; default -1)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getAccessibilityTree: async (scope, pid, maxDepth, options = {}) => {
            const localVarPath = `/computeruse/a11y/tree`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (scope !== undefined) {
                localVarQueryParameter['scope'] = scope;
            }
            if (pid !== undefined) {
                localVarQueryParameter['pid'] = pid;
            }
            if (maxDepth !== undefined) {
                localVarQueryParameter['maxDepth'] = maxDepth;
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get the status of all computer use processes
         * @summary Get computer use process status
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getComputerUseStatus: async (options = {}) => {
            const localVarPath = `/computeruse/process-status`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get the current status of the computer use system
         * @summary Get computer use status
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getComputerUseSystemStatus: async (options = {}) => {
            const localVarPath = `/computeruse/status`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get information about all available displays
         * @summary Get display information
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getDisplayInfo: async (options = {}) => {
            const localVarPath = `/computeruse/display/info`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get the current mouse cursor position
         * @summary Get mouse position
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMousePosition: async (options = {}) => {
            const localVarPath = `/computeruse/mouse/position`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get errors for a specific computer use process
         * @summary Get process errors
         * @param {string} processName Process name to get errors for
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getProcessErrors: async (processName, options = {}) => {
            // verify required parameter 'processName' is not null or undefined
            (0, common_1.assertParamExists)('getProcessErrors', 'processName', processName);
            const localVarPath = `/computeruse/process/{processName}/errors`
                .replace(`{${"processName"}}`, encodeURIComponent(String(processName)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get logs for a specific computer use process
         * @summary Get process logs
         * @param {string} processName Process name to get logs for
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getProcessLogs: async (processName, options = {}) => {
            // verify required parameter 'processName' is not null or undefined
            (0, common_1.assertParamExists)('getProcessLogs', 'processName', processName);
            const localVarPath = `/computeruse/process/{processName}/logs`
                .replace(`{${"processName"}}`, encodeURIComponent(String(processName)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Check if a specific computer use process is running
         * @summary Get specific process status
         * @param {string} processName Process name to check
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getProcessStatus: async (processName, options = {}) => {
            // verify required parameter 'processName' is not null or undefined
            (0, common_1.assertParamExists)('getProcessStatus', 'processName', processName);
            const localVarPath = `/computeruse/process/{processName}/status`
                .replace(`{${"processName"}}`, encodeURIComponent(String(processName)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get details of a specific recording by ID
         * @summary Get recording details
         * @param {string} id Recording ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getRecording: async (id, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('getRecording', 'id', id);
            const localVarPath = `/computeruse/recordings/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get information about all open windows
         * @summary Get windows information
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getWindows: async (options = {}) => {
            const localVarPath = `/computeruse/display/windows`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Call an AT-SPI Action on the node. Leave action empty to invoke the node\'s primary (first) action.
         * @summary Invoke an action on an accessibility node
         * @param {AccessibilityInvokeRequest} request Invoke request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        invokeAccessibilityNode: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('invokeAccessibilityNode', 'request', request);
            const localVarPath = `/computeruse/a11y/node/invoke`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get a list of all recordings (active and completed)
         * @summary List all recordings
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listRecordings: async (options = {}) => {
            const localVarPath = `/computeruse/recordings`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Move the mouse cursor to the specified coordinates
         * @summary Move mouse cursor
         * @param {MouseMoveRequest} request Mouse move request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        moveMouse: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('moveMouse', 'request', request);
            const localVarPath = `/computeruse/mouse/move`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Press a hotkey combination (e.g., ctrl+c, cmd+v)
         * @summary Press hotkey
         * @param {KeyboardHotkeyRequest} request Hotkey press request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pressHotkey: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('pressHotkey', 'request', request);
            const localVarPath = `/computeruse/keyboard/hotkey`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Press a key with optional modifiers
         * @summary Press key
         * @param {KeyboardPressRequest} request Key press request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pressKey: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('pressKey', 'request', request);
            const localVarPath = `/computeruse/keyboard/key`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Restart a specific computer use process
         * @summary Restart specific process
         * @param {string} processName Process name to restart
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        restartProcess: async (processName, options = {}) => {
            // verify required parameter 'processName' is not null or undefined
            (0, common_1.assertParamExists)('restartProcess', 'processName', processName);
            const localVarPath = `/computeruse/process/{processName}/restart`
                .replace(`{${"processName"}}`, encodeURIComponent(String(processName)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Scroll the mouse wheel at the specified coordinates
         * @summary Scroll mouse wheel
         * @param {MouseScrollRequest} request Mouse scroll request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        scroll: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('scroll', 'request', request);
            const localVarPath = `/computeruse/mouse/scroll`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Write the given value to the node via EditableText.SetTextContents or, for numeric controls, Value.CurrentValue.
         * @summary Set the value of an accessibility node
         * @param {AccessibilitySetValueRequest} request Set value request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        setAccessibilityNodeValue: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('setAccessibilityNodeValue', 'request', request);
            const localVarPath = `/computeruse/a11y/node/value`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Start all computer use processes and return their status
         * @summary Start computer use processes
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        startComputerUse: async (options = {}) => {
            const localVarPath = `/computeruse/start`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Start a new screen recording session
         * @summary Start a new recording
         * @param {StartRecordingRequest} [request] Recording options
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        startRecording: async (request, options = {}) => {
            const localVarPath = `/computeruse/recordings/start`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Stop all computer use processes and return their status
         * @summary Stop computer use processes
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        stopComputerUse: async (options = {}) => {
            const localVarPath = `/computeruse/stop`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Stop an active screen recording session
         * @summary Stop a recording
         * @param {StopRecordingRequest} request Recording ID to stop
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        stopRecording: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('stopRecording', 'request', request);
            const localVarPath = `/computeruse/recordings/stop`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Take a compressed screenshot of a specific region of the screen
         * @summary Take a compressed region screenshot
         * @param {number} x X coordinate of the region
         * @param {number} y Y coordinate of the region
         * @param {number} width Width of the region
         * @param {number} height Height of the region
         * @param {boolean} [showCursor] Whether to show cursor in screenshot
         * @param {string} [format] Image format (png or jpeg)
         * @param {number} [quality] JPEG quality (1-100)
         * @param {number} [scale] Scale factor (0.1-1.0)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        takeCompressedRegionScreenshot: async (x, y, width, height, showCursor, format, quality, scale, options = {}) => {
            // verify required parameter 'x' is not null or undefined
            (0, common_1.assertParamExists)('takeCompressedRegionScreenshot', 'x', x);
            // verify required parameter 'y' is not null or undefined
            (0, common_1.assertParamExists)('takeCompressedRegionScreenshot', 'y', y);
            // verify required parameter 'width' is not null or undefined
            (0, common_1.assertParamExists)('takeCompressedRegionScreenshot', 'width', width);
            // verify required parameter 'height' is not null or undefined
            (0, common_1.assertParamExists)('takeCompressedRegionScreenshot', 'height', height);
            const localVarPath = `/computeruse/screenshot/region/compressed`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (x !== undefined) {
                localVarQueryParameter['x'] = x;
            }
            if (y !== undefined) {
                localVarQueryParameter['y'] = y;
            }
            if (width !== undefined) {
                localVarQueryParameter['width'] = width;
            }
            if (height !== undefined) {
                localVarQueryParameter['height'] = height;
            }
            if (showCursor !== undefined) {
                localVarQueryParameter['showCursor'] = showCursor;
            }
            if (format !== undefined) {
                localVarQueryParameter['format'] = format;
            }
            if (quality !== undefined) {
                localVarQueryParameter['quality'] = quality;
            }
            if (scale !== undefined) {
                localVarQueryParameter['scale'] = scale;
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Take a compressed screenshot of the entire screen
         * @summary Take a compressed screenshot
         * @param {boolean} [showCursor] Whether to show cursor in screenshot
         * @param {string} [format] Image format (png or jpeg)
         * @param {number} [quality] JPEG quality (1-100)
         * @param {number} [scale] Scale factor (0.1-1.0)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        takeCompressedScreenshot: async (showCursor, format, quality, scale, options = {}) => {
            const localVarPath = `/computeruse/screenshot/compressed`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (showCursor !== undefined) {
                localVarQueryParameter['showCursor'] = showCursor;
            }
            if (format !== undefined) {
                localVarQueryParameter['format'] = format;
            }
            if (quality !== undefined) {
                localVarQueryParameter['quality'] = quality;
            }
            if (scale !== undefined) {
                localVarQueryParameter['scale'] = scale;
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Take a screenshot of a specific region of the screen
         * @summary Take a region screenshot
         * @param {number} x X coordinate of the region
         * @param {number} y Y coordinate of the region
         * @param {number} width Width of the region
         * @param {number} height Height of the region
         * @param {boolean} [showCursor] Whether to show cursor in screenshot
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        takeRegionScreenshot: async (x, y, width, height, showCursor, options = {}) => {
            // verify required parameter 'x' is not null or undefined
            (0, common_1.assertParamExists)('takeRegionScreenshot', 'x', x);
            // verify required parameter 'y' is not null or undefined
            (0, common_1.assertParamExists)('takeRegionScreenshot', 'y', y);
            // verify required parameter 'width' is not null or undefined
            (0, common_1.assertParamExists)('takeRegionScreenshot', 'width', width);
            // verify required parameter 'height' is not null or undefined
            (0, common_1.assertParamExists)('takeRegionScreenshot', 'height', height);
            const localVarPath = `/computeruse/screenshot/region`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (x !== undefined) {
                localVarQueryParameter['x'] = x;
            }
            if (y !== undefined) {
                localVarQueryParameter['y'] = y;
            }
            if (width !== undefined) {
                localVarQueryParameter['width'] = width;
            }
            if (height !== undefined) {
                localVarQueryParameter['height'] = height;
            }
            if (showCursor !== undefined) {
                localVarQueryParameter['showCursor'] = showCursor;
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Take a screenshot of the entire screen
         * @summary Take a screenshot
         * @param {boolean} [showCursor] Whether to show cursor in screenshot
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        takeScreenshot: async (showCursor, options = {}) => {
            const localVarPath = `/computeruse/screenshot`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (showCursor !== undefined) {
                localVarQueryParameter['showCursor'] = showCursor;
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Type text with optional delay between keystrokes
         * @summary Type text
         * @param {KeyboardTypeRequest} request Text typing request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        typeText: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('typeText', 'request', request);
            const localVarPath = `/computeruse/keyboard/type`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.ComputerUseApiAxiosParamCreator = ComputerUseApiAxiosParamCreator;
/**
 * ComputerUseApi - functional programming interface
 */
const ComputerUseApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.ComputerUseApiAxiosParamCreator)(configuration);
    return {
        /**
         * Click the mouse button at the specified coordinates
         * @summary Click mouse button
         * @param {MouseClickRequest} request Mouse click request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async click(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.click(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.click']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Delete a recording file by ID
         * @summary Delete a recording
         * @param {string} id Recording ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async deleteRecording(id, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.deleteRecording(id, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.deleteRecording']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Download a recording by providing its ID
         * @summary Download a recording
         * @param {string} id Recording ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async downloadRecording(id, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.downloadRecording(id, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.downloadRecording']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Drag the mouse from start to end coordinates
         * @summary Drag mouse
         * @param {MouseDragRequest} request Mouse drag request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async drag(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.drag(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.drag']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Search the AT-SPI tree for nodes matching a role/name/state filter and return a flat list.
         * @summary Find accessibility nodes
         * @param {FindAccessibilityNodesRequest} request Find request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async findAccessibilityNodes(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.findAccessibilityNodes(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.findAccessibilityNodes']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Move keyboard focus to the AT-SPI node identified by id (bus-name:object-path).
         * @summary Focus an accessibility node
         * @param {AccessibilityNodeRequest} request Node focus request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async focusAccessibilityNode(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.focusAccessibilityNode(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.focusAccessibilityNode']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Fetch the AT-SPI accessibility tree for the focused application, a specific PID, or all registered applications.
         * @summary Get accessibility tree
         * @param {string} [scope] Scope: focused | pid | all (default: focused)
         * @param {number} [pid] Process ID when scope&#x3D;pid
         * @param {number} [maxDepth] Max tree depth (-1 unbounded, 0 root only; default -1)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getAccessibilityTree(scope, pid, maxDepth, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getAccessibilityTree(scope, pid, maxDepth, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.getAccessibilityTree']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get the status of all computer use processes
         * @summary Get computer use process status
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getComputerUseStatus(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getComputerUseStatus(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.getComputerUseStatus']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get the current status of the computer use system
         * @summary Get computer use status
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getComputerUseSystemStatus(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getComputerUseSystemStatus(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.getComputerUseSystemStatus']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get information about all available displays
         * @summary Get display information
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getDisplayInfo(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getDisplayInfo(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.getDisplayInfo']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get the current mouse cursor position
         * @summary Get mouse position
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getMousePosition(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getMousePosition(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.getMousePosition']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get errors for a specific computer use process
         * @summary Get process errors
         * @param {string} processName Process name to get errors for
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getProcessErrors(processName, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getProcessErrors(processName, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.getProcessErrors']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get logs for a specific computer use process
         * @summary Get process logs
         * @param {string} processName Process name to get logs for
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getProcessLogs(processName, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getProcessLogs(processName, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.getProcessLogs']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Check if a specific computer use process is running
         * @summary Get specific process status
         * @param {string} processName Process name to check
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getProcessStatus(processName, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getProcessStatus(processName, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.getProcessStatus']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get details of a specific recording by ID
         * @summary Get recording details
         * @param {string} id Recording ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getRecording(id, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getRecording(id, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.getRecording']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get information about all open windows
         * @summary Get windows information
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getWindows(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getWindows(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.getWindows']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Call an AT-SPI Action on the node. Leave action empty to invoke the node\'s primary (first) action.
         * @summary Invoke an action on an accessibility node
         * @param {AccessibilityInvokeRequest} request Invoke request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async invokeAccessibilityNode(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.invokeAccessibilityNode(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.invokeAccessibilityNode']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get a list of all recordings (active and completed)
         * @summary List all recordings
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listRecordings(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listRecordings(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.listRecordings']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Move the mouse cursor to the specified coordinates
         * @summary Move mouse cursor
         * @param {MouseMoveRequest} request Mouse move request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async moveMouse(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.moveMouse(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.moveMouse']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Press a hotkey combination (e.g., ctrl+c, cmd+v)
         * @summary Press hotkey
         * @param {KeyboardHotkeyRequest} request Hotkey press request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async pressHotkey(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.pressHotkey(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.pressHotkey']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Press a key with optional modifiers
         * @summary Press key
         * @param {KeyboardPressRequest} request Key press request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async pressKey(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.pressKey(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.pressKey']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Restart a specific computer use process
         * @summary Restart specific process
         * @param {string} processName Process name to restart
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async restartProcess(processName, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.restartProcess(processName, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.restartProcess']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Scroll the mouse wheel at the specified coordinates
         * @summary Scroll mouse wheel
         * @param {MouseScrollRequest} request Mouse scroll request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async scroll(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.scroll(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.scroll']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Write the given value to the node via EditableText.SetTextContents or, for numeric controls, Value.CurrentValue.
         * @summary Set the value of an accessibility node
         * @param {AccessibilitySetValueRequest} request Set value request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async setAccessibilityNodeValue(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.setAccessibilityNodeValue(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.setAccessibilityNodeValue']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Start all computer use processes and return their status
         * @summary Start computer use processes
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async startComputerUse(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.startComputerUse(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.startComputerUse']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Start a new screen recording session
         * @summary Start a new recording
         * @param {StartRecordingRequest} [request] Recording options
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async startRecording(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.startRecording(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.startRecording']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Stop all computer use processes and return their status
         * @summary Stop computer use processes
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async stopComputerUse(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.stopComputerUse(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.stopComputerUse']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Stop an active screen recording session
         * @summary Stop a recording
         * @param {StopRecordingRequest} request Recording ID to stop
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async stopRecording(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.stopRecording(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.stopRecording']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Take a compressed screenshot of a specific region of the screen
         * @summary Take a compressed region screenshot
         * @param {number} x X coordinate of the region
         * @param {number} y Y coordinate of the region
         * @param {number} width Width of the region
         * @param {number} height Height of the region
         * @param {boolean} [showCursor] Whether to show cursor in screenshot
         * @param {string} [format] Image format (png or jpeg)
         * @param {number} [quality] JPEG quality (1-100)
         * @param {number} [scale] Scale factor (0.1-1.0)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async takeCompressedRegionScreenshot(x, y, width, height, showCursor, format, quality, scale, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.takeCompressedRegionScreenshot(x, y, width, height, showCursor, format, quality, scale, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.takeCompressedRegionScreenshot']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Take a compressed screenshot of the entire screen
         * @summary Take a compressed screenshot
         * @param {boolean} [showCursor] Whether to show cursor in screenshot
         * @param {string} [format] Image format (png or jpeg)
         * @param {number} [quality] JPEG quality (1-100)
         * @param {number} [scale] Scale factor (0.1-1.0)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async takeCompressedScreenshot(showCursor, format, quality, scale, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.takeCompressedScreenshot(showCursor, format, quality, scale, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.takeCompressedScreenshot']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Take a screenshot of a specific region of the screen
         * @summary Take a region screenshot
         * @param {number} x X coordinate of the region
         * @param {number} y Y coordinate of the region
         * @param {number} width Width of the region
         * @param {number} height Height of the region
         * @param {boolean} [showCursor] Whether to show cursor in screenshot
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async takeRegionScreenshot(x, y, width, height, showCursor, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.takeRegionScreenshot(x, y, width, height, showCursor, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.takeRegionScreenshot']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Take a screenshot of the entire screen
         * @summary Take a screenshot
         * @param {boolean} [showCursor] Whether to show cursor in screenshot
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async takeScreenshot(showCursor, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.takeScreenshot(showCursor, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.takeScreenshot']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Type text with optional delay between keystrokes
         * @summary Type text
         * @param {KeyboardTypeRequest} request Text typing request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async typeText(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.typeText(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ComputerUseApi.typeText']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.ComputerUseApiFp = ComputerUseApiFp;
/**
 * ComputerUseApi - factory interface
 */
const ComputerUseApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.ComputerUseApiFp)(configuration);
    return {
        /**
         * Click the mouse button at the specified coordinates
         * @summary Click mouse button
         * @param {MouseClickRequest} request Mouse click request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        click(request, options) {
            return localVarFp.click(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Delete a recording file by ID
         * @summary Delete a recording
         * @param {string} id Recording ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteRecording(id, options) {
            return localVarFp.deleteRecording(id, options).then((request) => request(axios, basePath));
        },
        /**
         * Download a recording by providing its ID
         * @summary Download a recording
         * @param {string} id Recording ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        downloadRecording(id, options) {
            return localVarFp.downloadRecording(id, options).then((request) => request(axios, basePath));
        },
        /**
         * Drag the mouse from start to end coordinates
         * @summary Drag mouse
         * @param {MouseDragRequest} request Mouse drag request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        drag(request, options) {
            return localVarFp.drag(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Search the AT-SPI tree for nodes matching a role/name/state filter and return a flat list.
         * @summary Find accessibility nodes
         * @param {FindAccessibilityNodesRequest} request Find request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        findAccessibilityNodes(request, options) {
            return localVarFp.findAccessibilityNodes(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Move keyboard focus to the AT-SPI node identified by id (bus-name:object-path).
         * @summary Focus an accessibility node
         * @param {AccessibilityNodeRequest} request Node focus request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        focusAccessibilityNode(request, options) {
            return localVarFp.focusAccessibilityNode(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Fetch the AT-SPI accessibility tree for the focused application, a specific PID, or all registered applications.
         * @summary Get accessibility tree
         * @param {string} [scope] Scope: focused | pid | all (default: focused)
         * @param {number} [pid] Process ID when scope&#x3D;pid
         * @param {number} [maxDepth] Max tree depth (-1 unbounded, 0 root only; default -1)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getAccessibilityTree(scope, pid, maxDepth, options) {
            return localVarFp.getAccessibilityTree(scope, pid, maxDepth, options).then((request) => request(axios, basePath));
        },
        /**
         * Get the status of all computer use processes
         * @summary Get computer use process status
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getComputerUseStatus(options) {
            return localVarFp.getComputerUseStatus(options).then((request) => request(axios, basePath));
        },
        /**
         * Get the current status of the computer use system
         * @summary Get computer use status
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getComputerUseSystemStatus(options) {
            return localVarFp.getComputerUseSystemStatus(options).then((request) => request(axios, basePath));
        },
        /**
         * Get information about all available displays
         * @summary Get display information
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getDisplayInfo(options) {
            return localVarFp.getDisplayInfo(options).then((request) => request(axios, basePath));
        },
        /**
         * Get the current mouse cursor position
         * @summary Get mouse position
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMousePosition(options) {
            return localVarFp.getMousePosition(options).then((request) => request(axios, basePath));
        },
        /**
         * Get errors for a specific computer use process
         * @summary Get process errors
         * @param {string} processName Process name to get errors for
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getProcessErrors(processName, options) {
            return localVarFp.getProcessErrors(processName, options).then((request) => request(axios, basePath));
        },
        /**
         * Get logs for a specific computer use process
         * @summary Get process logs
         * @param {string} processName Process name to get logs for
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getProcessLogs(processName, options) {
            return localVarFp.getProcessLogs(processName, options).then((request) => request(axios, basePath));
        },
        /**
         * Check if a specific computer use process is running
         * @summary Get specific process status
         * @param {string} processName Process name to check
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getProcessStatus(processName, options) {
            return localVarFp.getProcessStatus(processName, options).then((request) => request(axios, basePath));
        },
        /**
         * Get details of a specific recording by ID
         * @summary Get recording details
         * @param {string} id Recording ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getRecording(id, options) {
            return localVarFp.getRecording(id, options).then((request) => request(axios, basePath));
        },
        /**
         * Get information about all open windows
         * @summary Get windows information
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getWindows(options) {
            return localVarFp.getWindows(options).then((request) => request(axios, basePath));
        },
        /**
         * Call an AT-SPI Action on the node. Leave action empty to invoke the node\'s primary (first) action.
         * @summary Invoke an action on an accessibility node
         * @param {AccessibilityInvokeRequest} request Invoke request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        invokeAccessibilityNode(request, options) {
            return localVarFp.invokeAccessibilityNode(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Get a list of all recordings (active and completed)
         * @summary List all recordings
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listRecordings(options) {
            return localVarFp.listRecordings(options).then((request) => request(axios, basePath));
        },
        /**
         * Move the mouse cursor to the specified coordinates
         * @summary Move mouse cursor
         * @param {MouseMoveRequest} request Mouse move request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        moveMouse(request, options) {
            return localVarFp.moveMouse(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Press a hotkey combination (e.g., ctrl+c, cmd+v)
         * @summary Press hotkey
         * @param {KeyboardHotkeyRequest} request Hotkey press request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pressHotkey(request, options) {
            return localVarFp.pressHotkey(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Press a key with optional modifiers
         * @summary Press key
         * @param {KeyboardPressRequest} request Key press request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pressKey(request, options) {
            return localVarFp.pressKey(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Restart a specific computer use process
         * @summary Restart specific process
         * @param {string} processName Process name to restart
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        restartProcess(processName, options) {
            return localVarFp.restartProcess(processName, options).then((request) => request(axios, basePath));
        },
        /**
         * Scroll the mouse wheel at the specified coordinates
         * @summary Scroll mouse wheel
         * @param {MouseScrollRequest} request Mouse scroll request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        scroll(request, options) {
            return localVarFp.scroll(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Write the given value to the node via EditableText.SetTextContents or, for numeric controls, Value.CurrentValue.
         * @summary Set the value of an accessibility node
         * @param {AccessibilitySetValueRequest} request Set value request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        setAccessibilityNodeValue(request, options) {
            return localVarFp.setAccessibilityNodeValue(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Start all computer use processes and return their status
         * @summary Start computer use processes
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        startComputerUse(options) {
            return localVarFp.startComputerUse(options).then((request) => request(axios, basePath));
        },
        /**
         * Start a new screen recording session
         * @summary Start a new recording
         * @param {StartRecordingRequest} [request] Recording options
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        startRecording(request, options) {
            return localVarFp.startRecording(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Stop all computer use processes and return their status
         * @summary Stop computer use processes
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        stopComputerUse(options) {
            return localVarFp.stopComputerUse(options).then((request) => request(axios, basePath));
        },
        /**
         * Stop an active screen recording session
         * @summary Stop a recording
         * @param {StopRecordingRequest} request Recording ID to stop
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        stopRecording(request, options) {
            return localVarFp.stopRecording(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Take a compressed screenshot of a specific region of the screen
         * @summary Take a compressed region screenshot
         * @param {number} x X coordinate of the region
         * @param {number} y Y coordinate of the region
         * @param {number} width Width of the region
         * @param {number} height Height of the region
         * @param {boolean} [showCursor] Whether to show cursor in screenshot
         * @param {string} [format] Image format (png or jpeg)
         * @param {number} [quality] JPEG quality (1-100)
         * @param {number} [scale] Scale factor (0.1-1.0)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        takeCompressedRegionScreenshot(x, y, width, height, showCursor, format, quality, scale, options) {
            return localVarFp.takeCompressedRegionScreenshot(x, y, width, height, showCursor, format, quality, scale, options).then((request) => request(axios, basePath));
        },
        /**
         * Take a compressed screenshot of the entire screen
         * @summary Take a compressed screenshot
         * @param {boolean} [showCursor] Whether to show cursor in screenshot
         * @param {string} [format] Image format (png or jpeg)
         * @param {number} [quality] JPEG quality (1-100)
         * @param {number} [scale] Scale factor (0.1-1.0)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        takeCompressedScreenshot(showCursor, format, quality, scale, options) {
            return localVarFp.takeCompressedScreenshot(showCursor, format, quality, scale, options).then((request) => request(axios, basePath));
        },
        /**
         * Take a screenshot of a specific region of the screen
         * @summary Take a region screenshot
         * @param {number} x X coordinate of the region
         * @param {number} y Y coordinate of the region
         * @param {number} width Width of the region
         * @param {number} height Height of the region
         * @param {boolean} [showCursor] Whether to show cursor in screenshot
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        takeRegionScreenshot(x, y, width, height, showCursor, options) {
            return localVarFp.takeRegionScreenshot(x, y, width, height, showCursor, options).then((request) => request(axios, basePath));
        },
        /**
         * Take a screenshot of the entire screen
         * @summary Take a screenshot
         * @param {boolean} [showCursor] Whether to show cursor in screenshot
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        takeScreenshot(showCursor, options) {
            return localVarFp.takeScreenshot(showCursor, options).then((request) => request(axios, basePath));
        },
        /**
         * Type text with optional delay between keystrokes
         * @summary Type text
         * @param {KeyboardTypeRequest} request Text typing request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        typeText(request, options) {
            return localVarFp.typeText(request, options).then((request) => request(axios, basePath));
        },
    };
};
exports.ComputerUseApiFactory = ComputerUseApiFactory;
/**
 * ComputerUseApi - object-oriented interface
 */
class ComputerUseApi extends base_1.BaseAPI {
    /**
     * Click the mouse button at the specified coordinates
     * @summary Click mouse button
     * @param {MouseClickRequest} request Mouse click request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    click(request, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).click(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Delete a recording file by ID
     * @summary Delete a recording
     * @param {string} id Recording ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    deleteRecording(id, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).deleteRecording(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Download a recording by providing its ID
     * @summary Download a recording
     * @param {string} id Recording ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    downloadRecording(id, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).downloadRecording(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Drag the mouse from start to end coordinates
     * @summary Drag mouse
     * @param {MouseDragRequest} request Mouse drag request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    drag(request, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).drag(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Search the AT-SPI tree for nodes matching a role/name/state filter and return a flat list.
     * @summary Find accessibility nodes
     * @param {FindAccessibilityNodesRequest} request Find request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    findAccessibilityNodes(request, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).findAccessibilityNodes(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Move keyboard focus to the AT-SPI node identified by id (bus-name:object-path).
     * @summary Focus an accessibility node
     * @param {AccessibilityNodeRequest} request Node focus request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    focusAccessibilityNode(request, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).focusAccessibilityNode(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Fetch the AT-SPI accessibility tree for the focused application, a specific PID, or all registered applications.
     * @summary Get accessibility tree
     * @param {string} [scope] Scope: focused | pid | all (default: focused)
     * @param {number} [pid] Process ID when scope&#x3D;pid
     * @param {number} [maxDepth] Max tree depth (-1 unbounded, 0 root only; default -1)
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getAccessibilityTree(scope, pid, maxDepth, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).getAccessibilityTree(scope, pid, maxDepth, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get the status of all computer use processes
     * @summary Get computer use process status
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getComputerUseStatus(options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).getComputerUseStatus(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get the current status of the computer use system
     * @summary Get computer use status
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getComputerUseSystemStatus(options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).getComputerUseSystemStatus(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get information about all available displays
     * @summary Get display information
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getDisplayInfo(options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).getDisplayInfo(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get the current mouse cursor position
     * @summary Get mouse position
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getMousePosition(options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).getMousePosition(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get errors for a specific computer use process
     * @summary Get process errors
     * @param {string} processName Process name to get errors for
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProcessErrors(processName, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).getProcessErrors(processName, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get logs for a specific computer use process
     * @summary Get process logs
     * @param {string} processName Process name to get logs for
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProcessLogs(processName, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).getProcessLogs(processName, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Check if a specific computer use process is running
     * @summary Get specific process status
     * @param {string} processName Process name to check
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProcessStatus(processName, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).getProcessStatus(processName, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get details of a specific recording by ID
     * @summary Get recording details
     * @param {string} id Recording ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getRecording(id, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).getRecording(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get information about all open windows
     * @summary Get windows information
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getWindows(options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).getWindows(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Call an AT-SPI Action on the node. Leave action empty to invoke the node\'s primary (first) action.
     * @summary Invoke an action on an accessibility node
     * @param {AccessibilityInvokeRequest} request Invoke request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    invokeAccessibilityNode(request, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).invokeAccessibilityNode(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get a list of all recordings (active and completed)
     * @summary List all recordings
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    listRecordings(options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).listRecordings(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Move the mouse cursor to the specified coordinates
     * @summary Move mouse cursor
     * @param {MouseMoveRequest} request Mouse move request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    moveMouse(request, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).moveMouse(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Press a hotkey combination (e.g., ctrl+c, cmd+v)
     * @summary Press hotkey
     * @param {KeyboardHotkeyRequest} request Hotkey press request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    pressHotkey(request, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).pressHotkey(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Press a key with optional modifiers
     * @summary Press key
     * @param {KeyboardPressRequest} request Key press request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    pressKey(request, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).pressKey(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Restart a specific computer use process
     * @summary Restart specific process
     * @param {string} processName Process name to restart
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    restartProcess(processName, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).restartProcess(processName, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Scroll the mouse wheel at the specified coordinates
     * @summary Scroll mouse wheel
     * @param {MouseScrollRequest} request Mouse scroll request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    scroll(request, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).scroll(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Write the given value to the node via EditableText.SetTextContents or, for numeric controls, Value.CurrentValue.
     * @summary Set the value of an accessibility node
     * @param {AccessibilitySetValueRequest} request Set value request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    setAccessibilityNodeValue(request, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).setAccessibilityNodeValue(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Start all computer use processes and return their status
     * @summary Start computer use processes
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    startComputerUse(options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).startComputerUse(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Start a new screen recording session
     * @summary Start a new recording
     * @param {StartRecordingRequest} [request] Recording options
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    startRecording(request, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).startRecording(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Stop all computer use processes and return their status
     * @summary Stop computer use processes
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    stopComputerUse(options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).stopComputerUse(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Stop an active screen recording session
     * @summary Stop a recording
     * @param {StopRecordingRequest} request Recording ID to stop
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    stopRecording(request, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).stopRecording(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Take a compressed screenshot of a specific region of the screen
     * @summary Take a compressed region screenshot
     * @param {number} x X coordinate of the region
     * @param {number} y Y coordinate of the region
     * @param {number} width Width of the region
     * @param {number} height Height of the region
     * @param {boolean} [showCursor] Whether to show cursor in screenshot
     * @param {string} [format] Image format (png or jpeg)
     * @param {number} [quality] JPEG quality (1-100)
     * @param {number} [scale] Scale factor (0.1-1.0)
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    takeCompressedRegionScreenshot(x, y, width, height, showCursor, format, quality, scale, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).takeCompressedRegionScreenshot(x, y, width, height, showCursor, format, quality, scale, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Take a compressed screenshot of the entire screen
     * @summary Take a compressed screenshot
     * @param {boolean} [showCursor] Whether to show cursor in screenshot
     * @param {string} [format] Image format (png or jpeg)
     * @param {number} [quality] JPEG quality (1-100)
     * @param {number} [scale] Scale factor (0.1-1.0)
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    takeCompressedScreenshot(showCursor, format, quality, scale, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).takeCompressedScreenshot(showCursor, format, quality, scale, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Take a screenshot of a specific region of the screen
     * @summary Take a region screenshot
     * @param {number} x X coordinate of the region
     * @param {number} y Y coordinate of the region
     * @param {number} width Width of the region
     * @param {number} height Height of the region
     * @param {boolean} [showCursor] Whether to show cursor in screenshot
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    takeRegionScreenshot(x, y, width, height, showCursor, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).takeRegionScreenshot(x, y, width, height, showCursor, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Take a screenshot of the entire screen
     * @summary Take a screenshot
     * @param {boolean} [showCursor] Whether to show cursor in screenshot
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    takeScreenshot(showCursor, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).takeScreenshot(showCursor, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Type text with optional delay between keystrokes
     * @summary Type text
     * @param {KeyboardTypeRequest} request Text typing request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    typeText(request, options) {
        return (0, exports.ComputerUseApiFp)(this.configuration).typeText(request, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.ComputerUseApi = ComputerUseApi;
