// @daytona/toolbox-api-client/src/api/file-system-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Daytona Toolbox API
 * Daytona Toolbox API. The base URL comes from the sandbox\'s `toolboxProxyUrl` field (returned in sandbox DTO by the main Daytona API) plus the sandbox ID: `{toolboxProxyUrl}/{sandboxId}/{endpoint}`. Default for Daytona Cloud: `https://proxy.app.daytona.io/toolbox/{sandboxId}`.
 *
 * The version of the OpenAPI document: v0.0.0-dev
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileSystemApi = exports.FileSystemApiFactory = exports.FileSystemApiFp = exports.FileSystemApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * FileSystemApi - axios parameter creator
 */
const FileSystemApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Create a folder with the specified path and optional permissions
         * @summary Create a folder
         * @param {string} path Folder path to create
         * @param {string} mode Octal permission mode (default: 0755)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createFolder: async (path, mode, options = {}) => {
            // verify required parameter 'path' is not null or undefined
            (0, common_1.assertParamExists)('createFolder', 'path', path);
            // verify required parameter 'mode' is not null or undefined
            (0, common_1.assertParamExists)('createFolder', 'mode', mode);
            const localVarPath = `/files/folder`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (path !== undefined) {
                localVarQueryParameter['path'] = path;
            }
            if (mode !== undefined) {
                localVarQueryParameter['mode'] = mode;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Delete a file or directory at the specified path
         * @summary Delete a file or directory
         * @param {string} path File or directory path to delete
         * @param {boolean} [recursive] Enable recursive deletion for directories
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteFile: async (path, recursive, options = {}) => {
            // verify required parameter 'path' is not null or undefined
            (0, common_1.assertParamExists)('deleteFile', 'path', path);
            const localVarPath = `/files`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'DELETE', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (path !== undefined) {
                localVarQueryParameter['path'] = path;
            }
            if (recursive !== undefined) {
                localVarQueryParameter['recursive'] = recursive;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Download a file by providing its path
         * @summary Download a file
         * @param {string} path File path to download
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        downloadFile: async (path, options = {}) => {
            // verify required parameter 'path' is not null or undefined
            (0, common_1.assertParamExists)('downloadFile', 'path', path);
            const localVarPath = `/files/download`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (path !== undefined) {
                localVarQueryParameter['path'] = path;
            }
            localVarHeaderParameter['Accept'] = 'application/octet-stream';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Download multiple files by providing their paths. Successful files are returned as multipart parts named `file`. Per-file failures are returned as multipart parts named `error` with JSON payloads shaped like ErrorResponse.
         * @summary Download multiple files
         * @param {FilesDownloadRequest} downloadFiles Paths of files to download
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        downloadFiles: async (downloadFiles, options = {}) => {
            // verify required parameter 'downloadFiles' is not null or undefined
            (0, common_1.assertParamExists)('downloadFiles', 'downloadFiles', downloadFiles);
            const localVarPath = `/files/bulk-download`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'multipart/form-data';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(downloadFiles, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Search for text pattern within files in a directory
         * @summary Find text in files
         * @param {string} path Directory path to search in
         * @param {string} pattern Text pattern to search for
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        findInFiles: async (path, pattern, options = {}) => {
            // verify required parameter 'path' is not null or undefined
            (0, common_1.assertParamExists)('findInFiles', 'path', path);
            // verify required parameter 'pattern' is not null or undefined
            (0, common_1.assertParamExists)('findInFiles', 'pattern', pattern);
            const localVarPath = `/files/find`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (path !== undefined) {
                localVarQueryParameter['path'] = path;
            }
            if (pattern !== undefined) {
                localVarQueryParameter['pattern'] = pattern;
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get detailed information about a file or directory
         * @summary Get file information
         * @param {string} path File or directory path
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getFileInfo: async (path, options = {}) => {
            // verify required parameter 'path' is not null or undefined
            (0, common_1.assertParamExists)('getFileInfo', 'path', path);
            const localVarPath = `/files/info`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (path !== undefined) {
                localVarQueryParameter['path'] = path;
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * List files and directories in the specified path
         * @summary List files and directories
         * @param {string} [path] Directory path to list (defaults to working directory)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listFiles: async (path, options = {}) => {
            const localVarPath = `/files`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (path !== undefined) {
                localVarQueryParameter['path'] = path;
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Move or rename a file or directory from source to destination
         * @summary Move or rename file/directory
         * @param {string} source Source file or directory path
         * @param {string} destination Destination file or directory path
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        moveFile: async (source, destination, options = {}) => {
            // verify required parameter 'source' is not null or undefined
            (0, common_1.assertParamExists)('moveFile', 'source', source);
            // verify required parameter 'destination' is not null or undefined
            (0, common_1.assertParamExists)('moveFile', 'destination', destination);
            const localVarPath = `/files/move`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (source !== undefined) {
                localVarQueryParameter['source'] = source;
            }
            if (destination !== undefined) {
                localVarQueryParameter['destination'] = destination;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Replace text pattern with new value in multiple files
         * @summary Replace text in files
         * @param {ReplaceRequest} request Replace request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        replaceInFiles: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('replaceInFiles', 'request', request);
            const localVarPath = `/files/replace`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Search for files matching a specific pattern in a directory
         * @summary Search files by pattern
         * @param {string} path Directory path to search in
         * @param {string} pattern File pattern to match (e.g., *.txt, *.go)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        searchFiles: async (path, pattern, options = {}) => {
            // verify required parameter 'path' is not null or undefined
            (0, common_1.assertParamExists)('searchFiles', 'path', path);
            // verify required parameter 'pattern' is not null or undefined
            (0, common_1.assertParamExists)('searchFiles', 'pattern', pattern);
            const localVarPath = `/files/search`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (path !== undefined) {
                localVarQueryParameter['path'] = path;
            }
            if (pattern !== undefined) {
                localVarQueryParameter['pattern'] = pattern;
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Set file permissions, ownership, and group for a file or directory
         * @summary Set file permissions
         * @param {string} path File or directory path
         * @param {string} [owner] Owner (username or UID)
         * @param {string} [group] Group (group name or GID)
         * @param {string} [mode] File mode in octal format (e.g., 0755)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        setFilePermissions: async (path, owner, group, mode, options = {}) => {
            // verify required parameter 'path' is not null or undefined
            (0, common_1.assertParamExists)('setFilePermissions', 'path', path);
            const localVarPath = `/files/permissions`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (path !== undefined) {
                localVarQueryParameter['path'] = path;
            }
            if (owner !== undefined) {
                localVarQueryParameter['owner'] = owner;
            }
            if (group !== undefined) {
                localVarQueryParameter['group'] = group;
            }
            if (mode !== undefined) {
                localVarQueryParameter['mode'] = mode;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Upload a file to the specified path
         * @summary Upload a file
         * @param {string} path Destination path for the uploaded file
         * @param {File} file File to upload
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        uploadFile: async (path, file, options = {}) => {
            // verify required parameter 'path' is not null or undefined
            (0, common_1.assertParamExists)('uploadFile', 'path', path);
            // verify required parameter 'file' is not null or undefined
            (0, common_1.assertParamExists)('uploadFile', 'file', file);
            const localVarPath = `/files/upload`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            const localVarFormParams = new ((configuration && configuration.formDataCtor) || FormData)();
            if (path !== undefined) {
                localVarQueryParameter['path'] = path;
            }
            if (file !== undefined) {
                localVarFormParams.append('file', file);
            }
            localVarHeaderParameter['Content-Type'] = 'multipart/form-data';
            localVarHeaderParameter['Accept'] = '*/*';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = localVarFormParams;
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Upload multiple files with their destination paths
         * @summary Upload multiple files
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        uploadFiles: async (options = {}) => {
            const localVarPath = `/files/bulk-upload`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.FileSystemApiAxiosParamCreator = FileSystemApiAxiosParamCreator;
/**
 * FileSystemApi - functional programming interface
 */
const FileSystemApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.FileSystemApiAxiosParamCreator)(configuration);
    return {
        /**
         * Create a folder with the specified path and optional permissions
         * @summary Create a folder
         * @param {string} path Folder path to create
         * @param {string} mode Octal permission mode (default: 0755)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async createFolder(path, mode, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.createFolder(path, mode, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['FileSystemApi.createFolder']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Delete a file or directory at the specified path
         * @summary Delete a file or directory
         * @param {string} path File or directory path to delete
         * @param {boolean} [recursive] Enable recursive deletion for directories
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async deleteFile(path, recursive, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.deleteFile(path, recursive, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['FileSystemApi.deleteFile']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Download a file by providing its path
         * @summary Download a file
         * @param {string} path File path to download
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async downloadFile(path, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.downloadFile(path, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['FileSystemApi.downloadFile']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Download multiple files by providing their paths. Successful files are returned as multipart parts named `file`. Per-file failures are returned as multipart parts named `error` with JSON payloads shaped like ErrorResponse.
         * @summary Download multiple files
         * @param {FilesDownloadRequest} downloadFiles Paths of files to download
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async downloadFiles(downloadFiles, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.downloadFiles(downloadFiles, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['FileSystemApi.downloadFiles']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Search for text pattern within files in a directory
         * @summary Find text in files
         * @param {string} path Directory path to search in
         * @param {string} pattern Text pattern to search for
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async findInFiles(path, pattern, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.findInFiles(path, pattern, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['FileSystemApi.findInFiles']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get detailed information about a file or directory
         * @summary Get file information
         * @param {string} path File or directory path
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getFileInfo(path, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getFileInfo(path, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['FileSystemApi.getFileInfo']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * List files and directories in the specified path
         * @summary List files and directories
         * @param {string} [path] Directory path to list (defaults to working directory)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listFiles(path, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listFiles(path, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['FileSystemApi.listFiles']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Move or rename a file or directory from source to destination
         * @summary Move or rename file/directory
         * @param {string} source Source file or directory path
         * @param {string} destination Destination file or directory path
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async moveFile(source, destination, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.moveFile(source, destination, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['FileSystemApi.moveFile']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Replace text pattern with new value in multiple files
         * @summary Replace text in files
         * @param {ReplaceRequest} request Replace request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async replaceInFiles(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.replaceInFiles(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['FileSystemApi.replaceInFiles']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Search for files matching a specific pattern in a directory
         * @summary Search files by pattern
         * @param {string} path Directory path to search in
         * @param {string} pattern File pattern to match (e.g., *.txt, *.go)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async searchFiles(path, pattern, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.searchFiles(path, pattern, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['FileSystemApi.searchFiles']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Set file permissions, ownership, and group for a file or directory
         * @summary Set file permissions
         * @param {string} path File or directory path
         * @param {string} [owner] Owner (username or UID)
         * @param {string} [group] Group (group name or GID)
         * @param {string} [mode] File mode in octal format (e.g., 0755)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async setFilePermissions(path, owner, group, mode, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.setFilePermissions(path, owner, group, mode, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['FileSystemApi.setFilePermissions']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Upload a file to the specified path
         * @summary Upload a file
         * @param {string} path Destination path for the uploaded file
         * @param {File} file File to upload
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async uploadFile(path, file, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.uploadFile(path, file, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['FileSystemApi.uploadFile']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Upload multiple files with their destination paths
         * @summary Upload multiple files
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async uploadFiles(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.uploadFiles(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['FileSystemApi.uploadFiles']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.FileSystemApiFp = FileSystemApiFp;
/**
 * FileSystemApi - factory interface
 */
const FileSystemApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.FileSystemApiFp)(configuration);
    return {
        /**
         * Create a folder with the specified path and optional permissions
         * @summary Create a folder
         * @param {string} path Folder path to create
         * @param {string} mode Octal permission mode (default: 0755)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createFolder(path, mode, options) {
            return localVarFp.createFolder(path, mode, options).then((request) => request(axios, basePath));
        },
        /**
         * Delete a file or directory at the specified path
         * @summary Delete a file or directory
         * @param {string} path File or directory path to delete
         * @param {boolean} [recursive] Enable recursive deletion for directories
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteFile(path, recursive, options) {
            return localVarFp.deleteFile(path, recursive, options).then((request) => request(axios, basePath));
        },
        /**
         * Download a file by providing its path
         * @summary Download a file
         * @param {string} path File path to download
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        downloadFile(path, options) {
            return localVarFp.downloadFile(path, options).then((request) => request(axios, basePath));
        },
        /**
         * Download multiple files by providing their paths. Successful files are returned as multipart parts named `file`. Per-file failures are returned as multipart parts named `error` with JSON payloads shaped like ErrorResponse.
         * @summary Download multiple files
         * @param {FilesDownloadRequest} downloadFiles Paths of files to download
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        downloadFiles(downloadFiles, options) {
            return localVarFp.downloadFiles(downloadFiles, options).then((request) => request(axios, basePath));
        },
        /**
         * Search for text pattern within files in a directory
         * @summary Find text in files
         * @param {string} path Directory path to search in
         * @param {string} pattern Text pattern to search for
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        findInFiles(path, pattern, options) {
            return localVarFp.findInFiles(path, pattern, options).then((request) => request(axios, basePath));
        },
        /**
         * Get detailed information about a file or directory
         * @summary Get file information
         * @param {string} path File or directory path
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getFileInfo(path, options) {
            return localVarFp.getFileInfo(path, options).then((request) => request(axios, basePath));
        },
        /**
         * List files and directories in the specified path
         * @summary List files and directories
         * @param {string} [path] Directory path to list (defaults to working directory)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listFiles(path, options) {
            return localVarFp.listFiles(path, options).then((request) => request(axios, basePath));
        },
        /**
         * Move or rename a file or directory from source to destination
         * @summary Move or rename file/directory
         * @param {string} source Source file or directory path
         * @param {string} destination Destination file or directory path
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        moveFile(source, destination, options) {
            return localVarFp.moveFile(source, destination, options).then((request) => request(axios, basePath));
        },
        /**
         * Replace text pattern with new value in multiple files
         * @summary Replace text in files
         * @param {ReplaceRequest} request Replace request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        replaceInFiles(request, options) {
            return localVarFp.replaceInFiles(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Search for files matching a specific pattern in a directory
         * @summary Search files by pattern
         * @param {string} path Directory path to search in
         * @param {string} pattern File pattern to match (e.g., *.txt, *.go)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        searchFiles(path, pattern, options) {
            return localVarFp.searchFiles(path, pattern, options).then((request) => request(axios, basePath));
        },
        /**
         * Set file permissions, ownership, and group for a file or directory
         * @summary Set file permissions
         * @param {string} path File or directory path
         * @param {string} [owner] Owner (username or UID)
         * @param {string} [group] Group (group name or GID)
         * @param {string} [mode] File mode in octal format (e.g., 0755)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        setFilePermissions(path, owner, group, mode, options) {
            return localVarFp.setFilePermissions(path, owner, group, mode, options).then((request) => request(axios, basePath));
        },
        /**
         * Upload a file to the specified path
         * @summary Upload a file
         * @param {string} path Destination path for the uploaded file
         * @param {File} file File to upload
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        uploadFile(path, file, options) {
            return localVarFp.uploadFile(path, file, options).then((request) => request(axios, basePath));
        },
        /**
         * Upload multiple files with their destination paths
         * @summary Upload multiple files
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        uploadFiles(options) {
            return localVarFp.uploadFiles(options).then((request) => request(axios, basePath));
        },
    };
};
exports.FileSystemApiFactory = FileSystemApiFactory;
/**
 * FileSystemApi - object-oriented interface
 */
class FileSystemApi extends base_1.BaseAPI {
    /**
     * Create a folder with the specified path and optional permissions
     * @summary Create a folder
     * @param {string} path Folder path to create
     * @param {string} mode Octal permission mode (default: 0755)
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    createFolder(path, mode, options) {
        return (0, exports.FileSystemApiFp)(this.configuration).createFolder(path, mode, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Delete a file or directory at the specified path
     * @summary Delete a file or directory
     * @param {string} path File or directory path to delete
     * @param {boolean} [recursive] Enable recursive deletion for directories
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    deleteFile(path, recursive, options) {
        return (0, exports.FileSystemApiFp)(this.configuration).deleteFile(path, recursive, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Download a file by providing its path
     * @summary Download a file
     * @param {string} path File path to download
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    downloadFile(path, options) {
        return (0, exports.FileSystemApiFp)(this.configuration).downloadFile(path, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Download multiple files by providing their paths. Successful files are returned as multipart parts named `file`. Per-file failures are returned as multipart parts named `error` with JSON payloads shaped like ErrorResponse.
     * @summary Download multiple files
     * @param {FilesDownloadRequest} downloadFiles Paths of files to download
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    downloadFiles(downloadFiles, options) {
        return (0, exports.FileSystemApiFp)(this.configuration).downloadFiles(downloadFiles, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Search for text pattern within files in a directory
     * @summary Find text in files
     * @param {string} path Directory path to search in
     * @param {string} pattern Text pattern to search for
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    findInFiles(path, pattern, options) {
        return (0, exports.FileSystemApiFp)(this.configuration).findInFiles(path, pattern, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get detailed information about a file or directory
     * @summary Get file information
     * @param {string} path File or directory path
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getFileInfo(path, options) {
        return (0, exports.FileSystemApiFp)(this.configuration).getFileInfo(path, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * List files and directories in the specified path
     * @summary List files and directories
     * @param {string} [path] Directory path to list (defaults to working directory)
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    listFiles(path, options) {
        return (0, exports.FileSystemApiFp)(this.configuration).listFiles(path, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Move or rename a file or directory from source to destination
     * @summary Move or rename file/directory
     * @param {string} source Source file or directory path
     * @param {string} destination Destination file or directory path
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    moveFile(source, destination, options) {
        return (0, exports.FileSystemApiFp)(this.configuration).moveFile(source, destination, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Replace text pattern with new value in multiple files
     * @summary Replace text in files
     * @param {ReplaceRequest} request Replace request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    replaceInFiles(request, options) {
        return (0, exports.FileSystemApiFp)(this.configuration).replaceInFiles(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Search for files matching a specific pattern in a directory
     * @summary Search files by pattern
     * @param {string} path Directory path to search in
     * @param {string} pattern File pattern to match (e.g., *.txt, *.go)
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    searchFiles(path, pattern, options) {
        return (0, exports.FileSystemApiFp)(this.configuration).searchFiles(path, pattern, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Set file permissions, ownership, and group for a file or directory
     * @summary Set file permissions
     * @param {string} path File or directory path
     * @param {string} [owner] Owner (username or UID)
     * @param {string} [group] Group (group name or GID)
     * @param {string} [mode] File mode in octal format (e.g., 0755)
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    setFilePermissions(path, owner, group, mode, options) {
        return (0, exports.FileSystemApiFp)(this.configuration).setFilePermissions(path, owner, group, mode, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Upload a file to the specified path
     * @summary Upload a file
     * @param {string} path Destination path for the uploaded file
     * @param {File} file File to upload
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    uploadFile(path, file, options) {
        return (0, exports.FileSystemApiFp)(this.configuration).uploadFile(path, file, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Upload multiple files with their destination paths
     * @summary Upload multiple files
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    uploadFiles(options) {
        return (0, exports.FileSystemApiFp)(this.configuration).uploadFiles(options).then((request) => request(this.axios, this.basePath));
    }
}
exports.FileSystemApi = FileSystemApi;
