// @daytona/toolbox-api-client/src/api/git-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Daytona Toolbox API
 * Daytona Toolbox API. The base URL comes from the sandbox\'s `toolboxProxyUrl` field (returned in sandbox DTO by the main Daytona API) plus the sandbox ID: `{toolboxProxyUrl}/{sandboxId}/{endpoint}`. Default for Daytona Cloud: `https://proxy.app.daytona.io/toolbox/{sandboxId}`.
 *
 * The version of the OpenAPI document: v0.0.0-dev
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GitApi = exports.GitApiFactory = exports.GitApiFp = exports.GitApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * GitApi - axios parameter creator
 */
const GitApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Add files to the Git staging area
         * @summary Add files to Git staging
         * @param {GitAddRequest} request Add files request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        addFiles: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('addFiles', 'request', request);
            const localVarPath = `/git/add`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Switch to a different branch or commit in the Git repository
         * @summary Checkout branch or commit
         * @param {GitCheckoutRequest} request Checkout request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        checkoutBranch: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('checkoutBranch', 'request', request);
            const localVarPath = `/git/checkout`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Clone a Git repository to the specified path. Defaults to strict TLS verification; set insecure_skip_tls=true to skip verification for self-signed or private-CA Git servers.
         * @summary Clone a Git repository
         * @param {GitCloneRequest} request Clone repository request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        cloneRepository: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('cloneRepository', 'request', request);
            const localVarPath = `/git/clone`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Commit staged changes to the Git repository
         * @summary Commit changes
         * @param {GitCommitRequest} request Commit request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        commitChanges: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('commitChanges', 'request', request);
            const localVarPath = `/git/commit`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Create a new branch in the Git repository
         * @summary Create a new branch
         * @param {GitBranchRequest} request Create branch request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createBranch: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('createBranch', 'request', request);
            const localVarPath = `/git/branches`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Delete a branch from the Git repository
         * @summary Delete a branch
         * @param {GitDeleteBranchRequest} request Delete branch request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteBranch: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('deleteBranch', 'request', request);
            const localVarPath = `/git/branches`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'DELETE', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get the commit history of the Git repository
         * @summary Get commit history
         * @param {string} path Repository path
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getCommitHistory: async (path, options = {}) => {
            // verify required parameter 'path' is not null or undefined
            (0, common_1.assertParamExists)('getCommitHistory', 'path', path);
            const localVarPath = `/git/history`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (path !== undefined) {
                localVarQueryParameter['path'] = path;
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get the Git status of the repository at the specified path
         * @summary Get Git status
         * @param {string} path Repository path
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getStatus: async (path, options = {}) => {
            // verify required parameter 'path' is not null or undefined
            (0, common_1.assertParamExists)('getStatus', 'path', path);
            const localVarPath = `/git/status`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (path !== undefined) {
                localVarQueryParameter['path'] = path;
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get a list of all branches in the Git repository
         * @summary List branches
         * @param {string} path Repository path
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listBranches: async (path, options = {}) => {
            // verify required parameter 'path' is not null or undefined
            (0, common_1.assertParamExists)('listBranches', 'path', path);
            const localVarPath = `/git/branches`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (path !== undefined) {
                localVarQueryParameter['path'] = path;
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Pull changes from the remote Git repository
         * @summary Pull changes from remote
         * @param {GitRepoRequest} request Pull request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pullChanges: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('pullChanges', 'request', request);
            const localVarPath = `/git/pull`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Push local changes to the remote Git repository
         * @summary Push changes to remote
         * @param {GitRepoRequest} request Push request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pushChanges: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('pushChanges', 'request', request);
            const localVarPath = `/git/push`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.GitApiAxiosParamCreator = GitApiAxiosParamCreator;
/**
 * GitApi - functional programming interface
 */
const GitApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.GitApiAxiosParamCreator)(configuration);
    return {
        /**
         * Add files to the Git staging area
         * @summary Add files to Git staging
         * @param {GitAddRequest} request Add files request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async addFiles(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.addFiles(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['GitApi.addFiles']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Switch to a different branch or commit in the Git repository
         * @summary Checkout branch or commit
         * @param {GitCheckoutRequest} request Checkout request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async checkoutBranch(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.checkoutBranch(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['GitApi.checkoutBranch']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Clone a Git repository to the specified path. Defaults to strict TLS verification; set insecure_skip_tls=true to skip verification for self-signed or private-CA Git servers.
         * @summary Clone a Git repository
         * @param {GitCloneRequest} request Clone repository request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async cloneRepository(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.cloneRepository(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['GitApi.cloneRepository']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Commit staged changes to the Git repository
         * @summary Commit changes
         * @param {GitCommitRequest} request Commit request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async commitChanges(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.commitChanges(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['GitApi.commitChanges']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Create a new branch in the Git repository
         * @summary Create a new branch
         * @param {GitBranchRequest} request Create branch request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async createBranch(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.createBranch(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['GitApi.createBranch']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Delete a branch from the Git repository
         * @summary Delete a branch
         * @param {GitDeleteBranchRequest} request Delete branch request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async deleteBranch(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.deleteBranch(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['GitApi.deleteBranch']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get the commit history of the Git repository
         * @summary Get commit history
         * @param {string} path Repository path
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getCommitHistory(path, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getCommitHistory(path, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['GitApi.getCommitHistory']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get the Git status of the repository at the specified path
         * @summary Get Git status
         * @param {string} path Repository path
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getStatus(path, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getStatus(path, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['GitApi.getStatus']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get a list of all branches in the Git repository
         * @summary List branches
         * @param {string} path Repository path
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listBranches(path, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listBranches(path, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['GitApi.listBranches']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Pull changes from the remote Git repository
         * @summary Pull changes from remote
         * @param {GitRepoRequest} request Pull request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async pullChanges(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.pullChanges(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['GitApi.pullChanges']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Push local changes to the remote Git repository
         * @summary Push changes to remote
         * @param {GitRepoRequest} request Push request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async pushChanges(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.pushChanges(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['GitApi.pushChanges']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.GitApiFp = GitApiFp;
/**
 * GitApi - factory interface
 */
const GitApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.GitApiFp)(configuration);
    return {
        /**
         * Add files to the Git staging area
         * @summary Add files to Git staging
         * @param {GitAddRequest} request Add files request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        addFiles(request, options) {
            return localVarFp.addFiles(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Switch to a different branch or commit in the Git repository
         * @summary Checkout branch or commit
         * @param {GitCheckoutRequest} request Checkout request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        checkoutBranch(request, options) {
            return localVarFp.checkoutBranch(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Clone a Git repository to the specified path. Defaults to strict TLS verification; set insecure_skip_tls=true to skip verification for self-signed or private-CA Git servers.
         * @summary Clone a Git repository
         * @param {GitCloneRequest} request Clone repository request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        cloneRepository(request, options) {
            return localVarFp.cloneRepository(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Commit staged changes to the Git repository
         * @summary Commit changes
         * @param {GitCommitRequest} request Commit request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        commitChanges(request, options) {
            return localVarFp.commitChanges(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Create a new branch in the Git repository
         * @summary Create a new branch
         * @param {GitBranchRequest} request Create branch request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createBranch(request, options) {
            return localVarFp.createBranch(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Delete a branch from the Git repository
         * @summary Delete a branch
         * @param {GitDeleteBranchRequest} request Delete branch request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteBranch(request, options) {
            return localVarFp.deleteBranch(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Get the commit history of the Git repository
         * @summary Get commit history
         * @param {string} path Repository path
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getCommitHistory(path, options) {
            return localVarFp.getCommitHistory(path, options).then((request) => request(axios, basePath));
        },
        /**
         * Get the Git status of the repository at the specified path
         * @summary Get Git status
         * @param {string} path Repository path
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getStatus(path, options) {
            return localVarFp.getStatus(path, options).then((request) => request(axios, basePath));
        },
        /**
         * Get a list of all branches in the Git repository
         * @summary List branches
         * @param {string} path Repository path
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listBranches(path, options) {
            return localVarFp.listBranches(path, options).then((request) => request(axios, basePath));
        },
        /**
         * Pull changes from the remote Git repository
         * @summary Pull changes from remote
         * @param {GitRepoRequest} request Pull request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pullChanges(request, options) {
            return localVarFp.pullChanges(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Push local changes to the remote Git repository
         * @summary Push changes to remote
         * @param {GitRepoRequest} request Push request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pushChanges(request, options) {
            return localVarFp.pushChanges(request, options).then((request) => request(axios, basePath));
        },
    };
};
exports.GitApiFactory = GitApiFactory;
/**
 * GitApi - object-oriented interface
 */
class GitApi extends base_1.BaseAPI {
    /**
     * Add files to the Git staging area
     * @summary Add files to Git staging
     * @param {GitAddRequest} request Add files request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    addFiles(request, options) {
        return (0, exports.GitApiFp)(this.configuration).addFiles(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Switch to a different branch or commit in the Git repository
     * @summary Checkout branch or commit
     * @param {GitCheckoutRequest} request Checkout request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    checkoutBranch(request, options) {
        return (0, exports.GitApiFp)(this.configuration).checkoutBranch(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Clone a Git repository to the specified path. Defaults to strict TLS verification; set insecure_skip_tls=true to skip verification for self-signed or private-CA Git servers.
     * @summary Clone a Git repository
     * @param {GitCloneRequest} request Clone repository request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    cloneRepository(request, options) {
        return (0, exports.GitApiFp)(this.configuration).cloneRepository(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Commit staged changes to the Git repository
     * @summary Commit changes
     * @param {GitCommitRequest} request Commit request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    commitChanges(request, options) {
        return (0, exports.GitApiFp)(this.configuration).commitChanges(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Create a new branch in the Git repository
     * @summary Create a new branch
     * @param {GitBranchRequest} request Create branch request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    createBranch(request, options) {
        return (0, exports.GitApiFp)(this.configuration).createBranch(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Delete a branch from the Git repository
     * @summary Delete a branch
     * @param {GitDeleteBranchRequest} request Delete branch request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    deleteBranch(request, options) {
        return (0, exports.GitApiFp)(this.configuration).deleteBranch(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get the commit history of the Git repository
     * @summary Get commit history
     * @param {string} path Repository path
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getCommitHistory(path, options) {
        return (0, exports.GitApiFp)(this.configuration).getCommitHistory(path, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get the Git status of the repository at the specified path
     * @summary Get Git status
     * @param {string} path Repository path
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getStatus(path, options) {
        return (0, exports.GitApiFp)(this.configuration).getStatus(path, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get a list of all branches in the Git repository
     * @summary List branches
     * @param {string} path Repository path
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    listBranches(path, options) {
        return (0, exports.GitApiFp)(this.configuration).listBranches(path, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Pull changes from the remote Git repository
     * @summary Pull changes from remote
     * @param {GitRepoRequest} request Pull request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    pullChanges(request, options) {
        return (0, exports.GitApiFp)(this.configuration).pullChanges(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Push local changes to the remote Git repository
     * @summary Push changes to remote
     * @param {GitRepoRequest} request Push request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    pushChanges(request, options) {
        return (0, exports.GitApiFp)(this.configuration).pushChanges(request, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.GitApi = GitApi;
