// @daytona/toolbox-api-client/src/api/info-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Daytona Toolbox API
 * Daytona Toolbox API. The base URL comes from the sandbox\'s `toolboxProxyUrl` field (returned in sandbox DTO by the main Daytona API) plus the sandbox ID: `{toolboxProxyUrl}/{sandboxId}/{endpoint}`. Default for Daytona Cloud: `https://proxy.app.daytona.io/toolbox/{sandboxId}`.
 *
 * The version of the OpenAPI document: v0.0.0-dev
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InfoApi = exports.InfoApiFactory = exports.InfoApiFp = exports.InfoApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * InfoApi - axios parameter creator
 */
const InfoApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Get the current user home directory path.
         * @summary Get user home directory
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getUserHomeDir: async (options = {}) => {
            const localVarPath = `/user-home-dir`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get the current daemon version
         * @summary Get version
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getVersion: async (options = {}) => {
            const localVarPath = `/version`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get the current working directory path. This is default directory used for running commands.
         * @summary Get working directory
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getWorkDir: async (options = {}) => {
            const localVarPath = `/work-dir`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.InfoApiAxiosParamCreator = InfoApiAxiosParamCreator;
/**
 * InfoApi - functional programming interface
 */
const InfoApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.InfoApiAxiosParamCreator)(configuration);
    return {
        /**
         * Get the current user home directory path.
         * @summary Get user home directory
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getUserHomeDir(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getUserHomeDir(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['InfoApi.getUserHomeDir']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get the current daemon version
         * @summary Get version
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getVersion(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getVersion(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['InfoApi.getVersion']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get the current working directory path. This is default directory used for running commands.
         * @summary Get working directory
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getWorkDir(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getWorkDir(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['InfoApi.getWorkDir']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.InfoApiFp = InfoApiFp;
/**
 * InfoApi - factory interface
 */
const InfoApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.InfoApiFp)(configuration);
    return {
        /**
         * Get the current user home directory path.
         * @summary Get user home directory
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getUserHomeDir(options) {
            return localVarFp.getUserHomeDir(options).then((request) => request(axios, basePath));
        },
        /**
         * Get the current daemon version
         * @summary Get version
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getVersion(options) {
            return localVarFp.getVersion(options).then((request) => request(axios, basePath));
        },
        /**
         * Get the current working directory path. This is default directory used for running commands.
         * @summary Get working directory
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getWorkDir(options) {
            return localVarFp.getWorkDir(options).then((request) => request(axios, basePath));
        },
    };
};
exports.InfoApiFactory = InfoApiFactory;
/**
 * InfoApi - object-oriented interface
 */
class InfoApi extends base_1.BaseAPI {
    /**
     * Get the current user home directory path.
     * @summary Get user home directory
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getUserHomeDir(options) {
        return (0, exports.InfoApiFp)(this.configuration).getUserHomeDir(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get the current daemon version
     * @summary Get version
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getVersion(options) {
        return (0, exports.InfoApiFp)(this.configuration).getVersion(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get the current working directory path. This is default directory used for running commands.
     * @summary Get working directory
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getWorkDir(options) {
        return (0, exports.InfoApiFp)(this.configuration).getWorkDir(options).then((request) => request(this.axios, this.basePath));
    }
}
exports.InfoApi = InfoApi;
