// @daytona/toolbox-api-client/src/api/interpreter-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Daytona Toolbox API
 * Daytona Toolbox API. The base URL comes from the sandbox\'s `toolboxProxyUrl` field (returned in sandbox DTO by the main Daytona API) plus the sandbox ID: `{toolboxProxyUrl}/{sandboxId}/{endpoint}`. Default for Daytona Cloud: `https://proxy.app.daytona.io/toolbox/{sandboxId}`.
 *
 * The version of the OpenAPI document: v0.0.0-dev
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InterpreterApi = exports.InterpreterApiFactory = exports.InterpreterApiFp = exports.InterpreterApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * InterpreterApi - axios parameter creator
 */
const InterpreterApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Creates a new isolated interpreter context with optional working directory and language
         * @summary Create a new interpreter context
         * @param {CreateContextRequest} request Context configuration
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createInterpreterContext: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('createInterpreterContext', 'request', request);
            const localVarPath = `/process/interpreter/context`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Deletes an interpreter context and shuts down its worker process
         * @summary Delete an interpreter context
         * @param {string} id Context ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteInterpreterContext: async (id, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('deleteInterpreterContext', 'id', id);
            const localVarPath = `/process/interpreter/context/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'DELETE', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Executes code in a specified context (or default context if not specified) via WebSocket streaming
         * @summary Execute code in an interpreter context
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        executeInterpreterCode: async (options = {}) => {
            const localVarPath = `/process/interpreter/execute`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Returns information about all user-created interpreter contexts (excludes default context)
         * @summary List all user-created interpreter contexts
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listInterpreterContexts: async (options = {}) => {
            const localVarPath = `/process/interpreter/context`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.InterpreterApiAxiosParamCreator = InterpreterApiAxiosParamCreator;
/**
 * InterpreterApi - functional programming interface
 */
const InterpreterApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.InterpreterApiAxiosParamCreator)(configuration);
    return {
        /**
         * Creates a new isolated interpreter context with optional working directory and language
         * @summary Create a new interpreter context
         * @param {CreateContextRequest} request Context configuration
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async createInterpreterContext(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.createInterpreterContext(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['InterpreterApi.createInterpreterContext']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Deletes an interpreter context and shuts down its worker process
         * @summary Delete an interpreter context
         * @param {string} id Context ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async deleteInterpreterContext(id, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.deleteInterpreterContext(id, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['InterpreterApi.deleteInterpreterContext']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Executes code in a specified context (or default context if not specified) via WebSocket streaming
         * @summary Execute code in an interpreter context
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async executeInterpreterCode(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.executeInterpreterCode(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['InterpreterApi.executeInterpreterCode']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Returns information about all user-created interpreter contexts (excludes default context)
         * @summary List all user-created interpreter contexts
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listInterpreterContexts(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listInterpreterContexts(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['InterpreterApi.listInterpreterContexts']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.InterpreterApiFp = InterpreterApiFp;
/**
 * InterpreterApi - factory interface
 */
const InterpreterApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.InterpreterApiFp)(configuration);
    return {
        /**
         * Creates a new isolated interpreter context with optional working directory and language
         * @summary Create a new interpreter context
         * @param {CreateContextRequest} request Context configuration
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createInterpreterContext(request, options) {
            return localVarFp.createInterpreterContext(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Deletes an interpreter context and shuts down its worker process
         * @summary Delete an interpreter context
         * @param {string} id Context ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteInterpreterContext(id, options) {
            return localVarFp.deleteInterpreterContext(id, options).then((request) => request(axios, basePath));
        },
        /**
         * Executes code in a specified context (or default context if not specified) via WebSocket streaming
         * @summary Execute code in an interpreter context
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        executeInterpreterCode(options) {
            return localVarFp.executeInterpreterCode(options).then((request) => request(axios, basePath));
        },
        /**
         * Returns information about all user-created interpreter contexts (excludes default context)
         * @summary List all user-created interpreter contexts
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listInterpreterContexts(options) {
            return localVarFp.listInterpreterContexts(options).then((request) => request(axios, basePath));
        },
    };
};
exports.InterpreterApiFactory = InterpreterApiFactory;
/**
 * InterpreterApi - object-oriented interface
 */
class InterpreterApi extends base_1.BaseAPI {
    /**
     * Creates a new isolated interpreter context with optional working directory and language
     * @summary Create a new interpreter context
     * @param {CreateContextRequest} request Context configuration
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    createInterpreterContext(request, options) {
        return (0, exports.InterpreterApiFp)(this.configuration).createInterpreterContext(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Deletes an interpreter context and shuts down its worker process
     * @summary Delete an interpreter context
     * @param {string} id Context ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    deleteInterpreterContext(id, options) {
        return (0, exports.InterpreterApiFp)(this.configuration).deleteInterpreterContext(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Executes code in a specified context (or default context if not specified) via WebSocket streaming
     * @summary Execute code in an interpreter context
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    executeInterpreterCode(options) {
        return (0, exports.InterpreterApiFp)(this.configuration).executeInterpreterCode(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Returns information about all user-created interpreter contexts (excludes default context)
     * @summary List all user-created interpreter contexts
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    listInterpreterContexts(options) {
        return (0, exports.InterpreterApiFp)(this.configuration).listInterpreterContexts(options).then((request) => request(this.axios, this.basePath));
    }
}
exports.InterpreterApi = InterpreterApi;
