// @daytona/toolbox-api-client/src/api/lsp-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Daytona Toolbox API
 * Daytona Toolbox API. The base URL comes from the sandbox\'s `toolboxProxyUrl` field (returned in sandbox DTO by the main Daytona API) plus the sandbox ID: `{toolboxProxyUrl}/{sandboxId}/{endpoint}`. Default for Daytona Cloud: `https://proxy.app.daytona.io/toolbox/{sandboxId}`.
 *
 * The version of the OpenAPI document: v0.0.0-dev
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LspApi = exports.LspApiFactory = exports.LspApiFp = exports.LspApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * LspApi - axios parameter creator
 */
const LspApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Get code completion suggestions from the LSP server
         * @summary Get code completions
         * @param {LspCompletionParams} request Completion request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        completions: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('completions', 'request', request);
            const localVarPath = `/lsp/completions`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Notify the LSP server that a document has been closed
         * @summary Notify document closed
         * @param {LspDocumentRequest} request Document request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        didClose: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('didClose', 'request', request);
            const localVarPath = `/lsp/did-close`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Notify the LSP server that a document has been opened
         * @summary Notify document opened
         * @param {LspDocumentRequest} request Document request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        didOpen: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('didOpen', 'request', request);
            const localVarPath = `/lsp/did-open`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get symbols (functions, classes, etc.) from a document
         * @summary Get document symbols
         * @param {string} languageId Language ID (e.g., python, typescript)
         * @param {string} pathToProject Path to project
         * @param {string} uri Document URI
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        documentSymbols: async (languageId, pathToProject, uri, options = {}) => {
            // verify required parameter 'languageId' is not null or undefined
            (0, common_1.assertParamExists)('documentSymbols', 'languageId', languageId);
            // verify required parameter 'pathToProject' is not null or undefined
            (0, common_1.assertParamExists)('documentSymbols', 'pathToProject', pathToProject);
            // verify required parameter 'uri' is not null or undefined
            (0, common_1.assertParamExists)('documentSymbols', 'uri', uri);
            const localVarPath = `/lsp/document-symbols`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (languageId !== undefined) {
                localVarQueryParameter['languageId'] = languageId;
            }
            if (pathToProject !== undefined) {
                localVarQueryParameter['pathToProject'] = pathToProject;
            }
            if (uri !== undefined) {
                localVarQueryParameter['uri'] = uri;
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Start a Language Server Protocol server for the specified language
         * @summary Start LSP server
         * @param {LspServerRequest} request LSP server request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        start: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('start', 'request', request);
            const localVarPath = `/lsp/start`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Stop a Language Server Protocol server
         * @summary Stop LSP server
         * @param {LspServerRequest} request LSP server request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        stop: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('stop', 'request', request);
            const localVarPath = `/lsp/stop`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Search for symbols across the entire workspace
         * @summary Get workspace symbols
         * @param {string} query Search query
         * @param {string} languageId Language ID (e.g., python, typescript)
         * @param {string} pathToProject Path to project
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        workspaceSymbols: async (query, languageId, pathToProject, options = {}) => {
            // verify required parameter 'query' is not null or undefined
            (0, common_1.assertParamExists)('workspaceSymbols', 'query', query);
            // verify required parameter 'languageId' is not null or undefined
            (0, common_1.assertParamExists)('workspaceSymbols', 'languageId', languageId);
            // verify required parameter 'pathToProject' is not null or undefined
            (0, common_1.assertParamExists)('workspaceSymbols', 'pathToProject', pathToProject);
            const localVarPath = `/lsp/workspacesymbols`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (query !== undefined) {
                localVarQueryParameter['query'] = query;
            }
            if (languageId !== undefined) {
                localVarQueryParameter['languageId'] = languageId;
            }
            if (pathToProject !== undefined) {
                localVarQueryParameter['pathToProject'] = pathToProject;
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.LspApiAxiosParamCreator = LspApiAxiosParamCreator;
/**
 * LspApi - functional programming interface
 */
const LspApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.LspApiAxiosParamCreator)(configuration);
    return {
        /**
         * Get code completion suggestions from the LSP server
         * @summary Get code completions
         * @param {LspCompletionParams} request Completion request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async completions(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.completions(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['LspApi.completions']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Notify the LSP server that a document has been closed
         * @summary Notify document closed
         * @param {LspDocumentRequest} request Document request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async didClose(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.didClose(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['LspApi.didClose']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Notify the LSP server that a document has been opened
         * @summary Notify document opened
         * @param {LspDocumentRequest} request Document request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async didOpen(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.didOpen(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['LspApi.didOpen']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get symbols (functions, classes, etc.) from a document
         * @summary Get document symbols
         * @param {string} languageId Language ID (e.g., python, typescript)
         * @param {string} pathToProject Path to project
         * @param {string} uri Document URI
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async documentSymbols(languageId, pathToProject, uri, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.documentSymbols(languageId, pathToProject, uri, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['LspApi.documentSymbols']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Start a Language Server Protocol server for the specified language
         * @summary Start LSP server
         * @param {LspServerRequest} request LSP server request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async start(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.start(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['LspApi.start']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Stop a Language Server Protocol server
         * @summary Stop LSP server
         * @param {LspServerRequest} request LSP server request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async stop(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.stop(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['LspApi.stop']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Search for symbols across the entire workspace
         * @summary Get workspace symbols
         * @param {string} query Search query
         * @param {string} languageId Language ID (e.g., python, typescript)
         * @param {string} pathToProject Path to project
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async workspaceSymbols(query, languageId, pathToProject, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.workspaceSymbols(query, languageId, pathToProject, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['LspApi.workspaceSymbols']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.LspApiFp = LspApiFp;
/**
 * LspApi - factory interface
 */
const LspApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.LspApiFp)(configuration);
    return {
        /**
         * Get code completion suggestions from the LSP server
         * @summary Get code completions
         * @param {LspCompletionParams} request Completion request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        completions(request, options) {
            return localVarFp.completions(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Notify the LSP server that a document has been closed
         * @summary Notify document closed
         * @param {LspDocumentRequest} request Document request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        didClose(request, options) {
            return localVarFp.didClose(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Notify the LSP server that a document has been opened
         * @summary Notify document opened
         * @param {LspDocumentRequest} request Document request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        didOpen(request, options) {
            return localVarFp.didOpen(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Get symbols (functions, classes, etc.) from a document
         * @summary Get document symbols
         * @param {string} languageId Language ID (e.g., python, typescript)
         * @param {string} pathToProject Path to project
         * @param {string} uri Document URI
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        documentSymbols(languageId, pathToProject, uri, options) {
            return localVarFp.documentSymbols(languageId, pathToProject, uri, options).then((request) => request(axios, basePath));
        },
        /**
         * Start a Language Server Protocol server for the specified language
         * @summary Start LSP server
         * @param {LspServerRequest} request LSP server request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        start(request, options) {
            return localVarFp.start(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Stop a Language Server Protocol server
         * @summary Stop LSP server
         * @param {LspServerRequest} request LSP server request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        stop(request, options) {
            return localVarFp.stop(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Search for symbols across the entire workspace
         * @summary Get workspace symbols
         * @param {string} query Search query
         * @param {string} languageId Language ID (e.g., python, typescript)
         * @param {string} pathToProject Path to project
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        workspaceSymbols(query, languageId, pathToProject, options) {
            return localVarFp.workspaceSymbols(query, languageId, pathToProject, options).then((request) => request(axios, basePath));
        },
    };
};
exports.LspApiFactory = LspApiFactory;
/**
 * LspApi - object-oriented interface
 */
class LspApi extends base_1.BaseAPI {
    /**
     * Get code completion suggestions from the LSP server
     * @summary Get code completions
     * @param {LspCompletionParams} request Completion request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    completions(request, options) {
        return (0, exports.LspApiFp)(this.configuration).completions(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Notify the LSP server that a document has been closed
     * @summary Notify document closed
     * @param {LspDocumentRequest} request Document request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    didClose(request, options) {
        return (0, exports.LspApiFp)(this.configuration).didClose(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Notify the LSP server that a document has been opened
     * @summary Notify document opened
     * @param {LspDocumentRequest} request Document request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    didOpen(request, options) {
        return (0, exports.LspApiFp)(this.configuration).didOpen(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get symbols (functions, classes, etc.) from a document
     * @summary Get document symbols
     * @param {string} languageId Language ID (e.g., python, typescript)
     * @param {string} pathToProject Path to project
     * @param {string} uri Document URI
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    documentSymbols(languageId, pathToProject, uri, options) {
        return (0, exports.LspApiFp)(this.configuration).documentSymbols(languageId, pathToProject, uri, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Start a Language Server Protocol server for the specified language
     * @summary Start LSP server
     * @param {LspServerRequest} request LSP server request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    start(request, options) {
        return (0, exports.LspApiFp)(this.configuration).start(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Stop a Language Server Protocol server
     * @summary Stop LSP server
     * @param {LspServerRequest} request LSP server request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    stop(request, options) {
        return (0, exports.LspApiFp)(this.configuration).stop(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Search for symbols across the entire workspace
     * @summary Get workspace symbols
     * @param {string} query Search query
     * @param {string} languageId Language ID (e.g., python, typescript)
     * @param {string} pathToProject Path to project
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    workspaceSymbols(query, languageId, pathToProject, options) {
        return (0, exports.LspApiFp)(this.configuration).workspaceSymbols(query, languageId, pathToProject, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.LspApi = LspApi;
