// @daytona/toolbox-api-client/src/api/port-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Daytona Toolbox API
 * Daytona Toolbox API. The base URL comes from the sandbox\'s `toolboxProxyUrl` field (returned in sandbox DTO by the main Daytona API) plus the sandbox ID: `{toolboxProxyUrl}/{sandboxId}/{endpoint}`. Default for Daytona Cloud: `https://proxy.app.daytona.io/toolbox/{sandboxId}`.
 *
 * The version of the OpenAPI document: v0.0.0-dev
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PortApi = exports.PortApiFactory = exports.PortApiFp = exports.PortApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * PortApi - axios parameter creator
 */
const PortApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Get a list of all currently active ports
         * @summary Get active ports
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPorts: async (options = {}) => {
            const localVarPath = `/port`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Check if a specific port is currently in use
         * @summary Check if port is in use
         * @param {number} port Port number (3000-9999)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        isPortInUse: async (port, options = {}) => {
            // verify required parameter 'port' is not null or undefined
            (0, common_1.assertParamExists)('isPortInUse', 'port', port);
            const localVarPath = `/port/{port}/in-use`
                .replace(`{${"port"}}`, encodeURIComponent(String(port)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.PortApiAxiosParamCreator = PortApiAxiosParamCreator;
/**
 * PortApi - functional programming interface
 */
const PortApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.PortApiAxiosParamCreator)(configuration);
    return {
        /**
         * Get a list of all currently active ports
         * @summary Get active ports
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getPorts(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getPorts(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['PortApi.getPorts']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Check if a specific port is currently in use
         * @summary Check if port is in use
         * @param {number} port Port number (3000-9999)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async isPortInUse(port, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.isPortInUse(port, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['PortApi.isPortInUse']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.PortApiFp = PortApiFp;
/**
 * PortApi - factory interface
 */
const PortApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.PortApiFp)(configuration);
    return {
        /**
         * Get a list of all currently active ports
         * @summary Get active ports
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPorts(options) {
            return localVarFp.getPorts(options).then((request) => request(axios, basePath));
        },
        /**
         * Check if a specific port is currently in use
         * @summary Check if port is in use
         * @param {number} port Port number (3000-9999)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        isPortInUse(port, options) {
            return localVarFp.isPortInUse(port, options).then((request) => request(axios, basePath));
        },
    };
};
exports.PortApiFactory = PortApiFactory;
/**
 * PortApi - object-oriented interface
 */
class PortApi extends base_1.BaseAPI {
    /**
     * Get a list of all currently active ports
     * @summary Get active ports
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getPorts(options) {
        return (0, exports.PortApiFp)(this.configuration).getPorts(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Check if a specific port is currently in use
     * @summary Check if port is in use
     * @param {number} port Port number (3000-9999)
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    isPortInUse(port, options) {
        return (0, exports.PortApiFp)(this.configuration).isPortInUse(port, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.PortApi = PortApi;
