// @daytona/toolbox-api-client/src/api/process-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Daytona Toolbox API
 * Daytona Toolbox API. The base URL comes from the sandbox\'s `toolboxProxyUrl` field (returned in sandbox DTO by the main Daytona API) plus the sandbox ID: `{toolboxProxyUrl}/{sandboxId}/{endpoint}`. Default for Daytona Cloud: `https://proxy.app.daytona.io/toolbox/{sandboxId}`.
 *
 * The version of the OpenAPI document: v0.0.0-dev
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProcessApi = exports.ProcessApiFactory = exports.ProcessApiFp = exports.ProcessApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * ProcessApi - axios parameter creator
 */
const ProcessApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Execute Python, JavaScript, or TypeScript code and return output, exit code, and artifacts
         * @summary Execute code
         * @param {CodeRunRequest} request Code execution request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        codeRun: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('codeRun', 'request', request);
            const localVarPath = `/process/code-run`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Establish a WebSocket connection to interact with a pseudo-terminal session
         * @summary Connect to PTY session via WebSocket
         * @param {string} sessionId PTY session ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        connectPtySession: async (sessionId, options = {}) => {
            // verify required parameter 'sessionId' is not null or undefined
            (0, common_1.assertParamExists)('connectPtySession', 'sessionId', sessionId);
            const localVarPath = `/process/pty/{sessionId}/connect`
                .replace(`{${"sessionId"}}`, encodeURIComponent(String(sessionId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Create a new pseudo-terminal session with specified configuration
         * @summary Create a new PTY session
         * @param {PtyCreateRequest} request PTY session creation request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createPtySession: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('createPtySession', 'request', request);
            const localVarPath = `/process/pty`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Create a new shell session for command execution
         * @summary Create a new session
         * @param {CreateSessionRequest} request Session creation request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createSession: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('createSession', 'request', request);
            const localVarPath = `/process/session`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Delete a pseudo-terminal session and terminate its process
         * @summary Delete a PTY session
         * @param {string} sessionId PTY session ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deletePtySession: async (sessionId, options = {}) => {
            // verify required parameter 'sessionId' is not null or undefined
            (0, common_1.assertParamExists)('deletePtySession', 'sessionId', sessionId);
            const localVarPath = `/process/pty/{sessionId}`
                .replace(`{${"sessionId"}}`, encodeURIComponent(String(sessionId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'DELETE', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Delete an existing shell session
         * @summary Delete a session
         * @param {string} sessionId Session ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteSession: async (sessionId, options = {}) => {
            // verify required parameter 'sessionId' is not null or undefined
            (0, common_1.assertParamExists)('deleteSession', 'sessionId', sessionId);
            const localVarPath = `/process/session/{sessionId}`
                .replace(`{${"sessionId"}}`, encodeURIComponent(String(sessionId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'DELETE', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Execute a shell command and return the output and exit code
         * @summary Execute a command
         * @param {ExecuteRequest} request Command execution request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        executeCommand: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('executeCommand', 'request', request);
            const localVarPath = `/process/execute`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get logs for a sandbox entrypoint session. Returns JSON with separated stdout/stderr for SDK >= 0.161.0, plain text otherwise. Supports WebSocket streaming.
         * @summary Get entrypoint logs
         * @param {boolean} [follow] Follow logs in real-time (WebSocket only)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getEntrypointLogs: async (follow, options = {}) => {
            const localVarPath = `/process/session/entrypoint/logs`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (follow !== undefined) {
                localVarQueryParameter['follow'] = follow;
            }
            localVarHeaderParameter['Accept'] = 'application/json,text/plain';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get details of an entrypoint session including its commands
         * @summary Get entrypoint session details
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getEntrypointSession: async (options = {}) => {
            const localVarPath = `/process/session/entrypoint`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get detailed information about a specific pseudo-terminal session
         * @summary Get PTY session information
         * @param {string} sessionId PTY session ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPtySession: async (sessionId, options = {}) => {
            // verify required parameter 'sessionId' is not null or undefined
            (0, common_1.assertParamExists)('getPtySession', 'sessionId', sessionId);
            const localVarPath = `/process/pty/{sessionId}`
                .replace(`{${"sessionId"}}`, encodeURIComponent(String(sessionId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get details of a specific session including its commands
         * @summary Get session details
         * @param {string} sessionId Session ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getSession: async (sessionId, options = {}) => {
            // verify required parameter 'sessionId' is not null or undefined
            (0, common_1.assertParamExists)('getSession', 'sessionId', sessionId);
            const localVarPath = `/process/session/{sessionId}`
                .replace(`{${"sessionId"}}`, encodeURIComponent(String(sessionId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get details of a specific command within a session
         * @summary Get session command details
         * @param {string} sessionId Session ID
         * @param {string} commandId Command ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getSessionCommand: async (sessionId, commandId, options = {}) => {
            // verify required parameter 'sessionId' is not null or undefined
            (0, common_1.assertParamExists)('getSessionCommand', 'sessionId', sessionId);
            // verify required parameter 'commandId' is not null or undefined
            (0, common_1.assertParamExists)('getSessionCommand', 'commandId', commandId);
            const localVarPath = `/process/session/{sessionId}/command/{commandId}`
                .replace(`{${"sessionId"}}`, encodeURIComponent(String(sessionId)))
                .replace(`{${"commandId"}}`, encodeURIComponent(String(commandId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get logs for a specific command within a session. Returns JSON with separated stdout/stderr for SDK >= 0.167.0, plain text otherwise. Supports WebSocket streaming.
         * @summary Get session command logs
         * @param {string} sessionId Session ID
         * @param {string} commandId Command ID
         * @param {boolean} [follow] Follow logs in real-time (WebSocket only)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getSessionCommandLogs: async (sessionId, commandId, follow, options = {}) => {
            // verify required parameter 'sessionId' is not null or undefined
            (0, common_1.assertParamExists)('getSessionCommandLogs', 'sessionId', sessionId);
            // verify required parameter 'commandId' is not null or undefined
            (0, common_1.assertParamExists)('getSessionCommandLogs', 'commandId', commandId);
            const localVarPath = `/process/session/{sessionId}/command/{commandId}/logs`
                .replace(`{${"sessionId"}}`, encodeURIComponent(String(sessionId)))
                .replace(`{${"commandId"}}`, encodeURIComponent(String(commandId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (follow !== undefined) {
                localVarQueryParameter['follow'] = follow;
            }
            localVarHeaderParameter['Accept'] = 'application/json,text/plain';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get a list of all active pseudo-terminal sessions
         * @summary List all PTY sessions
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listPtySessions: async (options = {}) => {
            const localVarPath = `/process/pty`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Get a list of all active shell sessions
         * @summary List all sessions
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listSessions: async (options = {}) => {
            const localVarPath = `/process/session`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Resize the terminal dimensions of a pseudo-terminal session
         * @summary Resize a PTY session
         * @param {string} sessionId PTY session ID
         * @param {PtyResizeRequest} request Resize request with new dimensions
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        resizePtySession: async (sessionId, request, options = {}) => {
            // verify required parameter 'sessionId' is not null or undefined
            (0, common_1.assertParamExists)('resizePtySession', 'sessionId', sessionId);
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('resizePtySession', 'request', request);
            const localVarPath = `/process/pty/{sessionId}/resize`
                .replace(`{${"sessionId"}}`, encodeURIComponent(String(sessionId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Send input data to a running command in a session for interactive execution
         * @summary Send input to command
         * @param {string} sessionId Session ID
         * @param {string} commandId Command ID
         * @param {SessionSendInputRequest} request Input send request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        sendInput: async (sessionId, commandId, request, options = {}) => {
            // verify required parameter 'sessionId' is not null or undefined
            (0, common_1.assertParamExists)('sendInput', 'sessionId', sessionId);
            // verify required parameter 'commandId' is not null or undefined
            (0, common_1.assertParamExists)('sendInput', 'commandId', commandId);
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('sendInput', 'request', request);
            const localVarPath = `/process/session/{sessionId}/command/{commandId}/input`
                .replace(`{${"sessionId"}}`, encodeURIComponent(String(sessionId)))
                .replace(`{${"commandId"}}`, encodeURIComponent(String(commandId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Execute a command within an existing shell session
         * @summary Execute command in session
         * @param {string} sessionId Session ID
         * @param {SessionExecuteRequest} request Command execution request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        sessionExecuteCommand: async (sessionId, request, options = {}) => {
            // verify required parameter 'sessionId' is not null or undefined
            (0, common_1.assertParamExists)('sessionExecuteCommand', 'sessionId', sessionId);
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('sessionExecuteCommand', 'request', request);
            const localVarPath = `/process/session/{sessionId}/exec`
                .replace(`{${"sessionId"}}`, encodeURIComponent(String(sessionId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.ProcessApiAxiosParamCreator = ProcessApiAxiosParamCreator;
/**
 * ProcessApi - functional programming interface
 */
const ProcessApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.ProcessApiAxiosParamCreator)(configuration);
    return {
        /**
         * Execute Python, JavaScript, or TypeScript code and return output, exit code, and artifacts
         * @summary Execute code
         * @param {CodeRunRequest} request Code execution request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async codeRun(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.codeRun(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ProcessApi.codeRun']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Establish a WebSocket connection to interact with a pseudo-terminal session
         * @summary Connect to PTY session via WebSocket
         * @param {string} sessionId PTY session ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async connectPtySession(sessionId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.connectPtySession(sessionId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ProcessApi.connectPtySession']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Create a new pseudo-terminal session with specified configuration
         * @summary Create a new PTY session
         * @param {PtyCreateRequest} request PTY session creation request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async createPtySession(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.createPtySession(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ProcessApi.createPtySession']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Create a new shell session for command execution
         * @summary Create a new session
         * @param {CreateSessionRequest} request Session creation request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async createSession(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.createSession(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ProcessApi.createSession']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Delete a pseudo-terminal session and terminate its process
         * @summary Delete a PTY session
         * @param {string} sessionId PTY session ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async deletePtySession(sessionId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.deletePtySession(sessionId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ProcessApi.deletePtySession']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Delete an existing shell session
         * @summary Delete a session
         * @param {string} sessionId Session ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async deleteSession(sessionId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.deleteSession(sessionId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ProcessApi.deleteSession']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Execute a shell command and return the output and exit code
         * @summary Execute a command
         * @param {ExecuteRequest} request Command execution request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async executeCommand(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.executeCommand(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ProcessApi.executeCommand']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get logs for a sandbox entrypoint session. Returns JSON with separated stdout/stderr for SDK >= 0.161.0, plain text otherwise. Supports WebSocket streaming.
         * @summary Get entrypoint logs
         * @param {boolean} [follow] Follow logs in real-time (WebSocket only)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getEntrypointLogs(follow, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getEntrypointLogs(follow, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ProcessApi.getEntrypointLogs']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get details of an entrypoint session including its commands
         * @summary Get entrypoint session details
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getEntrypointSession(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getEntrypointSession(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ProcessApi.getEntrypointSession']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get detailed information about a specific pseudo-terminal session
         * @summary Get PTY session information
         * @param {string} sessionId PTY session ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getPtySession(sessionId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getPtySession(sessionId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ProcessApi.getPtySession']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get details of a specific session including its commands
         * @summary Get session details
         * @param {string} sessionId Session ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getSession(sessionId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getSession(sessionId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ProcessApi.getSession']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get details of a specific command within a session
         * @summary Get session command details
         * @param {string} sessionId Session ID
         * @param {string} commandId Command ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getSessionCommand(sessionId, commandId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getSessionCommand(sessionId, commandId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ProcessApi.getSessionCommand']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get logs for a specific command within a session. Returns JSON with separated stdout/stderr for SDK >= 0.167.0, plain text otherwise. Supports WebSocket streaming.
         * @summary Get session command logs
         * @param {string} sessionId Session ID
         * @param {string} commandId Command ID
         * @param {boolean} [follow] Follow logs in real-time (WebSocket only)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getSessionCommandLogs(sessionId, commandId, follow, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getSessionCommandLogs(sessionId, commandId, follow, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ProcessApi.getSessionCommandLogs']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get a list of all active pseudo-terminal sessions
         * @summary List all PTY sessions
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listPtySessions(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listPtySessions(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ProcessApi.listPtySessions']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Get a list of all active shell sessions
         * @summary List all sessions
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listSessions(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listSessions(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ProcessApi.listSessions']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Resize the terminal dimensions of a pseudo-terminal session
         * @summary Resize a PTY session
         * @param {string} sessionId PTY session ID
         * @param {PtyResizeRequest} request Resize request with new dimensions
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async resizePtySession(sessionId, request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.resizePtySession(sessionId, request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ProcessApi.resizePtySession']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Send input data to a running command in a session for interactive execution
         * @summary Send input to command
         * @param {string} sessionId Session ID
         * @param {string} commandId Command ID
         * @param {SessionSendInputRequest} request Input send request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async sendInput(sessionId, commandId, request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.sendInput(sessionId, commandId, request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ProcessApi.sendInput']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Execute a command within an existing shell session
         * @summary Execute command in session
         * @param {string} sessionId Session ID
         * @param {SessionExecuteRequest} request Command execution request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async sessionExecuteCommand(sessionId, request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.sessionExecuteCommand(sessionId, request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ProcessApi.sessionExecuteCommand']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.ProcessApiFp = ProcessApiFp;
/**
 * ProcessApi - factory interface
 */
const ProcessApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.ProcessApiFp)(configuration);
    return {
        /**
         * Execute Python, JavaScript, or TypeScript code and return output, exit code, and artifacts
         * @summary Execute code
         * @param {CodeRunRequest} request Code execution request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        codeRun(request, options) {
            return localVarFp.codeRun(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Establish a WebSocket connection to interact with a pseudo-terminal session
         * @summary Connect to PTY session via WebSocket
         * @param {string} sessionId PTY session ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        connectPtySession(sessionId, options) {
            return localVarFp.connectPtySession(sessionId, options).then((request) => request(axios, basePath));
        },
        /**
         * Create a new pseudo-terminal session with specified configuration
         * @summary Create a new PTY session
         * @param {PtyCreateRequest} request PTY session creation request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createPtySession(request, options) {
            return localVarFp.createPtySession(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Create a new shell session for command execution
         * @summary Create a new session
         * @param {CreateSessionRequest} request Session creation request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createSession(request, options) {
            return localVarFp.createSession(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Delete a pseudo-terminal session and terminate its process
         * @summary Delete a PTY session
         * @param {string} sessionId PTY session ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deletePtySession(sessionId, options) {
            return localVarFp.deletePtySession(sessionId, options).then((request) => request(axios, basePath));
        },
        /**
         * Delete an existing shell session
         * @summary Delete a session
         * @param {string} sessionId Session ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteSession(sessionId, options) {
            return localVarFp.deleteSession(sessionId, options).then((request) => request(axios, basePath));
        },
        /**
         * Execute a shell command and return the output and exit code
         * @summary Execute a command
         * @param {ExecuteRequest} request Command execution request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        executeCommand(request, options) {
            return localVarFp.executeCommand(request, options).then((request) => request(axios, basePath));
        },
        /**
         * Get logs for a sandbox entrypoint session. Returns JSON with separated stdout/stderr for SDK >= 0.161.0, plain text otherwise. Supports WebSocket streaming.
         * @summary Get entrypoint logs
         * @param {boolean} [follow] Follow logs in real-time (WebSocket only)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getEntrypointLogs(follow, options) {
            return localVarFp.getEntrypointLogs(follow, options).then((request) => request(axios, basePath));
        },
        /**
         * Get details of an entrypoint session including its commands
         * @summary Get entrypoint session details
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getEntrypointSession(options) {
            return localVarFp.getEntrypointSession(options).then((request) => request(axios, basePath));
        },
        /**
         * Get detailed information about a specific pseudo-terminal session
         * @summary Get PTY session information
         * @param {string} sessionId PTY session ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPtySession(sessionId, options) {
            return localVarFp.getPtySession(sessionId, options).then((request) => request(axios, basePath));
        },
        /**
         * Get details of a specific session including its commands
         * @summary Get session details
         * @param {string} sessionId Session ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getSession(sessionId, options) {
            return localVarFp.getSession(sessionId, options).then((request) => request(axios, basePath));
        },
        /**
         * Get details of a specific command within a session
         * @summary Get session command details
         * @param {string} sessionId Session ID
         * @param {string} commandId Command ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getSessionCommand(sessionId, commandId, options) {
            return localVarFp.getSessionCommand(sessionId, commandId, options).then((request) => request(axios, basePath));
        },
        /**
         * Get logs for a specific command within a session. Returns JSON with separated stdout/stderr for SDK >= 0.167.0, plain text otherwise. Supports WebSocket streaming.
         * @summary Get session command logs
         * @param {string} sessionId Session ID
         * @param {string} commandId Command ID
         * @param {boolean} [follow] Follow logs in real-time (WebSocket only)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getSessionCommandLogs(sessionId, commandId, follow, options) {
            return localVarFp.getSessionCommandLogs(sessionId, commandId, follow, options).then((request) => request(axios, basePath));
        },
        /**
         * Get a list of all active pseudo-terminal sessions
         * @summary List all PTY sessions
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listPtySessions(options) {
            return localVarFp.listPtySessions(options).then((request) => request(axios, basePath));
        },
        /**
         * Get a list of all active shell sessions
         * @summary List all sessions
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listSessions(options) {
            return localVarFp.listSessions(options).then((request) => request(axios, basePath));
        },
        /**
         * Resize the terminal dimensions of a pseudo-terminal session
         * @summary Resize a PTY session
         * @param {string} sessionId PTY session ID
         * @param {PtyResizeRequest} request Resize request with new dimensions
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        resizePtySession(sessionId, request, options) {
            return localVarFp.resizePtySession(sessionId, request, options).then((request) => request(axios, basePath));
        },
        /**
         * Send input data to a running command in a session for interactive execution
         * @summary Send input to command
         * @param {string} sessionId Session ID
         * @param {string} commandId Command ID
         * @param {SessionSendInputRequest} request Input send request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        sendInput(sessionId, commandId, request, options) {
            return localVarFp.sendInput(sessionId, commandId, request, options).then((request) => request(axios, basePath));
        },
        /**
         * Execute a command within an existing shell session
         * @summary Execute command in session
         * @param {string} sessionId Session ID
         * @param {SessionExecuteRequest} request Command execution request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        sessionExecuteCommand(sessionId, request, options) {
            return localVarFp.sessionExecuteCommand(sessionId, request, options).then((request) => request(axios, basePath));
        },
    };
};
exports.ProcessApiFactory = ProcessApiFactory;
/**
 * ProcessApi - object-oriented interface
 */
class ProcessApi extends base_1.BaseAPI {
    /**
     * Execute Python, JavaScript, or TypeScript code and return output, exit code, and artifacts
     * @summary Execute code
     * @param {CodeRunRequest} request Code execution request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    codeRun(request, options) {
        return (0, exports.ProcessApiFp)(this.configuration).codeRun(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Establish a WebSocket connection to interact with a pseudo-terminal session
     * @summary Connect to PTY session via WebSocket
     * @param {string} sessionId PTY session ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    connectPtySession(sessionId, options) {
        return (0, exports.ProcessApiFp)(this.configuration).connectPtySession(sessionId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Create a new pseudo-terminal session with specified configuration
     * @summary Create a new PTY session
     * @param {PtyCreateRequest} request PTY session creation request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    createPtySession(request, options) {
        return (0, exports.ProcessApiFp)(this.configuration).createPtySession(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Create a new shell session for command execution
     * @summary Create a new session
     * @param {CreateSessionRequest} request Session creation request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    createSession(request, options) {
        return (0, exports.ProcessApiFp)(this.configuration).createSession(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Delete a pseudo-terminal session and terminate its process
     * @summary Delete a PTY session
     * @param {string} sessionId PTY session ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    deletePtySession(sessionId, options) {
        return (0, exports.ProcessApiFp)(this.configuration).deletePtySession(sessionId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Delete an existing shell session
     * @summary Delete a session
     * @param {string} sessionId Session ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    deleteSession(sessionId, options) {
        return (0, exports.ProcessApiFp)(this.configuration).deleteSession(sessionId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Execute a shell command and return the output and exit code
     * @summary Execute a command
     * @param {ExecuteRequest} request Command execution request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    executeCommand(request, options) {
        return (0, exports.ProcessApiFp)(this.configuration).executeCommand(request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get logs for a sandbox entrypoint session. Returns JSON with separated stdout/stderr for SDK >= 0.161.0, plain text otherwise. Supports WebSocket streaming.
     * @summary Get entrypoint logs
     * @param {boolean} [follow] Follow logs in real-time (WebSocket only)
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getEntrypointLogs(follow, options) {
        return (0, exports.ProcessApiFp)(this.configuration).getEntrypointLogs(follow, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get details of an entrypoint session including its commands
     * @summary Get entrypoint session details
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getEntrypointSession(options) {
        return (0, exports.ProcessApiFp)(this.configuration).getEntrypointSession(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get detailed information about a specific pseudo-terminal session
     * @summary Get PTY session information
     * @param {string} sessionId PTY session ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getPtySession(sessionId, options) {
        return (0, exports.ProcessApiFp)(this.configuration).getPtySession(sessionId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get details of a specific session including its commands
     * @summary Get session details
     * @param {string} sessionId Session ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getSession(sessionId, options) {
        return (0, exports.ProcessApiFp)(this.configuration).getSession(sessionId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get details of a specific command within a session
     * @summary Get session command details
     * @param {string} sessionId Session ID
     * @param {string} commandId Command ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getSessionCommand(sessionId, commandId, options) {
        return (0, exports.ProcessApiFp)(this.configuration).getSessionCommand(sessionId, commandId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get logs for a specific command within a session. Returns JSON with separated stdout/stderr for SDK >= 0.167.0, plain text otherwise. Supports WebSocket streaming.
     * @summary Get session command logs
     * @param {string} sessionId Session ID
     * @param {string} commandId Command ID
     * @param {boolean} [follow] Follow logs in real-time (WebSocket only)
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getSessionCommandLogs(sessionId, commandId, follow, options) {
        return (0, exports.ProcessApiFp)(this.configuration).getSessionCommandLogs(sessionId, commandId, follow, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get a list of all active pseudo-terminal sessions
     * @summary List all PTY sessions
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    listPtySessions(options) {
        return (0, exports.ProcessApiFp)(this.configuration).listPtySessions(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get a list of all active shell sessions
     * @summary List all sessions
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    listSessions(options) {
        return (0, exports.ProcessApiFp)(this.configuration).listSessions(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Resize the terminal dimensions of a pseudo-terminal session
     * @summary Resize a PTY session
     * @param {string} sessionId PTY session ID
     * @param {PtyResizeRequest} request Resize request with new dimensions
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    resizePtySession(sessionId, request, options) {
        return (0, exports.ProcessApiFp)(this.configuration).resizePtySession(sessionId, request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Send input data to a running command in a session for interactive execution
     * @summary Send input to command
     * @param {string} sessionId Session ID
     * @param {string} commandId Command ID
     * @param {SessionSendInputRequest} request Input send request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    sendInput(sessionId, commandId, request, options) {
        return (0, exports.ProcessApiFp)(this.configuration).sendInput(sessionId, commandId, request, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Execute a command within an existing shell session
     * @summary Execute command in session
     * @param {string} sessionId Session ID
     * @param {SessionExecuteRequest} request Command execution request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    sessionExecuteCommand(sessionId, request, options) {
        return (0, exports.ProcessApiFp)(this.configuration).sessionExecuteCommand(sessionId, request, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.ProcessApi = ProcessApi;
