// @daytona/toolbox-api-client/src/api/server-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Daytona Toolbox API
 * Daytona Toolbox API. The base URL comes from the sandbox\'s `toolboxProxyUrl` field (returned in sandbox DTO by the main Daytona API) plus the sandbox ID: `{toolboxProxyUrl}/{sandboxId}/{endpoint}`. Default for Daytona Cloud: `https://proxy.app.daytona.io/toolbox/{sandboxId}`.
 *
 * The version of the OpenAPI document: v0.0.0-dev
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerApi = exports.ServerApiFactory = exports.ServerApiFp = exports.ServerApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * ServerApi - axios parameter creator
 */
const ServerApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Set the auth token and initialize telemetry for the toolbox server
         * @summary Initialize toolbox server
         * @param {InitializeRequest} request Initialization request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        initialize: async (request, options = {}) => {
            // verify required parameter 'request' is not null or undefined
            (0, common_1.assertParamExists)('initialize', 'request', request);
            const localVarPath = `/init`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(request, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.ServerApiAxiosParamCreator = ServerApiAxiosParamCreator;
/**
 * ServerApi - functional programming interface
 */
const ServerApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.ServerApiAxiosParamCreator)(configuration);
    return {
        /**
         * Set the auth token and initialize telemetry for the toolbox server
         * @summary Initialize toolbox server
         * @param {InitializeRequest} request Initialization request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async initialize(request, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.initialize(request, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['ServerApi.initialize']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.ServerApiFp = ServerApiFp;
/**
 * ServerApi - factory interface
 */
const ServerApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.ServerApiFp)(configuration);
    return {
        /**
         * Set the auth token and initialize telemetry for the toolbox server
         * @summary Initialize toolbox server
         * @param {InitializeRequest} request Initialization request
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        initialize(request, options) {
            return localVarFp.initialize(request, options).then((request) => request(axios, basePath));
        },
    };
};
exports.ServerApiFactory = ServerApiFactory;
/**
 * ServerApi - object-oriented interface
 */
class ServerApi extends base_1.BaseAPI {
    /**
     * Set the auth token and initialize telemetry for the toolbox server
     * @summary Initialize toolbox server
     * @param {InitializeRequest} request Initialization request
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    initialize(request, options) {
        return (0, exports.ServerApiFp)(this.configuration).initialize(request, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.ServerApi = ServerApi;
