// @daytona/sdk/cjs/utils/Import.js
"use strict";
/*
 * Copyright Daytona Platforms Inc.
 * SPDX-License-Identifier: Apache-2.0
 *
 * post-build.js rewrites every `require(` in the ESM output of this file to
 * `__esmRequire(` — avoid that literal token in string literals here.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.dynamicImport = dynamicImport;
exports.dynamicRequire = dynamicRequire;
exports.getPackageInfo = getPackageInfo;
const DaytonaError_1 = require("../errors/DaytonaError");
const Runtime_1 = require("./Runtime");
const loaderMap = {
    'fast-glob': async () => require('fast-glob'),
    '@iarna/toml': async () => require('@iarna/toml'),
    stream: async () => require('stream'),
    tar: async () => require('tar'),
    'expand-tilde': async () => require('expand-tilde'),
    // Stored in a variable + /* @vite-ignore */ so Vite/Rollup don't statically
    // chunk this; otherwise apps bundling the SDK source (e.g. the dashboard)
    // would pull @aws-sdk transitively. Node consumers resolve at runtime.
    ObjectStorage: () => {
        const path = '../ObjectStorage.js';
        return Promise.resolve(`${path}`).then(s => __importStar(require(s)));
    },
    fs: async () => require('fs'),
    'form-data': async () => require('form-data'),
    util: async () => require('util'),
};
const requireMap = {
    'fast-glob': () => require('fast-glob'),
    '@iarna/toml': () => require('@iarna/toml'),
    stream: () => require('stream'),
    tar: () => require('tar'),
    'expand-tilde': () => require('expand-tilde'),
    fs: () => require('fs'),
    'form-data': () => require('form-data'),
    buffer: () => require('buffer'),
    busboy: () => require('busboy'),
    '@opentelemetry/api': () => require('@opentelemetry/api'),
    '@opentelemetry/sdk-node': () => require('@opentelemetry/sdk-node'),
    '@opentelemetry/instrumentation-http': () => require('@opentelemetry/instrumentation-http'),
    '@opentelemetry/sdk-trace-base': () => require('@opentelemetry/sdk-trace-base'),
    '@opentelemetry/exporter-trace-otlp-http': () => require('@opentelemetry/exporter-trace-otlp-http'),
    '@opentelemetry/otlp-exporter-base': () => require('@opentelemetry/otlp-exporter-base'),
    '@opentelemetry/semantic-conventions': () => require('@opentelemetry/semantic-conventions'),
    '@opentelemetry/resources': () => require('@opentelemetry/resources'),
};
const validateMap = {
    'fast-glob': (mod) => typeof mod === 'function' && typeof mod?.sync === 'function',
    '@iarna/toml': (mod) => typeof mod.parse === 'function' && typeof mod.stringify === 'function',
    stream: (mod) => typeof mod.Readable === 'function' && typeof mod.Writable === 'function',
    tar: (mod) => typeof mod.extract === 'function' && typeof mod.create === 'function',
    'expand-tilde': (mod) => typeof mod === 'function',
    fs: (mod) => typeof mod.createReadStream === 'function' && typeof mod.readFile === 'function',
    'form-data': (mod) => typeof mod === 'function',
    util: (mod) => typeof mod.promisify === 'function',
};
async function dynamicImport(name, errorPrefix) {
    const loader = loaderMap[name];
    if (!loader) {
        throw new DaytonaError_1.DaytonaError(`${errorPrefix || ''} Unknown module "${name}"`);
    }
    let mod;
    try {
        mod = (await loader());
        mod = unwrapInterop(mod);
    }
    catch (err) {
        const msg = err instanceof Error ? err.message : String(err);
        throw new DaytonaError_1.DaytonaError(`${errorPrefix || ''} Module "${name}" is not available in the "${Runtime_1.RUNTIME}" runtime: ${msg}`);
    }
    if (validateMap[name] && !validateMap[name](mod)) {
        throw new DaytonaError_1.DaytonaError(`${errorPrefix || ''} Module "${name}" didn't pass import validation in the "${Runtime_1.RUNTIME}" runtime`);
    }
    return mod;
}
function unwrapInterop(mod) {
    if (!mod || typeof mod !== 'object' || mod.default === undefined)
        return mod;
    const namedKeys = Object.keys(mod).filter((k) => k !== 'default');
    if (namedKeys.length === 0)
        return mod.default;
    return mod;
}
function dynamicRequire(name, errorPrefix) {
    const loader = requireMap[name];
    if (!loader) {
        throw new DaytonaError_1.DaytonaError(`${errorPrefix || ''} Unknown module "${name}"`);
    }
    let mod;
    try {
        mod = loader();
        mod = unwrapInterop(mod);
    }
    catch (err) {
        const msg = err instanceof Error ? err.message : String(err);
        throw new DaytonaError_1.DaytonaError(`${errorPrefix || ''} Module "${name}" is not available in the "${Runtime_1.RUNTIME}" runtime: ${msg}`);
    }
    if (validateMap[name] && !validateMap[name](mod)) {
        throw new DaytonaError_1.DaytonaError(`${errorPrefix || ''} Module "${name}" didn't pass import validation in the "${Runtime_1.RUNTIME}" runtime`);
    }
    return mod;
}
let _packageInfo = null;
function getPackageInfo() {
    if (_packageInfo)
        return _packageInfo;
    try {
        const pkg = require('../../package.json');
        _packageInfo = { name: pkg.name, version: pkg.version };
    }
    catch {
        _packageInfo = { name: '@daytona/sdk', version: '0.0.0' };
    }
    return _packageInfo;
}
