// @daytona/sdk/esm/utils/Import.js
import * as _m from 'module';
const __esmRequire = (() => {
  try { return _m.createRequire(import.meta.url); } catch {}
  try { if (typeof require !== 'undefined') return require; } catch {}
  return (id) => { throw new Error(
    'cannot require("' + id + '"): no CommonJS require available. ' +
    'If re-bundling @daytona/sdk to CJS, ensure createRequire or the host require is accessible.'
  ); };
})();
/*
 * Copyright Daytona Platforms Inc.
 * SPDX-License-Identifier: Apache-2.0
 *
 * post-build.js rewrites every `__esmRequire(` in the ESM output of this file to
 * `__esmRequire(` — avoid that literal token in string literals here.
 */
import { DaytonaError } from '../errors/DaytonaError.js';
import { RUNTIME } from './Runtime.js';
const loaderMap = {
    'fast-glob': async () => __esmRequire('fast-glob'),
    '@iarna/toml': async () => __esmRequire('@iarna/toml'),
    stream: async () => __esmRequire('stream'),
    tar: async () => __esmRequire('tar'),
    'expand-tilde': async () => __esmRequire('expand-tilde'),
    // Stored in a variable + /* @vite-ignore */ so Vite/Rollup don't statically
    // chunk this; otherwise apps bundling the SDK source (e.g. the dashboard)
    // would pull @aws-sdk transitively. Node consumers resolve at runtime.
    ObjectStorage: () => {
        const path = '../ObjectStorage.js';
        return import(/* @vite-ignore */ path);
    },
    fs: async () => __esmRequire('fs'),
    'form-data': async () => __esmRequire('form-data'),
    util: async () => __esmRequire('util'),
};
const requireMap = {
    'fast-glob': () => __esmRequire('fast-glob'),
    '@iarna/toml': () => __esmRequire('@iarna/toml'),
    stream: () => __esmRequire('stream'),
    tar: () => __esmRequire('tar'),
    'expand-tilde': () => __esmRequire('expand-tilde'),
    fs: () => __esmRequire('fs'),
    'form-data': () => __esmRequire('form-data'),
    buffer: () => __esmRequire('buffer'),
    busboy: () => __esmRequire('busboy'),
    '@opentelemetry/api': () => __esmRequire('@opentelemetry/api'),
    '@opentelemetry/sdk-node': () => __esmRequire('@opentelemetry/sdk-node'),
    '@opentelemetry/instrumentation-http': () => __esmRequire('@opentelemetry/instrumentation-http'),
    '@opentelemetry/sdk-trace-base': () => __esmRequire('@opentelemetry/sdk-trace-base'),
    '@opentelemetry/exporter-trace-otlp-http': () => __esmRequire('@opentelemetry/exporter-trace-otlp-http'),
    '@opentelemetry/otlp-exporter-base': () => __esmRequire('@opentelemetry/otlp-exporter-base'),
    '@opentelemetry/semantic-conventions': () => __esmRequire('@opentelemetry/semantic-conventions'),
    '@opentelemetry/resources': () => __esmRequire('@opentelemetry/resources'),
};
const validateMap = {
    'fast-glob': (mod) => typeof mod === 'function' && typeof mod?.sync === 'function',
    '@iarna/toml': (mod) => typeof mod.parse === 'function' && typeof mod.stringify === 'function',
    stream: (mod) => typeof mod.Readable === 'function' && typeof mod.Writable === 'function',
    tar: (mod) => typeof mod.extract === 'function' && typeof mod.create === 'function',
    'expand-tilde': (mod) => typeof mod === 'function',
    fs: (mod) => typeof mod.createReadStream === 'function' && typeof mod.readFile === 'function',
    'form-data': (mod) => typeof mod === 'function',
    util: (mod) => typeof mod.promisify === 'function',
};
export async function dynamicImport(name, errorPrefix) {
    const loader = loaderMap[name];
    if (!loader) {
        throw new DaytonaError(`${errorPrefix || ''} Unknown module "${name}"`);
    }
    let mod;
    try {
        mod = (await loader());
        mod = unwrapInterop(mod);
    }
    catch (err) {
        const msg = err instanceof Error ? err.message : String(err);
        throw new DaytonaError(`${errorPrefix || ''} Module "${name}" is not available in the "${RUNTIME}" runtime: ${msg}`);
    }
    if (validateMap[name] && !validateMap[name](mod)) {
        throw new DaytonaError(`${errorPrefix || ''} Module "${name}" didn't pass import validation in the "${RUNTIME}" runtime`);
    }
    return mod;
}
function unwrapInterop(mod) {
    if (!mod || typeof mod !== 'object' || mod.default === undefined)
        return mod;
    const namedKeys = Object.keys(mod).filter((k) => k !== 'default');
    if (namedKeys.length === 0)
        return mod.default;
    return mod;
}
export function dynamicRequire(name, errorPrefix) {
    const loader = requireMap[name];
    if (!loader) {
        throw new DaytonaError(`${errorPrefix || ''} Unknown module "${name}"`);
    }
    let mod;
    try {
        mod = loader();
        mod = unwrapInterop(mod);
    }
    catch (err) {
        const msg = err instanceof Error ? err.message : String(err);
        throw new DaytonaError(`${errorPrefix || ''} Module "${name}" is not available in the "${RUNTIME}" runtime: ${msg}`);
    }
    if (validateMap[name] && !validateMap[name](mod)) {
        throw new DaytonaError(`${errorPrefix || ''} Module "${name}" didn't pass import validation in the "${RUNTIME}" runtime`);
    }
    return mod;
}
let _packageInfo = null;
export function getPackageInfo() {
    if (_packageInfo)
        return _packageInfo;
    try {
        const pkg = {"name":"@daytona/sdk","version":"0.193.0"};
        _packageInfo = { name: pkg.name, version: pkg.version };
    }
    catch {
        _packageInfo = { name: '@daytona/sdk', version: '0.0.0' };
    }
    return _packageInfo;
}
