// @daytona/sdk/esm/utils/Runtime.js
/*
 * Copyright Daytona Platforms Inc.
 * SPDX-License-Identifier: Apache-2.0
 */
export var Runtime;
(function (Runtime) {
    Runtime["NODE"] = "node";
    Runtime["DENO"] = "deno";
    Runtime["BUN"] = "bun";
    Runtime["BROWSER"] = "browser";
    Runtime["SERVERLESS"] = "serverless";
    Runtime["UNKNOWN"] = "unknown";
})(Runtime || (Runtime = {}));
export const RUNTIME = typeof Deno !== 'undefined'
    ? Runtime.DENO
    : typeof Bun !== 'undefined' && !!Bun.version
        ? Runtime.BUN
        : isServerlessRuntime()
            ? Runtime.SERVERLESS
            : typeof window !== 'undefined'
                ? Runtime.BROWSER
                : typeof process !== 'undefined' && !!process.versions?.node
                    ? Runtime.NODE
                    : Runtime.UNKNOWN;
export function getEnvVar(name) {
    if (typeof process !== 'undefined' && process.env) {
        return process.env[name];
    }
    if (RUNTIME === Runtime.DENO) {
        return Deno.env.get(name);
    }
    return undefined;
}
export class DaytonaEnvReader {
    envLocalVars;
    envVars;
    constructor() {
        this.envLocalVars = DaytonaEnvReader.parseFileVars('.env.local');
        this.envVars = DaytonaEnvReader.parseFileVars('.env');
    }
    get(name) {
        if (!name.startsWith('DAYTONA_')) {
            throw new Error(`DaytonaEnvReader: variable name must start with 'DAYTONA_', got '${name}'`);
        }
        // 1. Runtime env
        const runtimeVal = getEnvVar(name);
        if (runtimeVal !== undefined)
            return runtimeVal;
        // 2. .env.local
        if (name in this.envLocalVars)
            return this.envLocalVars[name];
        // 3. .env
        return this.envVars[name];
    }
    static parseFileVars(path) {
        if (RUNTIME !== Runtime.NODE || typeof require === 'undefined')
            return {};
        const fs = require('fs');
        if (!fs.existsSync(path))
            return {};
        const dotenv = require('dotenv');
        const parsed = dotenv.parse(fs.readFileSync(path));
        return Object.fromEntries(Object.entries(parsed).filter(([k]) => k.startsWith('DAYTONA_')));
    }
}
export function isServerlessRuntime() {
    // Safely grab env vars, even if `process` is undeclared
    const env = typeof process !== 'undefined' ? process.env : {};
    // Worker-specific globals
    const globalObj = globalThis;
    return Boolean(
    // Cloudflare Workers (V8 isolate API)
    typeof globalObj.WebSocketPair === 'function' ||
        // Cloudflare Pages
        env.CF_PAGES === '1' ||
        // AWS Lambda (incl. SAM local)
        env.AWS_EXECUTION_ENV?.startsWith('AWS_Lambda') ||
        env.LAMBDA_TASK_ROOT !== undefined ||
        env.AWS_SAM_LOCAL === 'true' ||
        // Azure Functions
        env.FUNCTIONS_WORKER_RUNTIME !== undefined ||
        // Google Cloud Functions / Cloud Run
        (env.FUNCTION_TARGET !== undefined && env.FUNCTION_SIGNATURE_TYPE !== undefined) ||
        // Vercel
        env.VERCEL === '1' ||
        // Netlify Functions
        env.SITE_NAME !== undefined);
}
