// @daytona/api-client/src/api/admin-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Daytona
 * Daytona AI platform API Docs
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@daytona.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdminApi = exports.AdminApiFactory = exports.AdminApiFp = exports.AdminApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * AdminApi - axios parameter creator
 */
const AdminApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @summary Check if an image can be cleaned up
         * @param {string} imageName Image name with tag to check
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminCanCleanupImage: async (imageName, options = {}) => {
            // verify required parameter 'imageName' is not null or undefined
            (0, common_1.assertParamExists)('adminCanCleanupImage', 'imageName', imageName);
            const localVarPath = `/admin/snapshots/can-cleanup-image`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            if (imageName !== undefined) {
                localVarQueryParameter['imageName'] = imageName;
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Create organization region quota
         * @param {string} organizationId Organization ID
         * @param {string} regionId ID of the region the new quota applies to
         * @param {CreateOrganizationRegionQuota} createOrganizationRegionQuota
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminCreateOrganizationRegionQuota: async (organizationId, regionId, createOrganizationRegionQuota, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('adminCreateOrganizationRegionQuota', 'organizationId', organizationId);
            // verify required parameter 'regionId' is not null or undefined
            (0, common_1.assertParamExists)('adminCreateOrganizationRegionQuota', 'regionId', regionId);
            // verify required parameter 'createOrganizationRegionQuota' is not null or undefined
            (0, common_1.assertParamExists)('adminCreateOrganizationRegionQuota', 'createOrganizationRegionQuota', createOrganizationRegionQuota);
            const localVarPath = `/admin/organizations/{organizationId}/quota/{regionId}`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)))
                .replace(`{${"regionId"}}`, encodeURIComponent(String(regionId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(createOrganizationRegionQuota, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Create runner
         * @param {AdminCreateRunner} adminCreateRunner
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminCreateRunner: async (adminCreateRunner, options = {}) => {
            // verify required parameter 'adminCreateRunner' is not null or undefined
            (0, common_1.assertParamExists)('adminCreateRunner', 'adminCreateRunner', adminCreateRunner);
            const localVarPath = `/admin/runners`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(adminCreateRunner, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Create user
         * @param {CreateUser} createUser
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminCreateUser: async (createUser, options = {}) => {
            // verify required parameter 'createUser' is not null or undefined
            (0, common_1.assertParamExists)('adminCreateUser', 'createUser', createUser);
            const localVarPath = `/admin/users`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(createUser, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Delete organization region quota
         * @param {string} organizationId Organization ID
         * @param {string} regionId Region ID
         * @param {SandboxClass} sandboxClass Sandbox class
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminDeleteOrganizationRegionQuota: async (organizationId, regionId, sandboxClass, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('adminDeleteOrganizationRegionQuota', 'organizationId', organizationId);
            // verify required parameter 'regionId' is not null or undefined
            (0, common_1.assertParamExists)('adminDeleteOrganizationRegionQuota', 'regionId', regionId);
            // verify required parameter 'sandboxClass' is not null or undefined
            (0, common_1.assertParamExists)('adminDeleteOrganizationRegionQuota', 'sandboxClass', sandboxClass);
            const localVarPath = `/admin/organizations/{organizationId}/quota/{regionId}/{sandboxClass}`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)))
                .replace(`{${"regionId"}}`, encodeURIComponent(String(regionId)))
                .replace(`{${"sandboxClass"}}`, encodeURIComponent(String(sandboxClass)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'DELETE', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Delete runner
         * @param {string} id Runner ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminDeleteRunner: async (id, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('adminDeleteRunner', 'id', id);
            const localVarPath = `/admin/runners/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'DELETE', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Get all audit logs
         * @param {number} [page] Page number of the results
         * @param {number} [limit] Number of results per page
         * @param {Date} [from] Deprecated alias for &#x60;createdAt[gte]&#x60;. From date (ISO 8601 format).
         * @param {Date} [to] Deprecated alias for &#x60;createdAt[lte]&#x60;. To date (ISO 8601 format).
         * @param {string} [nextToken] Token for cursor-based pagination. When provided, takes precedence over page parameter.
         * @param {StringFilter} [id] Filter by audit log ID.
         * @param {StringFilter} [actorId] Filter by actor user ID.
         * @param {StringFilter} [actorEmail] Filter by actor email.
         * @param {StringFilter} [actorApiKeyPrefix] Filter by actor API key prefix.
         * @param {StringFilter} [actorApiKeySuffix] Filter by actor API key suffix.
         * @param {StringFilter} [action] Filter by action.
         * @param {StringFilter} [targetType] Filter by target type.
         * @param {StringFilter} [targetId] Filter by target ID.
         * @param {IntFilter} [statusCode] Filter by HTTP status code.
         * @param {DateFilter} [createdAt] Filter by creation timestamp.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminGetAllAuditLogs: async (page, limit, from, to, nextToken, id, actorId, actorEmail, actorApiKeyPrefix, actorApiKeySuffix, action, targetType, targetId, statusCode, createdAt, options = {}) => {
            const localVarPath = `/admin/audit`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            if (page !== undefined) {
                localVarQueryParameter['page'] = page;
            }
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (from !== undefined) {
                localVarQueryParameter['from'] = (from instanceof Date) ?
                    from.toISOString() :
                    from;
            }
            if (to !== undefined) {
                localVarQueryParameter['to'] = (to instanceof Date) ?
                    to.toISOString() :
                    to;
            }
            if (nextToken !== undefined) {
                localVarQueryParameter['nextToken'] = nextToken;
            }
            if (id !== undefined) {
                for (const [key, value] of Object.entries(id)) {
                    localVarQueryParameter[key] = value;
                }
            }
            if (actorId !== undefined) {
                for (const [key, value] of Object.entries(actorId)) {
                    localVarQueryParameter[key] = value;
                }
            }
            if (actorEmail !== undefined) {
                for (const [key, value] of Object.entries(actorEmail)) {
                    localVarQueryParameter[key] = value;
                }
            }
            if (actorApiKeyPrefix !== undefined) {
                for (const [key, value] of Object.entries(actorApiKeyPrefix)) {
                    localVarQueryParameter[key] = value;
                }
            }
            if (actorApiKeySuffix !== undefined) {
                for (const [key, value] of Object.entries(actorApiKeySuffix)) {
                    localVarQueryParameter[key] = value;
                }
            }
            if (action !== undefined) {
                for (const [key, value] of Object.entries(action)) {
                    localVarQueryParameter[key] = value;
                }
            }
            if (targetType !== undefined) {
                for (const [key, value] of Object.entries(targetType)) {
                    localVarQueryParameter[key] = value;
                }
            }
            if (targetId !== undefined) {
                for (const [key, value] of Object.entries(targetId)) {
                    localVarQueryParameter[key] = value;
                }
            }
            if (statusCode !== undefined) {
                for (const [key, value] of Object.entries(statusCode)) {
                    localVarQueryParameter[key] = value;
                }
            }
            if (createdAt !== undefined) {
                for (const [key, value] of Object.entries(createdAt)) {
                    localVarQueryParameter[key] = value;
                }
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Get delivery attempts for a webhook message
         * @param {string} organizationId
         * @param {string} messageId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminGetMessageAttempts: async (organizationId, messageId, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('adminGetMessageAttempts', 'organizationId', organizationId);
            // verify required parameter 'messageId' is not null or undefined
            (0, common_1.assertParamExists)('adminGetMessageAttempts', 'messageId', messageId);
            const localVarPath = `/admin/webhooks/organizations/{organizationId}/messages/{messageId}/attempts`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)))
                .replace(`{${"messageId"}}`, encodeURIComponent(String(messageId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Get organization region quota
         * @param {string} organizationId Organization ID
         * @param {string} regionId Region ID
         * @param {SandboxClass} sandboxClass Sandbox class
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminGetOrganizationRegionQuota: async (organizationId, regionId, sandboxClass, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('adminGetOrganizationRegionQuota', 'organizationId', organizationId);
            // verify required parameter 'regionId' is not null or undefined
            (0, common_1.assertParamExists)('adminGetOrganizationRegionQuota', 'regionId', regionId);
            // verify required parameter 'sandboxClass' is not null or undefined
            (0, common_1.assertParamExists)('adminGetOrganizationRegionQuota', 'sandboxClass', sandboxClass);
            const localVarPath = `/admin/organizations/{organizationId}/quota/{regionId}/{sandboxClass}`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)))
                .replace(`{${"regionId"}}`, encodeURIComponent(String(regionId)))
                .replace(`{${"sandboxClass"}}`, encodeURIComponent(String(sandboxClass)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Get runner by ID
         * @param {string} id Runner ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminGetRunnerById: async (id, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('adminGetRunnerById', 'id', id);
            const localVarPath = `/admin/runners/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Get user by ID
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminGetUser: async (id, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('adminGetUser', 'id', id);
            const localVarPath = `/admin/users/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Get webhook service status
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminGetWebhookStatus: async (options = {}) => {
            const localVarPath = `/admin/webhooks/status`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Initialize webhooks for an organization
         * @param {string} organizationId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminInitializeWebhooks: async (organizationId, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('adminInitializeWebhooks', 'organizationId', organizationId);
            const localVarPath = `/admin/webhooks/organizations/{organizationId}/initialize`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary List all runners
         * @param {string} [regionId] Filter runners by region ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminListRunners: async (regionId, options = {}) => {
            const localVarPath = `/admin/runners`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            if (regionId !== undefined) {
                localVarQueryParameter['regionId'] = regionId;
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary List all users
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminListUsers: async (options = {}) => {
            const localVarPath = `/admin/users`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Recover sandbox from error state as an admin
         * @param {string} sandboxId ID of the sandbox
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminRecoverSandbox: async (sandboxId, options = {}) => {
            // verify required parameter 'sandboxId' is not null or undefined
            (0, common_1.assertParamExists)('adminRecoverSandbox', 'sandboxId', sandboxId);
            const localVarPath = `/admin/sandbox/{sandboxId}/recover`
                .replace(`{${"sandboxId"}}`, encodeURIComponent(String(sandboxId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Regenerate user key pair
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminRegenerateKeyPair: async (id, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('adminRegenerateKeyPair', 'id', id);
            const localVarPath = `/admin/users/{id}/regenerate-key-pair`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Send a webhook message to an organization
         * @param {string} organizationId
         * @param {SendWebhookDto} sendWebhookDto
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminSendWebhook: async (organizationId, sendWebhookDto, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('adminSendWebhook', 'organizationId', organizationId);
            // verify required parameter 'sendWebhookDto' is not null or undefined
            (0, common_1.assertParamExists)('adminSendWebhook', 'sendWebhookDto', sendWebhookDto);
            const localVarPath = `/admin/webhooks/organizations/{organizationId}/send`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(sendWebhookDto, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Set default registry
         * @param {string} id ID of the docker registry
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminSetDefaultRegistry: async (id, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('adminSetDefaultRegistry', 'id', id);
            const localVarPath = `/admin/docker-registry/{id}/set-default`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Set snapshot general status
         * @param {string} id Snapshot ID
         * @param {SetSnapshotGeneralStatusDto} setSnapshotGeneralStatusDto
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminSetSnapshotGeneralStatus: async (id, setSnapshotGeneralStatusDto, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('adminSetSnapshotGeneralStatus', 'id', id);
            // verify required parameter 'setSnapshotGeneralStatusDto' is not null or undefined
            (0, common_1.assertParamExists)('adminSetSnapshotGeneralStatus', 'setSnapshotGeneralStatusDto', setSnapshotGeneralStatusDto);
            const localVarPath = `/admin/snapshots/{id}/general`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'PATCH', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(setSnapshotGeneralStatusDto, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Update organization region quota
         * @param {string} organizationId Organization ID
         * @param {string} regionId Region ID
         * @param {UpdateOrganizationRegionQuota} updateOrganizationRegionQuota
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminUpdateOrganizationRegionQuota: async (organizationId, regionId, updateOrganizationRegionQuota, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('adminUpdateOrganizationRegionQuota', 'organizationId', organizationId);
            // verify required parameter 'regionId' is not null or undefined
            (0, common_1.assertParamExists)('adminUpdateOrganizationRegionQuota', 'regionId', regionId);
            // verify required parameter 'updateOrganizationRegionQuota' is not null or undefined
            (0, common_1.assertParamExists)('adminUpdateOrganizationRegionQuota', 'updateOrganizationRegionQuota', updateOrganizationRegionQuota);
            const localVarPath = `/admin/organizations/{organizationId}/quota/{regionId}`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)))
                .replace(`{${"regionId"}}`, encodeURIComponent(String(regionId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'PATCH', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(updateOrganizationRegionQuota, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Update runner scheduling status
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminUpdateRunnerScheduling: async (id, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('adminUpdateRunnerScheduling', 'id', id);
            const localVarPath = `/admin/runners/{id}/scheduling`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'PATCH', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.AdminApiAxiosParamCreator = AdminApiAxiosParamCreator;
/**
 * AdminApi - functional programming interface
 */
const AdminApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.AdminApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @summary Check if an image can be cleaned up
         * @param {string} imageName Image name with tag to check
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminCanCleanupImage(imageName, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminCanCleanupImage(imageName, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminCanCleanupImage']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Create organization region quota
         * @param {string} organizationId Organization ID
         * @param {string} regionId ID of the region the new quota applies to
         * @param {CreateOrganizationRegionQuota} createOrganizationRegionQuota
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminCreateOrganizationRegionQuota(organizationId, regionId, createOrganizationRegionQuota, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminCreateOrganizationRegionQuota(organizationId, regionId, createOrganizationRegionQuota, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminCreateOrganizationRegionQuota']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Create runner
         * @param {AdminCreateRunner} adminCreateRunner
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminCreateRunner(adminCreateRunner, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminCreateRunner(adminCreateRunner, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminCreateRunner']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Create user
         * @param {CreateUser} createUser
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminCreateUser(createUser, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminCreateUser(createUser, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminCreateUser']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Delete organization region quota
         * @param {string} organizationId Organization ID
         * @param {string} regionId Region ID
         * @param {SandboxClass} sandboxClass Sandbox class
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminDeleteOrganizationRegionQuota(organizationId, regionId, sandboxClass, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminDeleteOrganizationRegionQuota(organizationId, regionId, sandboxClass, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminDeleteOrganizationRegionQuota']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Delete runner
         * @param {string} id Runner ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminDeleteRunner(id, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminDeleteRunner(id, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminDeleteRunner']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Get all audit logs
         * @param {number} [page] Page number of the results
         * @param {number} [limit] Number of results per page
         * @param {Date} [from] Deprecated alias for &#x60;createdAt[gte]&#x60;. From date (ISO 8601 format).
         * @param {Date} [to] Deprecated alias for &#x60;createdAt[lte]&#x60;. To date (ISO 8601 format).
         * @param {string} [nextToken] Token for cursor-based pagination. When provided, takes precedence over page parameter.
         * @param {StringFilter} [id] Filter by audit log ID.
         * @param {StringFilter} [actorId] Filter by actor user ID.
         * @param {StringFilter} [actorEmail] Filter by actor email.
         * @param {StringFilter} [actorApiKeyPrefix] Filter by actor API key prefix.
         * @param {StringFilter} [actorApiKeySuffix] Filter by actor API key suffix.
         * @param {StringFilter} [action] Filter by action.
         * @param {StringFilter} [targetType] Filter by target type.
         * @param {StringFilter} [targetId] Filter by target ID.
         * @param {IntFilter} [statusCode] Filter by HTTP status code.
         * @param {DateFilter} [createdAt] Filter by creation timestamp.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminGetAllAuditLogs(page, limit, from, to, nextToken, id, actorId, actorEmail, actorApiKeyPrefix, actorApiKeySuffix, action, targetType, targetId, statusCode, createdAt, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminGetAllAuditLogs(page, limit, from, to, nextToken, id, actorId, actorEmail, actorApiKeyPrefix, actorApiKeySuffix, action, targetType, targetId, statusCode, createdAt, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminGetAllAuditLogs']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Get delivery attempts for a webhook message
         * @param {string} organizationId
         * @param {string} messageId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminGetMessageAttempts(organizationId, messageId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminGetMessageAttempts(organizationId, messageId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminGetMessageAttempts']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Get organization region quota
         * @param {string} organizationId Organization ID
         * @param {string} regionId Region ID
         * @param {SandboxClass} sandboxClass Sandbox class
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminGetOrganizationRegionQuota(organizationId, regionId, sandboxClass, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminGetOrganizationRegionQuota(organizationId, regionId, sandboxClass, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminGetOrganizationRegionQuota']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Get runner by ID
         * @param {string} id Runner ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminGetRunnerById(id, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminGetRunnerById(id, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminGetRunnerById']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Get user by ID
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminGetUser(id, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminGetUser(id, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminGetUser']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Get webhook service status
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminGetWebhookStatus(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminGetWebhookStatus(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminGetWebhookStatus']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Initialize webhooks for an organization
         * @param {string} organizationId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminInitializeWebhooks(organizationId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminInitializeWebhooks(organizationId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminInitializeWebhooks']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary List all runners
         * @param {string} [regionId] Filter runners by region ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminListRunners(regionId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminListRunners(regionId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminListRunners']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary List all users
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminListUsers(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminListUsers(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminListUsers']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Recover sandbox from error state as an admin
         * @param {string} sandboxId ID of the sandbox
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminRecoverSandbox(sandboxId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminRecoverSandbox(sandboxId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminRecoverSandbox']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Regenerate user key pair
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminRegenerateKeyPair(id, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminRegenerateKeyPair(id, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminRegenerateKeyPair']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Send a webhook message to an organization
         * @param {string} organizationId
         * @param {SendWebhookDto} sendWebhookDto
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminSendWebhook(organizationId, sendWebhookDto, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminSendWebhook(organizationId, sendWebhookDto, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminSendWebhook']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Set default registry
         * @param {string} id ID of the docker registry
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminSetDefaultRegistry(id, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminSetDefaultRegistry(id, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminSetDefaultRegistry']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Set snapshot general status
         * @param {string} id Snapshot ID
         * @param {SetSnapshotGeneralStatusDto} setSnapshotGeneralStatusDto
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminSetSnapshotGeneralStatus(id, setSnapshotGeneralStatusDto, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminSetSnapshotGeneralStatus(id, setSnapshotGeneralStatusDto, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminSetSnapshotGeneralStatus']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Update organization region quota
         * @param {string} organizationId Organization ID
         * @param {string} regionId Region ID
         * @param {UpdateOrganizationRegionQuota} updateOrganizationRegionQuota
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminUpdateOrganizationRegionQuota(organizationId, regionId, updateOrganizationRegionQuota, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminUpdateOrganizationRegionQuota(organizationId, regionId, updateOrganizationRegionQuota, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminUpdateOrganizationRegionQuota']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Update runner scheduling status
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async adminUpdateRunnerScheduling(id, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.adminUpdateRunnerScheduling(id, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AdminApi.adminUpdateRunnerScheduling']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.AdminApiFp = AdminApiFp;
/**
 * AdminApi - factory interface
 */
const AdminApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.AdminApiFp)(configuration);
    return {
        /**
         *
         * @summary Check if an image can be cleaned up
         * @param {string} imageName Image name with tag to check
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminCanCleanupImage(imageName, options) {
            return localVarFp.adminCanCleanupImage(imageName, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Create organization region quota
         * @param {string} organizationId Organization ID
         * @param {string} regionId ID of the region the new quota applies to
         * @param {CreateOrganizationRegionQuota} createOrganizationRegionQuota
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminCreateOrganizationRegionQuota(organizationId, regionId, createOrganizationRegionQuota, options) {
            return localVarFp.adminCreateOrganizationRegionQuota(organizationId, regionId, createOrganizationRegionQuota, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Create runner
         * @param {AdminCreateRunner} adminCreateRunner
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminCreateRunner(adminCreateRunner, options) {
            return localVarFp.adminCreateRunner(adminCreateRunner, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Create user
         * @param {CreateUser} createUser
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminCreateUser(createUser, options) {
            return localVarFp.adminCreateUser(createUser, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Delete organization region quota
         * @param {string} organizationId Organization ID
         * @param {string} regionId Region ID
         * @param {SandboxClass} sandboxClass Sandbox class
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminDeleteOrganizationRegionQuota(organizationId, regionId, sandboxClass, options) {
            return localVarFp.adminDeleteOrganizationRegionQuota(organizationId, regionId, sandboxClass, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Delete runner
         * @param {string} id Runner ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminDeleteRunner(id, options) {
            return localVarFp.adminDeleteRunner(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get all audit logs
         * @param {number} [page] Page number of the results
         * @param {number} [limit] Number of results per page
         * @param {Date} [from] Deprecated alias for &#x60;createdAt[gte]&#x60;. From date (ISO 8601 format).
         * @param {Date} [to] Deprecated alias for &#x60;createdAt[lte]&#x60;. To date (ISO 8601 format).
         * @param {string} [nextToken] Token for cursor-based pagination. When provided, takes precedence over page parameter.
         * @param {StringFilter} [id] Filter by audit log ID.
         * @param {StringFilter} [actorId] Filter by actor user ID.
         * @param {StringFilter} [actorEmail] Filter by actor email.
         * @param {StringFilter} [actorApiKeyPrefix] Filter by actor API key prefix.
         * @param {StringFilter} [actorApiKeySuffix] Filter by actor API key suffix.
         * @param {StringFilter} [action] Filter by action.
         * @param {StringFilter} [targetType] Filter by target type.
         * @param {StringFilter} [targetId] Filter by target ID.
         * @param {IntFilter} [statusCode] Filter by HTTP status code.
         * @param {DateFilter} [createdAt] Filter by creation timestamp.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminGetAllAuditLogs(page, limit, from, to, nextToken, id, actorId, actorEmail, actorApiKeyPrefix, actorApiKeySuffix, action, targetType, targetId, statusCode, createdAt, options) {
            return localVarFp.adminGetAllAuditLogs(page, limit, from, to, nextToken, id, actorId, actorEmail, actorApiKeyPrefix, actorApiKeySuffix, action, targetType, targetId, statusCode, createdAt, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get delivery attempts for a webhook message
         * @param {string} organizationId
         * @param {string} messageId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminGetMessageAttempts(organizationId, messageId, options) {
            return localVarFp.adminGetMessageAttempts(organizationId, messageId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get organization region quota
         * @param {string} organizationId Organization ID
         * @param {string} regionId Region ID
         * @param {SandboxClass} sandboxClass Sandbox class
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminGetOrganizationRegionQuota(organizationId, regionId, sandboxClass, options) {
            return localVarFp.adminGetOrganizationRegionQuota(organizationId, regionId, sandboxClass, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get runner by ID
         * @param {string} id Runner ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminGetRunnerById(id, options) {
            return localVarFp.adminGetRunnerById(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get user by ID
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminGetUser(id, options) {
            return localVarFp.adminGetUser(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get webhook service status
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminGetWebhookStatus(options) {
            return localVarFp.adminGetWebhookStatus(options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Initialize webhooks for an organization
         * @param {string} organizationId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminInitializeWebhooks(organizationId, options) {
            return localVarFp.adminInitializeWebhooks(organizationId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary List all runners
         * @param {string} [regionId] Filter runners by region ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminListRunners(regionId, options) {
            return localVarFp.adminListRunners(regionId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary List all users
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminListUsers(options) {
            return localVarFp.adminListUsers(options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Recover sandbox from error state as an admin
         * @param {string} sandboxId ID of the sandbox
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminRecoverSandbox(sandboxId, options) {
            return localVarFp.adminRecoverSandbox(sandboxId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Regenerate user key pair
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminRegenerateKeyPair(id, options) {
            return localVarFp.adminRegenerateKeyPair(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Send a webhook message to an organization
         * @param {string} organizationId
         * @param {SendWebhookDto} sendWebhookDto
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminSendWebhook(organizationId, sendWebhookDto, options) {
            return localVarFp.adminSendWebhook(organizationId, sendWebhookDto, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Set default registry
         * @param {string} id ID of the docker registry
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminSetDefaultRegistry(id, options) {
            return localVarFp.adminSetDefaultRegistry(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Set snapshot general status
         * @param {string} id Snapshot ID
         * @param {SetSnapshotGeneralStatusDto} setSnapshotGeneralStatusDto
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminSetSnapshotGeneralStatus(id, setSnapshotGeneralStatusDto, options) {
            return localVarFp.adminSetSnapshotGeneralStatus(id, setSnapshotGeneralStatusDto, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Update organization region quota
         * @param {string} organizationId Organization ID
         * @param {string} regionId Region ID
         * @param {UpdateOrganizationRegionQuota} updateOrganizationRegionQuota
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminUpdateOrganizationRegionQuota(organizationId, regionId, updateOrganizationRegionQuota, options) {
            return localVarFp.adminUpdateOrganizationRegionQuota(organizationId, regionId, updateOrganizationRegionQuota, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Update runner scheduling status
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        adminUpdateRunnerScheduling(id, options) {
            return localVarFp.adminUpdateRunnerScheduling(id, options).then((request) => request(axios, basePath));
        },
    };
};
exports.AdminApiFactory = AdminApiFactory;
/**
 * AdminApi - object-oriented interface
 */
class AdminApi extends base_1.BaseAPI {
    /**
     *
     * @summary Check if an image can be cleaned up
     * @param {string} imageName Image name with tag to check
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminCanCleanupImage(imageName, options) {
        return (0, exports.AdminApiFp)(this.configuration).adminCanCleanupImage(imageName, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Create organization region quota
     * @param {string} organizationId Organization ID
     * @param {string} regionId ID of the region the new quota applies to
     * @param {CreateOrganizationRegionQuota} createOrganizationRegionQuota
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminCreateOrganizationRegionQuota(organizationId, regionId, createOrganizationRegionQuota, options) {
        return (0, exports.AdminApiFp)(this.configuration).adminCreateOrganizationRegionQuota(organizationId, regionId, createOrganizationRegionQuota, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Create runner
     * @param {AdminCreateRunner} adminCreateRunner
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminCreateRunner(adminCreateRunner, options) {
        return (0, exports.AdminApiFp)(this.configuration).adminCreateRunner(adminCreateRunner, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Create user
     * @param {CreateUser} createUser
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminCreateUser(createUser, options) {
        return (0, exports.AdminApiFp)(this.configuration).adminCreateUser(createUser, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Delete organization region quota
     * @param {string} organizationId Organization ID
     * @param {string} regionId Region ID
     * @param {SandboxClass} sandboxClass Sandbox class
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminDeleteOrganizationRegionQuota(organizationId, regionId, sandboxClass, options) {
        return (0, exports.AdminApiFp)(this.configuration).adminDeleteOrganizationRegionQuota(organizationId, regionId, sandboxClass, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Delete runner
     * @param {string} id Runner ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminDeleteRunner(id, options) {
        return (0, exports.AdminApiFp)(this.configuration).adminDeleteRunner(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get all audit logs
     * @param {number} [page] Page number of the results
     * @param {number} [limit] Number of results per page
     * @param {Date} [from] Deprecated alias for &#x60;createdAt[gte]&#x60;. From date (ISO 8601 format).
     * @param {Date} [to] Deprecated alias for &#x60;createdAt[lte]&#x60;. To date (ISO 8601 format).
     * @param {string} [nextToken] Token for cursor-based pagination. When provided, takes precedence over page parameter.
     * @param {StringFilter} [id] Filter by audit log ID.
     * @param {StringFilter} [actorId] Filter by actor user ID.
     * @param {StringFilter} [actorEmail] Filter by actor email.
     * @param {StringFilter} [actorApiKeyPrefix] Filter by actor API key prefix.
     * @param {StringFilter} [actorApiKeySuffix] Filter by actor API key suffix.
     * @param {StringFilter} [action] Filter by action.
     * @param {StringFilter} [targetType] Filter by target type.
     * @param {StringFilter} [targetId] Filter by target ID.
     * @param {IntFilter} [statusCode] Filter by HTTP status code.
     * @param {DateFilter} [createdAt] Filter by creation timestamp.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminGetAllAuditLogs(page, limit, from, to, nextToken, id, actorId, actorEmail, actorApiKeyPrefix, actorApiKeySuffix, action, targetType, targetId, statusCode, createdAt, options) {
        return (0, exports.AdminApiFp)(this.configuration).adminGetAllAuditLogs(page, limit, from, to, nextToken, id, actorId, actorEmail, actorApiKeyPrefix, actorApiKeySuffix, action, targetType, targetId, statusCode, createdAt, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get delivery attempts for a webhook message
     * @param {string} organizationId
     * @param {string} messageId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminGetMessageAttempts(organizationId, messageId, options) {
        return (0, exports.AdminApiFp)(this.configuration).adminGetMessageAttempts(organizationId, messageId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get organization region quota
     * @param {string} organizationId Organization ID
     * @param {string} regionId Region ID
     * @param {SandboxClass} sandboxClass Sandbox class
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminGetOrganizationRegionQuota(organizationId, regionId, sandboxClass, options) {
        return (0, exports.AdminApiFp)(this.configuration).adminGetOrganizationRegionQuota(organizationId, regionId, sandboxClass, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get runner by ID
     * @param {string} id Runner ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminGetRunnerById(id, options) {
        return (0, exports.AdminApiFp)(this.configuration).adminGetRunnerById(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get user by ID
     * @param {string} id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminGetUser(id, options) {
        return (0, exports.AdminApiFp)(this.configuration).adminGetUser(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get webhook service status
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminGetWebhookStatus(options) {
        return (0, exports.AdminApiFp)(this.configuration).adminGetWebhookStatus(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Initialize webhooks for an organization
     * @param {string} organizationId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminInitializeWebhooks(organizationId, options) {
        return (0, exports.AdminApiFp)(this.configuration).adminInitializeWebhooks(organizationId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary List all runners
     * @param {string} [regionId] Filter runners by region ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminListRunners(regionId, options) {
        return (0, exports.AdminApiFp)(this.configuration).adminListRunners(regionId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary List all users
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminListUsers(options) {
        return (0, exports.AdminApiFp)(this.configuration).adminListUsers(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Recover sandbox from error state as an admin
     * @param {string} sandboxId ID of the sandbox
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminRecoverSandbox(sandboxId, options) {
        return (0, exports.AdminApiFp)(this.configuration).adminRecoverSandbox(sandboxId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Regenerate user key pair
     * @param {string} id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminRegenerateKeyPair(id, options) {
        return (0, exports.AdminApiFp)(this.configuration).adminRegenerateKeyPair(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Send a webhook message to an organization
     * @param {string} organizationId
     * @param {SendWebhookDto} sendWebhookDto
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminSendWebhook(organizationId, sendWebhookDto, options) {
        return (0, exports.AdminApiFp)(this.configuration).adminSendWebhook(organizationId, sendWebhookDto, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Set default registry
     * @param {string} id ID of the docker registry
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminSetDefaultRegistry(id, options) {
        return (0, exports.AdminApiFp)(this.configuration).adminSetDefaultRegistry(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Set snapshot general status
     * @param {string} id Snapshot ID
     * @param {SetSnapshotGeneralStatusDto} setSnapshotGeneralStatusDto
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminSetSnapshotGeneralStatus(id, setSnapshotGeneralStatusDto, options) {
        return (0, exports.AdminApiFp)(this.configuration).adminSetSnapshotGeneralStatus(id, setSnapshotGeneralStatusDto, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Update organization region quota
     * @param {string} organizationId Organization ID
     * @param {string} regionId Region ID
     * @param {UpdateOrganizationRegionQuota} updateOrganizationRegionQuota
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminUpdateOrganizationRegionQuota(organizationId, regionId, updateOrganizationRegionQuota, options) {
        return (0, exports.AdminApiFp)(this.configuration).adminUpdateOrganizationRegionQuota(organizationId, regionId, updateOrganizationRegionQuota, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Update runner scheduling status
     * @param {string} id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    adminUpdateRunnerScheduling(id, options) {
        return (0, exports.AdminApiFp)(this.configuration).adminUpdateRunnerScheduling(id, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.AdminApi = AdminApi;
