// @daytona/api-client/src/api/audit-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Daytona
 * Daytona AI platform API Docs
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@daytona.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuditApi = exports.AuditApiFactory = exports.AuditApiFp = exports.AuditApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * AuditApi - axios parameter creator
 */
const AuditApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @summary Get audit logs for organization
         * @param {string} organizationId Organization ID
         * @param {number} [page] Page number of the results
         * @param {number} [limit] Number of results per page
         * @param {Date} [from] Deprecated alias for &#x60;createdAt[gte]&#x60;. From date (ISO 8601 format).
         * @param {Date} [to] Deprecated alias for &#x60;createdAt[lte]&#x60;. To date (ISO 8601 format).
         * @param {string} [nextToken] Token for cursor-based pagination. When provided, takes precedence over page parameter.
         * @param {StringFilter} [id] Filter by audit log ID.
         * @param {StringFilter} [actorId] Filter by actor user ID.
         * @param {StringFilter} [actorEmail] Filter by actor email.
         * @param {StringFilter} [actorApiKeyPrefix] Filter by actor API key prefix.
         * @param {StringFilter} [actorApiKeySuffix] Filter by actor API key suffix.
         * @param {StringFilter} [action] Filter by action.
         * @param {StringFilter} [targetType] Filter by target type.
         * @param {StringFilter} [targetId] Filter by target ID.
         * @param {IntFilter} [statusCode] Filter by HTTP status code.
         * @param {DateFilter} [createdAt] Filter by creation timestamp.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getOrganizationAuditLogs: async (organizationId, page, limit, from, to, nextToken, id, actorId, actorEmail, actorApiKeyPrefix, actorApiKeySuffix, action, targetType, targetId, statusCode, createdAt, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('getOrganizationAuditLogs', 'organizationId', organizationId);
            const localVarPath = `/audit/organizations/{organizationId}`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            if (page !== undefined) {
                localVarQueryParameter['page'] = page;
            }
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (from !== undefined) {
                localVarQueryParameter['from'] = (from instanceof Date) ?
                    from.toISOString() :
                    from;
            }
            if (to !== undefined) {
                localVarQueryParameter['to'] = (to instanceof Date) ?
                    to.toISOString() :
                    to;
            }
            if (nextToken !== undefined) {
                localVarQueryParameter['nextToken'] = nextToken;
            }
            if (id !== undefined) {
                for (const [key, value] of Object.entries(id)) {
                    localVarQueryParameter[key] = value;
                }
            }
            if (actorId !== undefined) {
                for (const [key, value] of Object.entries(actorId)) {
                    localVarQueryParameter[key] = value;
                }
            }
            if (actorEmail !== undefined) {
                for (const [key, value] of Object.entries(actorEmail)) {
                    localVarQueryParameter[key] = value;
                }
            }
            if (actorApiKeyPrefix !== undefined) {
                for (const [key, value] of Object.entries(actorApiKeyPrefix)) {
                    localVarQueryParameter[key] = value;
                }
            }
            if (actorApiKeySuffix !== undefined) {
                for (const [key, value] of Object.entries(actorApiKeySuffix)) {
                    localVarQueryParameter[key] = value;
                }
            }
            if (action !== undefined) {
                for (const [key, value] of Object.entries(action)) {
                    localVarQueryParameter[key] = value;
                }
            }
            if (targetType !== undefined) {
                for (const [key, value] of Object.entries(targetType)) {
                    localVarQueryParameter[key] = value;
                }
            }
            if (targetId !== undefined) {
                for (const [key, value] of Object.entries(targetId)) {
                    localVarQueryParameter[key] = value;
                }
            }
            if (statusCode !== undefined) {
                for (const [key, value] of Object.entries(statusCode)) {
                    localVarQueryParameter[key] = value;
                }
            }
            if (createdAt !== undefined) {
                for (const [key, value] of Object.entries(createdAt)) {
                    localVarQueryParameter[key] = value;
                }
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.AuditApiAxiosParamCreator = AuditApiAxiosParamCreator;
/**
 * AuditApi - functional programming interface
 */
const AuditApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.AuditApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @summary Get audit logs for organization
         * @param {string} organizationId Organization ID
         * @param {number} [page] Page number of the results
         * @param {number} [limit] Number of results per page
         * @param {Date} [from] Deprecated alias for &#x60;createdAt[gte]&#x60;. From date (ISO 8601 format).
         * @param {Date} [to] Deprecated alias for &#x60;createdAt[lte]&#x60;. To date (ISO 8601 format).
         * @param {string} [nextToken] Token for cursor-based pagination. When provided, takes precedence over page parameter.
         * @param {StringFilter} [id] Filter by audit log ID.
         * @param {StringFilter} [actorId] Filter by actor user ID.
         * @param {StringFilter} [actorEmail] Filter by actor email.
         * @param {StringFilter} [actorApiKeyPrefix] Filter by actor API key prefix.
         * @param {StringFilter} [actorApiKeySuffix] Filter by actor API key suffix.
         * @param {StringFilter} [action] Filter by action.
         * @param {StringFilter} [targetType] Filter by target type.
         * @param {StringFilter} [targetId] Filter by target ID.
         * @param {IntFilter} [statusCode] Filter by HTTP status code.
         * @param {DateFilter} [createdAt] Filter by creation timestamp.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getOrganizationAuditLogs(organizationId, page, limit, from, to, nextToken, id, actorId, actorEmail, actorApiKeyPrefix, actorApiKeySuffix, action, targetType, targetId, statusCode, createdAt, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getOrganizationAuditLogs(organizationId, page, limit, from, to, nextToken, id, actorId, actorEmail, actorApiKeyPrefix, actorApiKeySuffix, action, targetType, targetId, statusCode, createdAt, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['AuditApi.getOrganizationAuditLogs']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.AuditApiFp = AuditApiFp;
/**
 * AuditApi - factory interface
 */
const AuditApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.AuditApiFp)(configuration);
    return {
        /**
         *
         * @summary Get audit logs for organization
         * @param {string} organizationId Organization ID
         * @param {number} [page] Page number of the results
         * @param {number} [limit] Number of results per page
         * @param {Date} [from] Deprecated alias for &#x60;createdAt[gte]&#x60;. From date (ISO 8601 format).
         * @param {Date} [to] Deprecated alias for &#x60;createdAt[lte]&#x60;. To date (ISO 8601 format).
         * @param {string} [nextToken] Token for cursor-based pagination. When provided, takes precedence over page parameter.
         * @param {StringFilter} [id] Filter by audit log ID.
         * @param {StringFilter} [actorId] Filter by actor user ID.
         * @param {StringFilter} [actorEmail] Filter by actor email.
         * @param {StringFilter} [actorApiKeyPrefix] Filter by actor API key prefix.
         * @param {StringFilter} [actorApiKeySuffix] Filter by actor API key suffix.
         * @param {StringFilter} [action] Filter by action.
         * @param {StringFilter} [targetType] Filter by target type.
         * @param {StringFilter} [targetId] Filter by target ID.
         * @param {IntFilter} [statusCode] Filter by HTTP status code.
         * @param {DateFilter} [createdAt] Filter by creation timestamp.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getOrganizationAuditLogs(organizationId, page, limit, from, to, nextToken, id, actorId, actorEmail, actorApiKeyPrefix, actorApiKeySuffix, action, targetType, targetId, statusCode, createdAt, options) {
            return localVarFp.getOrganizationAuditLogs(organizationId, page, limit, from, to, nextToken, id, actorId, actorEmail, actorApiKeyPrefix, actorApiKeySuffix, action, targetType, targetId, statusCode, createdAt, options).then((request) => request(axios, basePath));
        },
    };
};
exports.AuditApiFactory = AuditApiFactory;
/**
 * AuditApi - object-oriented interface
 */
class AuditApi extends base_1.BaseAPI {
    /**
     *
     * @summary Get audit logs for organization
     * @param {string} organizationId Organization ID
     * @param {number} [page] Page number of the results
     * @param {number} [limit] Number of results per page
     * @param {Date} [from] Deprecated alias for &#x60;createdAt[gte]&#x60;. From date (ISO 8601 format).
     * @param {Date} [to] Deprecated alias for &#x60;createdAt[lte]&#x60;. To date (ISO 8601 format).
     * @param {string} [nextToken] Token for cursor-based pagination. When provided, takes precedence over page parameter.
     * @param {StringFilter} [id] Filter by audit log ID.
     * @param {StringFilter} [actorId] Filter by actor user ID.
     * @param {StringFilter} [actorEmail] Filter by actor email.
     * @param {StringFilter} [actorApiKeyPrefix] Filter by actor API key prefix.
     * @param {StringFilter} [actorApiKeySuffix] Filter by actor API key suffix.
     * @param {StringFilter} [action] Filter by action.
     * @param {StringFilter} [targetType] Filter by target type.
     * @param {StringFilter} [targetId] Filter by target ID.
     * @param {IntFilter} [statusCode] Filter by HTTP status code.
     * @param {DateFilter} [createdAt] Filter by creation timestamp.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getOrganizationAuditLogs(organizationId, page, limit, from, to, nextToken, id, actorId, actorEmail, actorApiKeyPrefix, actorApiKeySuffix, action, targetType, targetId, statusCode, createdAt, options) {
        return (0, exports.AuditApiFp)(this.configuration).getOrganizationAuditLogs(organizationId, page, limit, from, to, nextToken, id, actorId, actorEmail, actorApiKeyPrefix, actorApiKeySuffix, action, targetType, targetId, statusCode, createdAt, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.AuditApi = AuditApi;
