// @daytona/api-client/src/api/jobs-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Daytona
 * Daytona AI platform API Docs
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@daytona.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobsApi = exports.JobsApiFactory = exports.JobsApiFp = exports.JobsApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * JobsApi - axios parameter creator
 */
const JobsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @summary Get job details
         * @param {string} jobId ID of the job
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getJob: async (jobId, options = {}) => {
            // verify required parameter 'jobId' is not null or undefined
            (0, common_1.assertParamExists)('getJob', 'jobId', jobId);
            const localVarPath = `/jobs/{jobId}`
                .replace(`{${"jobId"}}`, encodeURIComponent(String(jobId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Returns a paginated list of jobs for the runner, optionally filtered by status.
         * @summary List jobs for the runner
         * @param {number} [page] Page number of the results
         * @param {number} [limit] Maximum number of jobs to return (default: 100, max: 500)
         * @param {JobStatus} [status] Filter jobs by status
         * @param {number} [offset] Number of jobs to skip for pagination (default: 0)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listJobs: async (page, limit, status, offset, options = {}) => {
            const localVarPath = `/jobs`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            if (page !== undefined) {
                localVarQueryParameter['page'] = page;
            }
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (status !== undefined) {
                localVarQueryParameter['status'] = status;
            }
            if (offset !== undefined) {
                localVarQueryParameter['offset'] = offset;
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Long poll endpoint for runners to fetch pending jobs. Returns immediately if jobs are available, otherwise waits up to timeout seconds.
         * @summary Long poll for jobs
         * @param {number} [timeout] Timeout in seconds for long polling (default: 30, max: 60)
         * @param {number} [limit] Maximum number of jobs to return (default: 10, max: 100)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pollJobs: async (timeout, limit, options = {}) => {
            const localVarPath = `/jobs/poll`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            if (timeout !== undefined) {
                localVarQueryParameter['timeout'] = timeout;
            }
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Update job status
         * @param {string} jobId ID of the job
         * @param {UpdateJobStatus} updateJobStatus
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateJobStatus: async (jobId, updateJobStatus, options = {}) => {
            // verify required parameter 'jobId' is not null or undefined
            (0, common_1.assertParamExists)('updateJobStatus', 'jobId', jobId);
            // verify required parameter 'updateJobStatus' is not null or undefined
            (0, common_1.assertParamExists)('updateJobStatus', 'updateJobStatus', updateJobStatus);
            const localVarPath = `/jobs/{jobId}/status`
                .replace(`{${"jobId"}}`, encodeURIComponent(String(jobId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(updateJobStatus, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.JobsApiAxiosParamCreator = JobsApiAxiosParamCreator;
/**
 * JobsApi - functional programming interface
 */
const JobsApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.JobsApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @summary Get job details
         * @param {string} jobId ID of the job
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getJob(jobId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getJob(jobId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['JobsApi.getJob']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Returns a paginated list of jobs for the runner, optionally filtered by status.
         * @summary List jobs for the runner
         * @param {number} [page] Page number of the results
         * @param {number} [limit] Maximum number of jobs to return (default: 100, max: 500)
         * @param {JobStatus} [status] Filter jobs by status
         * @param {number} [offset] Number of jobs to skip for pagination (default: 0)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listJobs(page, limit, status, offset, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listJobs(page, limit, status, offset, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['JobsApi.listJobs']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Long poll endpoint for runners to fetch pending jobs. Returns immediately if jobs are available, otherwise waits up to timeout seconds.
         * @summary Long poll for jobs
         * @param {number} [timeout] Timeout in seconds for long polling (default: 30, max: 60)
         * @param {number} [limit] Maximum number of jobs to return (default: 10, max: 100)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async pollJobs(timeout, limit, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.pollJobs(timeout, limit, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['JobsApi.pollJobs']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Update job status
         * @param {string} jobId ID of the job
         * @param {UpdateJobStatus} updateJobStatus
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async updateJobStatus(jobId, updateJobStatus, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.updateJobStatus(jobId, updateJobStatus, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['JobsApi.updateJobStatus']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.JobsApiFp = JobsApiFp;
/**
 * JobsApi - factory interface
 */
const JobsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.JobsApiFp)(configuration);
    return {
        /**
         *
         * @summary Get job details
         * @param {string} jobId ID of the job
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getJob(jobId, options) {
            return localVarFp.getJob(jobId, options).then((request) => request(axios, basePath));
        },
        /**
         * Returns a paginated list of jobs for the runner, optionally filtered by status.
         * @summary List jobs for the runner
         * @param {number} [page] Page number of the results
         * @param {number} [limit] Maximum number of jobs to return (default: 100, max: 500)
         * @param {JobStatus} [status] Filter jobs by status
         * @param {number} [offset] Number of jobs to skip for pagination (default: 0)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listJobs(page, limit, status, offset, options) {
            return localVarFp.listJobs(page, limit, status, offset, options).then((request) => request(axios, basePath));
        },
        /**
         * Long poll endpoint for runners to fetch pending jobs. Returns immediately if jobs are available, otherwise waits up to timeout seconds.
         * @summary Long poll for jobs
         * @param {number} [timeout] Timeout in seconds for long polling (default: 30, max: 60)
         * @param {number} [limit] Maximum number of jobs to return (default: 10, max: 100)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        pollJobs(timeout, limit, options) {
            return localVarFp.pollJobs(timeout, limit, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Update job status
         * @param {string} jobId ID of the job
         * @param {UpdateJobStatus} updateJobStatus
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateJobStatus(jobId, updateJobStatus, options) {
            return localVarFp.updateJobStatus(jobId, updateJobStatus, options).then((request) => request(axios, basePath));
        },
    };
};
exports.JobsApiFactory = JobsApiFactory;
/**
 * JobsApi - object-oriented interface
 */
class JobsApi extends base_1.BaseAPI {
    /**
     *
     * @summary Get job details
     * @param {string} jobId ID of the job
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getJob(jobId, options) {
        return (0, exports.JobsApiFp)(this.configuration).getJob(jobId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Returns a paginated list of jobs for the runner, optionally filtered by status.
     * @summary List jobs for the runner
     * @param {number} [page] Page number of the results
     * @param {number} [limit] Maximum number of jobs to return (default: 100, max: 500)
     * @param {JobStatus} [status] Filter jobs by status
     * @param {number} [offset] Number of jobs to skip for pagination (default: 0)
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    listJobs(page, limit, status, offset, options) {
        return (0, exports.JobsApiFp)(this.configuration).listJobs(page, limit, status, offset, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Long poll endpoint for runners to fetch pending jobs. Returns immediately if jobs are available, otherwise waits up to timeout seconds.
     * @summary Long poll for jobs
     * @param {number} [timeout] Timeout in seconds for long polling (default: 30, max: 60)
     * @param {number} [limit] Maximum number of jobs to return (default: 10, max: 100)
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    pollJobs(timeout, limit, options) {
        return (0, exports.JobsApiFp)(this.configuration).pollJobs(timeout, limit, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Update job status
     * @param {string} jobId ID of the job
     * @param {UpdateJobStatus} updateJobStatus
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    updateJobStatus(jobId, updateJobStatus, options) {
        return (0, exports.JobsApiFp)(this.configuration).updateJobStatus(jobId, updateJobStatus, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.JobsApi = JobsApi;
