// @daytona/api-client/src/api/organizations-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Daytona
 * Daytona AI platform API Docs
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@daytona.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationsApi = exports.OrganizationsApiFactory = exports.OrganizationsApiFp = exports.OrganizationsApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * OrganizationsApi - axios parameter creator
 */
const OrganizationsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @summary Accept organization invitation
         * @param {string} invitationId Invitation ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        acceptOrganizationInvitation: async (invitationId, options = {}) => {
            // verify required parameter 'invitationId' is not null or undefined
            (0, common_1.assertParamExists)('acceptOrganizationInvitation', 'invitationId', invitationId);
            const localVarPath = `/organizations/invitations/{invitationId}/accept`
                .replace(`{${"invitationId"}}`, encodeURIComponent(String(invitationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Cancel organization invitation
         * @param {string} organizationId Organization ID
         * @param {string} invitationId Invitation ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        cancelOrganizationInvitation: async (organizationId, invitationId, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('cancelOrganizationInvitation', 'organizationId', organizationId);
            // verify required parameter 'invitationId' is not null or undefined
            (0, common_1.assertParamExists)('cancelOrganizationInvitation', 'invitationId', invitationId);
            const localVarPath = `/organizations/{organizationId}/invitations/{invitationId}/cancel`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)))
                .replace(`{${"invitationId"}}`, encodeURIComponent(String(invitationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Create organization
         * @param {CreateOrganization} createOrganization
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createOrganization: async (createOrganization, options = {}) => {
            // verify required parameter 'createOrganization' is not null or undefined
            (0, common_1.assertParamExists)('createOrganization', 'createOrganization', createOrganization);
            const localVarPath = `/organizations`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(createOrganization, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Create organization invitation
         * @param {string} organizationId Organization ID
         * @param {CreateOrganizationInvitation} createOrganizationInvitation
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createOrganizationInvitation: async (organizationId, createOrganizationInvitation, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('createOrganizationInvitation', 'organizationId', organizationId);
            // verify required parameter 'createOrganizationInvitation' is not null or undefined
            (0, common_1.assertParamExists)('createOrganizationInvitation', 'createOrganizationInvitation', createOrganizationInvitation);
            const localVarPath = `/organizations/{organizationId}/invitations`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(createOrganizationInvitation, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Create organization role
         * @param {string} organizationId Organization ID
         * @param {CreateOrganizationRole} createOrganizationRole
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createOrganizationRole: async (organizationId, createOrganizationRole, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('createOrganizationRole', 'organizationId', organizationId);
            // verify required parameter 'createOrganizationRole' is not null or undefined
            (0, common_1.assertParamExists)('createOrganizationRole', 'createOrganizationRole', createOrganizationRole);
            const localVarPath = `/organizations/{organizationId}/roles`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(createOrganizationRole, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Create a new region
         * @param {CreateRegion} createRegion
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createRegion: async (createRegion, xDaytonaOrganizationID, options = {}) => {
            // verify required parameter 'createRegion' is not null or undefined
            (0, common_1.assertParamExists)('createRegion', 'createRegion', createRegion);
            const localVarPath = `/regions`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            if (xDaytonaOrganizationID != null) {
                localVarHeaderParameter['X-Daytona-Organization-ID'] = String(xDaytonaOrganizationID);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(createRegion, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Decline organization invitation
         * @param {string} invitationId Invitation ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        declineOrganizationInvitation: async (invitationId, options = {}) => {
            // verify required parameter 'invitationId' is not null or undefined
            (0, common_1.assertParamExists)('declineOrganizationInvitation', 'invitationId', invitationId);
            const localVarPath = `/organizations/invitations/{invitationId}/decline`
                .replace(`{${"invitationId"}}`, encodeURIComponent(String(invitationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Delete organization
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteOrganization: async (organizationId, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('deleteOrganization', 'organizationId', organizationId);
            const localVarPath = `/organizations/{organizationId}`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'DELETE', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Delete organization member
         * @param {string} organizationId Organization ID
         * @param {string} userId User ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteOrganizationMember: async (organizationId, userId, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('deleteOrganizationMember', 'organizationId', organizationId);
            // verify required parameter 'userId' is not null or undefined
            (0, common_1.assertParamExists)('deleteOrganizationMember', 'userId', userId);
            const localVarPath = `/organizations/{organizationId}/users/{userId}`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)))
                .replace(`{${"userId"}}`, encodeURIComponent(String(userId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'DELETE', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Delete organization OpenTelemetry configuration
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteOrganizationOtelConfig: async (organizationId, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('deleteOrganizationOtelConfig', 'organizationId', organizationId);
            const localVarPath = `/organizations/{organizationId}/otel-config`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'DELETE', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Delete organization role
         * @param {string} organizationId Organization ID
         * @param {string} roleId Role ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteOrganizationRole: async (organizationId, roleId, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('deleteOrganizationRole', 'organizationId', organizationId);
            // verify required parameter 'roleId' is not null or undefined
            (0, common_1.assertParamExists)('deleteOrganizationRole', 'roleId', roleId);
            const localVarPath = `/organizations/{organizationId}/roles/{roleId}`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)))
                .replace(`{${"roleId"}}`, encodeURIComponent(String(roleId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'DELETE', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Delete a region
         * @param {string} id Region ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteRegion: async (id, xDaytonaOrganizationID, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('deleteRegion', 'id', id);
            const localVarPath = `/regions/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'DELETE', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            if (xDaytonaOrganizationID != null) {
                localVarHeaderParameter['X-Daytona-Organization-ID'] = String(xDaytonaOrganizationID);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Get organization by ID
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getOrganization: async (organizationId, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('getOrganization', 'organizationId', organizationId);
            const localVarPath = `/organizations/{organizationId}`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Get count of organization invitations for authenticated user
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getOrganizationInvitationsCountForAuthenticatedUser: async (options = {}) => {
            const localVarPath = `/organizations/invitations/count`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Get organization OTEL config by organization ID
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getOrganizationOtelConfig: async (organizationId, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('getOrganizationOtelConfig', 'organizationId', organizationId);
            const localVarPath = `/organizations/{organizationId}/otel-config`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Get organization OTEL config by sandbox auth token
         * @param {string} authToken Sandbox Auth Token
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getOrganizationOtelConfigBySandboxAuthToken: async (authToken, options = {}) => {
            // verify required parameter 'authToken' is not null or undefined
            (0, common_1.assertParamExists)('getOrganizationOtelConfigBySandboxAuthToken', 'authToken', authToken);
            const localVarPath = `/organizations/otel-config/by-sandbox-auth-token/{authToken}`
                .replace(`{${"authToken"}}`, encodeURIComponent(String(authToken)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Get organization current usage overview
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getOrganizationUsageOverview: async (organizationId, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('getOrganizationUsageOverview', 'organizationId', organizationId);
            const localVarPath = `/organizations/{organizationId}/usage`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Get region by ID
         * @param {string} id Region ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getRegionById: async (id, xDaytonaOrganizationID, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('getRegionById', 'id', id);
            const localVarPath = `/regions/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            if (xDaytonaOrganizationID != null) {
                localVarHeaderParameter['X-Daytona-Organization-ID'] = String(xDaytonaOrganizationID);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Leave organization
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        leaveOrganization: async (organizationId, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('leaveOrganization', 'organizationId', organizationId);
            const localVarPath = `/organizations/{organizationId}/leave`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary List all available regions for the organization
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listAvailableRegions: async (xDaytonaOrganizationID, options = {}) => {
            const localVarPath = `/regions`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            if (xDaytonaOrganizationID != null) {
                localVarHeaderParameter['X-Daytona-Organization-ID'] = String(xDaytonaOrganizationID);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary List pending organization invitations
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listOrganizationInvitations: async (organizationId, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('listOrganizationInvitations', 'organizationId', organizationId);
            const localVarPath = `/organizations/{organizationId}/invitations`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary List organization invitations for authenticated user
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listOrganizationInvitationsForAuthenticatedUser: async (options = {}) => {
            const localVarPath = `/organizations/invitations`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary List organization members
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listOrganizationMembers: async (organizationId, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('listOrganizationMembers', 'organizationId', organizationId);
            const localVarPath = `/organizations/{organizationId}/users`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary List organization roles
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listOrganizationRoles: async (organizationId, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('listOrganizationRoles', 'organizationId', organizationId);
            const localVarPath = `/organizations/{organizationId}/roles`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary List organizations
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listOrganizations: async (options = {}) => {
            const localVarPath = `/organizations`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Regenerate proxy API key for a region
         * @param {string} id Region ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        regenerateProxyApiKey: async (id, xDaytonaOrganizationID, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('regenerateProxyApiKey', 'id', id);
            const localVarPath = `/regions/{id}/regenerate-proxy-api-key`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            if (xDaytonaOrganizationID != null) {
                localVarHeaderParameter['X-Daytona-Organization-ID'] = String(xDaytonaOrganizationID);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Regenerate snapshot manager credentials for a region
         * @param {string} id Region ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        regenerateSnapshotManagerCredentials: async (id, xDaytonaOrganizationID, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('regenerateSnapshotManagerCredentials', 'id', id);
            const localVarPath = `/regions/{id}/regenerate-snapshot-manager-credentials`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            if (xDaytonaOrganizationID != null) {
                localVarHeaderParameter['X-Daytona-Organization-ID'] = String(xDaytonaOrganizationID);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Regenerate SSH gateway API key for a region
         * @param {string} id Region ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        regenerateSshGatewayApiKey: async (id, xDaytonaOrganizationID, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('regenerateSshGatewayApiKey', 'id', id);
            const localVarPath = `/regions/{id}/regenerate-ssh-gateway-api-key`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            if (xDaytonaOrganizationID != null) {
                localVarHeaderParameter['X-Daytona-Organization-ID'] = String(xDaytonaOrganizationID);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Set default region for organization
         * @param {string} organizationId Organization ID
         * @param {UpdateOrganizationDefaultRegion} updateOrganizationDefaultRegion
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        setOrganizationDefaultRegion: async (organizationId, updateOrganizationDefaultRegion, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('setOrganizationDefaultRegion', 'organizationId', organizationId);
            // verify required parameter 'updateOrganizationDefaultRegion' is not null or undefined
            (0, common_1.assertParamExists)('setOrganizationDefaultRegion', 'updateOrganizationDefaultRegion', updateOrganizationDefaultRegion);
            const localVarPath = `/organizations/{organizationId}/default-region`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'PATCH', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(updateOrganizationDefaultRegion, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Suspend organization
         * @param {string} organizationId Organization ID
         * @param {OrganizationSuspension} [organizationSuspension]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        suspendOrganization: async (organizationId, organizationSuspension, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('suspendOrganization', 'organizationId', organizationId);
            const localVarPath = `/organizations/{organizationId}/suspend`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(organizationSuspension, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Unsuspend organization
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        unsuspendOrganization: async (organizationId, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('unsuspendOrganization', 'organizationId', organizationId);
            const localVarPath = `/organizations/{organizationId}/unsuspend`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Update access for organization member
         * @param {string} organizationId Organization ID
         * @param {string} userId User ID
         * @param {UpdateOrganizationMemberAccess} updateOrganizationMemberAccess
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateAccessForOrganizationMember: async (organizationId, userId, updateOrganizationMemberAccess, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('updateAccessForOrganizationMember', 'organizationId', organizationId);
            // verify required parameter 'userId' is not null or undefined
            (0, common_1.assertParamExists)('updateAccessForOrganizationMember', 'userId', userId);
            // verify required parameter 'updateOrganizationMemberAccess' is not null or undefined
            (0, common_1.assertParamExists)('updateAccessForOrganizationMember', 'updateOrganizationMemberAccess', updateOrganizationMemberAccess);
            const localVarPath = `/organizations/{organizationId}/users/{userId}/access`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)))
                .replace(`{${"userId"}}`, encodeURIComponent(String(userId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(updateOrganizationMemberAccess, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Update experimental configuration
         * @param {string} organizationId Organization ID
         * @param {{ [key: string]: any; }} [requestBody] Experimental configuration as a JSON object. Set to null to clear the configuration.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateExperimentalConfig: async (organizationId, requestBody, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('updateExperimentalConfig', 'organizationId', organizationId);
            const localVarPath = `/organizations/{organizationId}/experimental-config`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'PUT', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(requestBody, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Update organization invitation
         * @param {string} organizationId Organization ID
         * @param {string} invitationId Invitation ID
         * @param {UpdateOrganizationInvitation} updateOrganizationInvitation
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateOrganizationInvitation: async (organizationId, invitationId, updateOrganizationInvitation, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('updateOrganizationInvitation', 'organizationId', organizationId);
            // verify required parameter 'invitationId' is not null or undefined
            (0, common_1.assertParamExists)('updateOrganizationInvitation', 'invitationId', invitationId);
            // verify required parameter 'updateOrganizationInvitation' is not null or undefined
            (0, common_1.assertParamExists)('updateOrganizationInvitation', 'updateOrganizationInvitation', updateOrganizationInvitation);
            const localVarPath = `/organizations/{organizationId}/invitations/{invitationId}`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)))
                .replace(`{${"invitationId"}}`, encodeURIComponent(String(invitationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'PUT', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(updateOrganizationInvitation, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Update organization OpenTelemetry configuration
         * @param {string} organizationId Organization ID
         * @param {OtelConfig} otelConfig
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateOrganizationOtelConfig: async (organizationId, otelConfig, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('updateOrganizationOtelConfig', 'organizationId', organizationId);
            // verify required parameter 'otelConfig' is not null or undefined
            (0, common_1.assertParamExists)('updateOrganizationOtelConfig', 'otelConfig', otelConfig);
            const localVarPath = `/organizations/{organizationId}/otel-config`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'PUT', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(otelConfig, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Update organization quota
         * @param {string} organizationId Organization ID
         * @param {UpdateOrganizationQuota} updateOrganizationQuota
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateOrganizationQuota: async (organizationId, updateOrganizationQuota, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('updateOrganizationQuota', 'organizationId', organizationId);
            // verify required parameter 'updateOrganizationQuota' is not null or undefined
            (0, common_1.assertParamExists)('updateOrganizationQuota', 'updateOrganizationQuota', updateOrganizationQuota);
            const localVarPath = `/organizations/{organizationId}/quota`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'PATCH', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(updateOrganizationQuota, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Update organization region quota
         * @param {string} organizationId Organization ID
         * @param {string} regionId ID of the region where the updated quota will be applied
         * @param {UpdateOrganizationRegionQuota} updateOrganizationRegionQuota
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateOrganizationRegionQuota: async (organizationId, regionId, updateOrganizationRegionQuota, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('updateOrganizationRegionQuota', 'organizationId', organizationId);
            // verify required parameter 'regionId' is not null or undefined
            (0, common_1.assertParamExists)('updateOrganizationRegionQuota', 'regionId', regionId);
            // verify required parameter 'updateOrganizationRegionQuota' is not null or undefined
            (0, common_1.assertParamExists)('updateOrganizationRegionQuota', 'updateOrganizationRegionQuota', updateOrganizationRegionQuota);
            const localVarPath = `/organizations/{organizationId}/quota/{regionId}`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)))
                .replace(`{${"regionId"}}`, encodeURIComponent(String(regionId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'PATCH', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(updateOrganizationRegionQuota, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Update organization role
         * @param {string} organizationId Organization ID
         * @param {string} roleId Role ID
         * @param {UpdateOrganizationRole} updateOrganizationRole
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateOrganizationRole: async (organizationId, roleId, updateOrganizationRole, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('updateOrganizationRole', 'organizationId', organizationId);
            // verify required parameter 'roleId' is not null or undefined
            (0, common_1.assertParamExists)('updateOrganizationRole', 'roleId', roleId);
            // verify required parameter 'updateOrganizationRole' is not null or undefined
            (0, common_1.assertParamExists)('updateOrganizationRole', 'updateOrganizationRole', updateOrganizationRole);
            const localVarPath = `/organizations/{organizationId}/roles/{roleId}`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)))
                .replace(`{${"roleId"}}`, encodeURIComponent(String(roleId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'PUT', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(updateOrganizationRole, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Update region configuration
         * @param {string} id Region ID
         * @param {UpdateRegion} updateRegion
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateRegion: async (id, updateRegion, xDaytonaOrganizationID, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('updateRegion', 'id', id);
            // verify required parameter 'updateRegion' is not null or undefined
            (0, common_1.assertParamExists)('updateRegion', 'updateRegion', updateRegion);
            const localVarPath = `/regions/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'PATCH', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            if (xDaytonaOrganizationID != null) {
                localVarHeaderParameter['X-Daytona-Organization-ID'] = String(xDaytonaOrganizationID);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(updateRegion, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Update sandbox default limited network egress
         * @param {string} organizationId Organization ID
         * @param {OrganizationSandboxDefaultLimitedNetworkEgress} organizationSandboxDefaultLimitedNetworkEgress
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateSandboxDefaultLimitedNetworkEgress: async (organizationId, organizationSandboxDefaultLimitedNetworkEgress, options = {}) => {
            // verify required parameter 'organizationId' is not null or undefined
            (0, common_1.assertParamExists)('updateSandboxDefaultLimitedNetworkEgress', 'organizationId', organizationId);
            // verify required parameter 'organizationSandboxDefaultLimitedNetworkEgress' is not null or undefined
            (0, common_1.assertParamExists)('updateSandboxDefaultLimitedNetworkEgress', 'organizationSandboxDefaultLimitedNetworkEgress', organizationSandboxDefaultLimitedNetworkEgress);
            const localVarPath = `/organizations/{organizationId}/sandbox-default-limited-network-egress`
                .replace(`{${"organizationId"}}`, encodeURIComponent(String(organizationId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(organizationSandboxDefaultLimitedNetworkEgress, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.OrganizationsApiAxiosParamCreator = OrganizationsApiAxiosParamCreator;
/**
 * OrganizationsApi - functional programming interface
 */
const OrganizationsApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.OrganizationsApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @summary Accept organization invitation
         * @param {string} invitationId Invitation ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async acceptOrganizationInvitation(invitationId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.acceptOrganizationInvitation(invitationId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.acceptOrganizationInvitation']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Cancel organization invitation
         * @param {string} organizationId Organization ID
         * @param {string} invitationId Invitation ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async cancelOrganizationInvitation(organizationId, invitationId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.cancelOrganizationInvitation(organizationId, invitationId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.cancelOrganizationInvitation']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Create organization
         * @param {CreateOrganization} createOrganization
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async createOrganization(createOrganization, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.createOrganization(createOrganization, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.createOrganization']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Create organization invitation
         * @param {string} organizationId Organization ID
         * @param {CreateOrganizationInvitation} createOrganizationInvitation
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async createOrganizationInvitation(organizationId, createOrganizationInvitation, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.createOrganizationInvitation(organizationId, createOrganizationInvitation, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.createOrganizationInvitation']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Create organization role
         * @param {string} organizationId Organization ID
         * @param {CreateOrganizationRole} createOrganizationRole
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async createOrganizationRole(organizationId, createOrganizationRole, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.createOrganizationRole(organizationId, createOrganizationRole, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.createOrganizationRole']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Create a new region
         * @param {CreateRegion} createRegion
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async createRegion(createRegion, xDaytonaOrganizationID, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.createRegion(createRegion, xDaytonaOrganizationID, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.createRegion']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Decline organization invitation
         * @param {string} invitationId Invitation ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async declineOrganizationInvitation(invitationId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.declineOrganizationInvitation(invitationId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.declineOrganizationInvitation']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Delete organization
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async deleteOrganization(organizationId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.deleteOrganization(organizationId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.deleteOrganization']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Delete organization member
         * @param {string} organizationId Organization ID
         * @param {string} userId User ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async deleteOrganizationMember(organizationId, userId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.deleteOrganizationMember(organizationId, userId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.deleteOrganizationMember']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Delete organization OpenTelemetry configuration
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async deleteOrganizationOtelConfig(organizationId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.deleteOrganizationOtelConfig(organizationId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.deleteOrganizationOtelConfig']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Delete organization role
         * @param {string} organizationId Organization ID
         * @param {string} roleId Role ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async deleteOrganizationRole(organizationId, roleId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.deleteOrganizationRole(organizationId, roleId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.deleteOrganizationRole']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Delete a region
         * @param {string} id Region ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async deleteRegion(id, xDaytonaOrganizationID, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.deleteRegion(id, xDaytonaOrganizationID, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.deleteRegion']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Get organization by ID
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getOrganization(organizationId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getOrganization(organizationId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.getOrganization']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Get count of organization invitations for authenticated user
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getOrganizationInvitationsCountForAuthenticatedUser(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getOrganizationInvitationsCountForAuthenticatedUser(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.getOrganizationInvitationsCountForAuthenticatedUser']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Get organization OTEL config by organization ID
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getOrganizationOtelConfig(organizationId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getOrganizationOtelConfig(organizationId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.getOrganizationOtelConfig']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Get organization OTEL config by sandbox auth token
         * @param {string} authToken Sandbox Auth Token
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getOrganizationOtelConfigBySandboxAuthToken(authToken, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getOrganizationOtelConfigBySandboxAuthToken(authToken, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.getOrganizationOtelConfigBySandboxAuthToken']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Get organization current usage overview
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getOrganizationUsageOverview(organizationId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getOrganizationUsageOverview(organizationId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.getOrganizationUsageOverview']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Get region by ID
         * @param {string} id Region ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getRegionById(id, xDaytonaOrganizationID, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getRegionById(id, xDaytonaOrganizationID, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.getRegionById']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Leave organization
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async leaveOrganization(organizationId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.leaveOrganization(organizationId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.leaveOrganization']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary List all available regions for the organization
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listAvailableRegions(xDaytonaOrganizationID, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listAvailableRegions(xDaytonaOrganizationID, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.listAvailableRegions']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary List pending organization invitations
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listOrganizationInvitations(organizationId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listOrganizationInvitations(organizationId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.listOrganizationInvitations']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary List organization invitations for authenticated user
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listOrganizationInvitationsForAuthenticatedUser(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listOrganizationInvitationsForAuthenticatedUser(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.listOrganizationInvitationsForAuthenticatedUser']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary List organization members
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listOrganizationMembers(organizationId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listOrganizationMembers(organizationId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.listOrganizationMembers']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary List organization roles
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listOrganizationRoles(organizationId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listOrganizationRoles(organizationId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.listOrganizationRoles']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary List organizations
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listOrganizations(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listOrganizations(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.listOrganizations']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Regenerate proxy API key for a region
         * @param {string} id Region ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async regenerateProxyApiKey(id, xDaytonaOrganizationID, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.regenerateProxyApiKey(id, xDaytonaOrganizationID, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.regenerateProxyApiKey']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Regenerate snapshot manager credentials for a region
         * @param {string} id Region ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async regenerateSnapshotManagerCredentials(id, xDaytonaOrganizationID, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.regenerateSnapshotManagerCredentials(id, xDaytonaOrganizationID, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.regenerateSnapshotManagerCredentials']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Regenerate SSH gateway API key for a region
         * @param {string} id Region ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async regenerateSshGatewayApiKey(id, xDaytonaOrganizationID, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.regenerateSshGatewayApiKey(id, xDaytonaOrganizationID, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.regenerateSshGatewayApiKey']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Set default region for organization
         * @param {string} organizationId Organization ID
         * @param {UpdateOrganizationDefaultRegion} updateOrganizationDefaultRegion
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async setOrganizationDefaultRegion(organizationId, updateOrganizationDefaultRegion, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.setOrganizationDefaultRegion(organizationId, updateOrganizationDefaultRegion, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.setOrganizationDefaultRegion']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Suspend organization
         * @param {string} organizationId Organization ID
         * @param {OrganizationSuspension} [organizationSuspension]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async suspendOrganization(organizationId, organizationSuspension, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.suspendOrganization(organizationId, organizationSuspension, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.suspendOrganization']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Unsuspend organization
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async unsuspendOrganization(organizationId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.unsuspendOrganization(organizationId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.unsuspendOrganization']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Update access for organization member
         * @param {string} organizationId Organization ID
         * @param {string} userId User ID
         * @param {UpdateOrganizationMemberAccess} updateOrganizationMemberAccess
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async updateAccessForOrganizationMember(organizationId, userId, updateOrganizationMemberAccess, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.updateAccessForOrganizationMember(organizationId, userId, updateOrganizationMemberAccess, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.updateAccessForOrganizationMember']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Update experimental configuration
         * @param {string} organizationId Organization ID
         * @param {{ [key: string]: any; }} [requestBody] Experimental configuration as a JSON object. Set to null to clear the configuration.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async updateExperimentalConfig(organizationId, requestBody, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.updateExperimentalConfig(organizationId, requestBody, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.updateExperimentalConfig']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Update organization invitation
         * @param {string} organizationId Organization ID
         * @param {string} invitationId Invitation ID
         * @param {UpdateOrganizationInvitation} updateOrganizationInvitation
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async updateOrganizationInvitation(organizationId, invitationId, updateOrganizationInvitation, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.updateOrganizationInvitation(organizationId, invitationId, updateOrganizationInvitation, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.updateOrganizationInvitation']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Update organization OpenTelemetry configuration
         * @param {string} organizationId Organization ID
         * @param {OtelConfig} otelConfig
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async updateOrganizationOtelConfig(organizationId, otelConfig, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.updateOrganizationOtelConfig(organizationId, otelConfig, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.updateOrganizationOtelConfig']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Update organization quota
         * @param {string} organizationId Organization ID
         * @param {UpdateOrganizationQuota} updateOrganizationQuota
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async updateOrganizationQuota(organizationId, updateOrganizationQuota, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.updateOrganizationQuota(organizationId, updateOrganizationQuota, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.updateOrganizationQuota']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Update organization region quota
         * @param {string} organizationId Organization ID
         * @param {string} regionId ID of the region where the updated quota will be applied
         * @param {UpdateOrganizationRegionQuota} updateOrganizationRegionQuota
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async updateOrganizationRegionQuota(organizationId, regionId, updateOrganizationRegionQuota, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.updateOrganizationRegionQuota(organizationId, regionId, updateOrganizationRegionQuota, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.updateOrganizationRegionQuota']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Update organization role
         * @param {string} organizationId Organization ID
         * @param {string} roleId Role ID
         * @param {UpdateOrganizationRole} updateOrganizationRole
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async updateOrganizationRole(organizationId, roleId, updateOrganizationRole, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.updateOrganizationRole(organizationId, roleId, updateOrganizationRole, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.updateOrganizationRole']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Update region configuration
         * @param {string} id Region ID
         * @param {UpdateRegion} updateRegion
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async updateRegion(id, updateRegion, xDaytonaOrganizationID, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.updateRegion(id, updateRegion, xDaytonaOrganizationID, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.updateRegion']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Update sandbox default limited network egress
         * @param {string} organizationId Organization ID
         * @param {OrganizationSandboxDefaultLimitedNetworkEgress} organizationSandboxDefaultLimitedNetworkEgress
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async updateSandboxDefaultLimitedNetworkEgress(organizationId, organizationSandboxDefaultLimitedNetworkEgress, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.updateSandboxDefaultLimitedNetworkEgress(organizationId, organizationSandboxDefaultLimitedNetworkEgress, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['OrganizationsApi.updateSandboxDefaultLimitedNetworkEgress']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.OrganizationsApiFp = OrganizationsApiFp;
/**
 * OrganizationsApi - factory interface
 */
const OrganizationsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.OrganizationsApiFp)(configuration);
    return {
        /**
         *
         * @summary Accept organization invitation
         * @param {string} invitationId Invitation ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        acceptOrganizationInvitation(invitationId, options) {
            return localVarFp.acceptOrganizationInvitation(invitationId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Cancel organization invitation
         * @param {string} organizationId Organization ID
         * @param {string} invitationId Invitation ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        cancelOrganizationInvitation(organizationId, invitationId, options) {
            return localVarFp.cancelOrganizationInvitation(organizationId, invitationId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Create organization
         * @param {CreateOrganization} createOrganization
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createOrganization(createOrganization, options) {
            return localVarFp.createOrganization(createOrganization, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Create organization invitation
         * @param {string} organizationId Organization ID
         * @param {CreateOrganizationInvitation} createOrganizationInvitation
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createOrganizationInvitation(organizationId, createOrganizationInvitation, options) {
            return localVarFp.createOrganizationInvitation(organizationId, createOrganizationInvitation, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Create organization role
         * @param {string} organizationId Organization ID
         * @param {CreateOrganizationRole} createOrganizationRole
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createOrganizationRole(organizationId, createOrganizationRole, options) {
            return localVarFp.createOrganizationRole(organizationId, createOrganizationRole, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Create a new region
         * @param {CreateRegion} createRegion
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createRegion(createRegion, xDaytonaOrganizationID, options) {
            return localVarFp.createRegion(createRegion, xDaytonaOrganizationID, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Decline organization invitation
         * @param {string} invitationId Invitation ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        declineOrganizationInvitation(invitationId, options) {
            return localVarFp.declineOrganizationInvitation(invitationId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Delete organization
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteOrganization(organizationId, options) {
            return localVarFp.deleteOrganization(organizationId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Delete organization member
         * @param {string} organizationId Organization ID
         * @param {string} userId User ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteOrganizationMember(organizationId, userId, options) {
            return localVarFp.deleteOrganizationMember(organizationId, userId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Delete organization OpenTelemetry configuration
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteOrganizationOtelConfig(organizationId, options) {
            return localVarFp.deleteOrganizationOtelConfig(organizationId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Delete organization role
         * @param {string} organizationId Organization ID
         * @param {string} roleId Role ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteOrganizationRole(organizationId, roleId, options) {
            return localVarFp.deleteOrganizationRole(organizationId, roleId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Delete a region
         * @param {string} id Region ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteRegion(id, xDaytonaOrganizationID, options) {
            return localVarFp.deleteRegion(id, xDaytonaOrganizationID, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get organization by ID
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getOrganization(organizationId, options) {
            return localVarFp.getOrganization(organizationId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get count of organization invitations for authenticated user
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getOrganizationInvitationsCountForAuthenticatedUser(options) {
            return localVarFp.getOrganizationInvitationsCountForAuthenticatedUser(options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get organization OTEL config by organization ID
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getOrganizationOtelConfig(organizationId, options) {
            return localVarFp.getOrganizationOtelConfig(organizationId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get organization OTEL config by sandbox auth token
         * @param {string} authToken Sandbox Auth Token
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getOrganizationOtelConfigBySandboxAuthToken(authToken, options) {
            return localVarFp.getOrganizationOtelConfigBySandboxAuthToken(authToken, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get organization current usage overview
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getOrganizationUsageOverview(organizationId, options) {
            return localVarFp.getOrganizationUsageOverview(organizationId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get region by ID
         * @param {string} id Region ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getRegionById(id, xDaytonaOrganizationID, options) {
            return localVarFp.getRegionById(id, xDaytonaOrganizationID, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Leave organization
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        leaveOrganization(organizationId, options) {
            return localVarFp.leaveOrganization(organizationId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary List all available regions for the organization
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listAvailableRegions(xDaytonaOrganizationID, options) {
            return localVarFp.listAvailableRegions(xDaytonaOrganizationID, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary List pending organization invitations
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listOrganizationInvitations(organizationId, options) {
            return localVarFp.listOrganizationInvitations(organizationId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary List organization invitations for authenticated user
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listOrganizationInvitationsForAuthenticatedUser(options) {
            return localVarFp.listOrganizationInvitationsForAuthenticatedUser(options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary List organization members
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listOrganizationMembers(organizationId, options) {
            return localVarFp.listOrganizationMembers(organizationId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary List organization roles
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listOrganizationRoles(organizationId, options) {
            return localVarFp.listOrganizationRoles(organizationId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary List organizations
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listOrganizations(options) {
            return localVarFp.listOrganizations(options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Regenerate proxy API key for a region
         * @param {string} id Region ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        regenerateProxyApiKey(id, xDaytonaOrganizationID, options) {
            return localVarFp.regenerateProxyApiKey(id, xDaytonaOrganizationID, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Regenerate snapshot manager credentials for a region
         * @param {string} id Region ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        regenerateSnapshotManagerCredentials(id, xDaytonaOrganizationID, options) {
            return localVarFp.regenerateSnapshotManagerCredentials(id, xDaytonaOrganizationID, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Regenerate SSH gateway API key for a region
         * @param {string} id Region ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        regenerateSshGatewayApiKey(id, xDaytonaOrganizationID, options) {
            return localVarFp.regenerateSshGatewayApiKey(id, xDaytonaOrganizationID, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Set default region for organization
         * @param {string} organizationId Organization ID
         * @param {UpdateOrganizationDefaultRegion} updateOrganizationDefaultRegion
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        setOrganizationDefaultRegion(organizationId, updateOrganizationDefaultRegion, options) {
            return localVarFp.setOrganizationDefaultRegion(organizationId, updateOrganizationDefaultRegion, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Suspend organization
         * @param {string} organizationId Organization ID
         * @param {OrganizationSuspension} [organizationSuspension]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        suspendOrganization(organizationId, organizationSuspension, options) {
            return localVarFp.suspendOrganization(organizationId, organizationSuspension, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Unsuspend organization
         * @param {string} organizationId Organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        unsuspendOrganization(organizationId, options) {
            return localVarFp.unsuspendOrganization(organizationId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Update access for organization member
         * @param {string} organizationId Organization ID
         * @param {string} userId User ID
         * @param {UpdateOrganizationMemberAccess} updateOrganizationMemberAccess
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateAccessForOrganizationMember(organizationId, userId, updateOrganizationMemberAccess, options) {
            return localVarFp.updateAccessForOrganizationMember(organizationId, userId, updateOrganizationMemberAccess, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Update experimental configuration
         * @param {string} organizationId Organization ID
         * @param {{ [key: string]: any; }} [requestBody] Experimental configuration as a JSON object. Set to null to clear the configuration.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateExperimentalConfig(organizationId, requestBody, options) {
            return localVarFp.updateExperimentalConfig(organizationId, requestBody, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Update organization invitation
         * @param {string} organizationId Organization ID
         * @param {string} invitationId Invitation ID
         * @param {UpdateOrganizationInvitation} updateOrganizationInvitation
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateOrganizationInvitation(organizationId, invitationId, updateOrganizationInvitation, options) {
            return localVarFp.updateOrganizationInvitation(organizationId, invitationId, updateOrganizationInvitation, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Update organization OpenTelemetry configuration
         * @param {string} organizationId Organization ID
         * @param {OtelConfig} otelConfig
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateOrganizationOtelConfig(organizationId, otelConfig, options) {
            return localVarFp.updateOrganizationOtelConfig(organizationId, otelConfig, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Update organization quota
         * @param {string} organizationId Organization ID
         * @param {UpdateOrganizationQuota} updateOrganizationQuota
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateOrganizationQuota(organizationId, updateOrganizationQuota, options) {
            return localVarFp.updateOrganizationQuota(organizationId, updateOrganizationQuota, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Update organization region quota
         * @param {string} organizationId Organization ID
         * @param {string} regionId ID of the region where the updated quota will be applied
         * @param {UpdateOrganizationRegionQuota} updateOrganizationRegionQuota
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateOrganizationRegionQuota(organizationId, regionId, updateOrganizationRegionQuota, options) {
            return localVarFp.updateOrganizationRegionQuota(organizationId, regionId, updateOrganizationRegionQuota, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Update organization role
         * @param {string} organizationId Organization ID
         * @param {string} roleId Role ID
         * @param {UpdateOrganizationRole} updateOrganizationRole
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateOrganizationRole(organizationId, roleId, updateOrganizationRole, options) {
            return localVarFp.updateOrganizationRole(organizationId, roleId, updateOrganizationRole, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Update region configuration
         * @param {string} id Region ID
         * @param {UpdateRegion} updateRegion
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateRegion(id, updateRegion, xDaytonaOrganizationID, options) {
            return localVarFp.updateRegion(id, updateRegion, xDaytonaOrganizationID, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Update sandbox default limited network egress
         * @param {string} organizationId Organization ID
         * @param {OrganizationSandboxDefaultLimitedNetworkEgress} organizationSandboxDefaultLimitedNetworkEgress
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateSandboxDefaultLimitedNetworkEgress(organizationId, organizationSandboxDefaultLimitedNetworkEgress, options) {
            return localVarFp.updateSandboxDefaultLimitedNetworkEgress(organizationId, organizationSandboxDefaultLimitedNetworkEgress, options).then((request) => request(axios, basePath));
        },
    };
};
exports.OrganizationsApiFactory = OrganizationsApiFactory;
/**
 * OrganizationsApi - object-oriented interface
 */
class OrganizationsApi extends base_1.BaseAPI {
    /**
     *
     * @summary Accept organization invitation
     * @param {string} invitationId Invitation ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    acceptOrganizationInvitation(invitationId, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).acceptOrganizationInvitation(invitationId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Cancel organization invitation
     * @param {string} organizationId Organization ID
     * @param {string} invitationId Invitation ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    cancelOrganizationInvitation(organizationId, invitationId, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).cancelOrganizationInvitation(organizationId, invitationId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Create organization
     * @param {CreateOrganization} createOrganization
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    createOrganization(createOrganization, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).createOrganization(createOrganization, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Create organization invitation
     * @param {string} organizationId Organization ID
     * @param {CreateOrganizationInvitation} createOrganizationInvitation
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    createOrganizationInvitation(organizationId, createOrganizationInvitation, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).createOrganizationInvitation(organizationId, createOrganizationInvitation, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Create organization role
     * @param {string} organizationId Organization ID
     * @param {CreateOrganizationRole} createOrganizationRole
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    createOrganizationRole(organizationId, createOrganizationRole, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).createOrganizationRole(organizationId, createOrganizationRole, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Create a new region
     * @param {CreateRegion} createRegion
     * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    createRegion(createRegion, xDaytonaOrganizationID, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).createRegion(createRegion, xDaytonaOrganizationID, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Decline organization invitation
     * @param {string} invitationId Invitation ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    declineOrganizationInvitation(invitationId, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).declineOrganizationInvitation(invitationId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Delete organization
     * @param {string} organizationId Organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    deleteOrganization(organizationId, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).deleteOrganization(organizationId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Delete organization member
     * @param {string} organizationId Organization ID
     * @param {string} userId User ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    deleteOrganizationMember(organizationId, userId, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).deleteOrganizationMember(organizationId, userId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Delete organization OpenTelemetry configuration
     * @param {string} organizationId Organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    deleteOrganizationOtelConfig(organizationId, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).deleteOrganizationOtelConfig(organizationId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Delete organization role
     * @param {string} organizationId Organization ID
     * @param {string} roleId Role ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    deleteOrganizationRole(organizationId, roleId, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).deleteOrganizationRole(organizationId, roleId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Delete a region
     * @param {string} id Region ID
     * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    deleteRegion(id, xDaytonaOrganizationID, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).deleteRegion(id, xDaytonaOrganizationID, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get organization by ID
     * @param {string} organizationId Organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getOrganization(organizationId, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).getOrganization(organizationId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get count of organization invitations for authenticated user
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getOrganizationInvitationsCountForAuthenticatedUser(options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).getOrganizationInvitationsCountForAuthenticatedUser(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get organization OTEL config by organization ID
     * @param {string} organizationId Organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getOrganizationOtelConfig(organizationId, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).getOrganizationOtelConfig(organizationId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get organization OTEL config by sandbox auth token
     * @param {string} authToken Sandbox Auth Token
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getOrganizationOtelConfigBySandboxAuthToken(authToken, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).getOrganizationOtelConfigBySandboxAuthToken(authToken, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get organization current usage overview
     * @param {string} organizationId Organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getOrganizationUsageOverview(organizationId, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).getOrganizationUsageOverview(organizationId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get region by ID
     * @param {string} id Region ID
     * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getRegionById(id, xDaytonaOrganizationID, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).getRegionById(id, xDaytonaOrganizationID, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Leave organization
     * @param {string} organizationId Organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    leaveOrganization(organizationId, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).leaveOrganization(organizationId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary List all available regions for the organization
     * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    listAvailableRegions(xDaytonaOrganizationID, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).listAvailableRegions(xDaytonaOrganizationID, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary List pending organization invitations
     * @param {string} organizationId Organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    listOrganizationInvitations(organizationId, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).listOrganizationInvitations(organizationId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary List organization invitations for authenticated user
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    listOrganizationInvitationsForAuthenticatedUser(options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).listOrganizationInvitationsForAuthenticatedUser(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary List organization members
     * @param {string} organizationId Organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    listOrganizationMembers(organizationId, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).listOrganizationMembers(organizationId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary List organization roles
     * @param {string} organizationId Organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    listOrganizationRoles(organizationId, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).listOrganizationRoles(organizationId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary List organizations
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    listOrganizations(options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).listOrganizations(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Regenerate proxy API key for a region
     * @param {string} id Region ID
     * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    regenerateProxyApiKey(id, xDaytonaOrganizationID, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).regenerateProxyApiKey(id, xDaytonaOrganizationID, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Regenerate snapshot manager credentials for a region
     * @param {string} id Region ID
     * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    regenerateSnapshotManagerCredentials(id, xDaytonaOrganizationID, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).regenerateSnapshotManagerCredentials(id, xDaytonaOrganizationID, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Regenerate SSH gateway API key for a region
     * @param {string} id Region ID
     * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    regenerateSshGatewayApiKey(id, xDaytonaOrganizationID, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).regenerateSshGatewayApiKey(id, xDaytonaOrganizationID, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Set default region for organization
     * @param {string} organizationId Organization ID
     * @param {UpdateOrganizationDefaultRegion} updateOrganizationDefaultRegion
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    setOrganizationDefaultRegion(organizationId, updateOrganizationDefaultRegion, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).setOrganizationDefaultRegion(organizationId, updateOrganizationDefaultRegion, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Suspend organization
     * @param {string} organizationId Organization ID
     * @param {OrganizationSuspension} [organizationSuspension]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    suspendOrganization(organizationId, organizationSuspension, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).suspendOrganization(organizationId, organizationSuspension, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Unsuspend organization
     * @param {string} organizationId Organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    unsuspendOrganization(organizationId, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).unsuspendOrganization(organizationId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Update access for organization member
     * @param {string} organizationId Organization ID
     * @param {string} userId User ID
     * @param {UpdateOrganizationMemberAccess} updateOrganizationMemberAccess
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    updateAccessForOrganizationMember(organizationId, userId, updateOrganizationMemberAccess, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).updateAccessForOrganizationMember(organizationId, userId, updateOrganizationMemberAccess, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Update experimental configuration
     * @param {string} organizationId Organization ID
     * @param {{ [key: string]: any; }} [requestBody] Experimental configuration as a JSON object. Set to null to clear the configuration.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    updateExperimentalConfig(organizationId, requestBody, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).updateExperimentalConfig(organizationId, requestBody, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Update organization invitation
     * @param {string} organizationId Organization ID
     * @param {string} invitationId Invitation ID
     * @param {UpdateOrganizationInvitation} updateOrganizationInvitation
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    updateOrganizationInvitation(organizationId, invitationId, updateOrganizationInvitation, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).updateOrganizationInvitation(organizationId, invitationId, updateOrganizationInvitation, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Update organization OpenTelemetry configuration
     * @param {string} organizationId Organization ID
     * @param {OtelConfig} otelConfig
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    updateOrganizationOtelConfig(organizationId, otelConfig, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).updateOrganizationOtelConfig(organizationId, otelConfig, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Update organization quota
     * @param {string} organizationId Organization ID
     * @param {UpdateOrganizationQuota} updateOrganizationQuota
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    updateOrganizationQuota(organizationId, updateOrganizationQuota, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).updateOrganizationQuota(organizationId, updateOrganizationQuota, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Update organization region quota
     * @param {string} organizationId Organization ID
     * @param {string} regionId ID of the region where the updated quota will be applied
     * @param {UpdateOrganizationRegionQuota} updateOrganizationRegionQuota
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    updateOrganizationRegionQuota(organizationId, regionId, updateOrganizationRegionQuota, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).updateOrganizationRegionQuota(organizationId, regionId, updateOrganizationRegionQuota, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Update organization role
     * @param {string} organizationId Organization ID
     * @param {string} roleId Role ID
     * @param {UpdateOrganizationRole} updateOrganizationRole
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    updateOrganizationRole(organizationId, roleId, updateOrganizationRole, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).updateOrganizationRole(organizationId, roleId, updateOrganizationRole, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Update region configuration
     * @param {string} id Region ID
     * @param {UpdateRegion} updateRegion
     * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    updateRegion(id, updateRegion, xDaytonaOrganizationID, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).updateRegion(id, updateRegion, xDaytonaOrganizationID, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Update sandbox default limited network egress
     * @param {string} organizationId Organization ID
     * @param {OrganizationSandboxDefaultLimitedNetworkEgress} organizationSandboxDefaultLimitedNetworkEgress
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    updateSandboxDefaultLimitedNetworkEgress(organizationId, organizationSandboxDefaultLimitedNetworkEgress, options) {
        return (0, exports.OrganizationsApiFp)(this.configuration).updateSandboxDefaultLimitedNetworkEgress(organizationId, organizationSandboxDefaultLimitedNetworkEgress, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.OrganizationsApi = OrganizationsApi;
