// @daytona/api-client/src/api/runners-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Daytona
 * Daytona AI platform API Docs
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@daytona.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RunnersApi = exports.RunnersApiFactory = exports.RunnersApiFp = exports.RunnersApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * RunnersApi - axios parameter creator
 */
const RunnersApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @summary Create runner
         * @param {CreateRunner} createRunner
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createRunner: async (createRunner, xDaytonaOrganizationID, options = {}) => {
            // verify required parameter 'createRunner' is not null or undefined
            (0, common_1.assertParamExists)('createRunner', 'createRunner', createRunner);
            const localVarPath = `/runners`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            localVarHeaderParameter['Accept'] = 'application/json';
            if (xDaytonaOrganizationID != null) {
                localVarHeaderParameter['X-Daytona-Organization-ID'] = String(xDaytonaOrganizationID);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(createRunner, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Delete runner
         * @param {string} id Runner ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteRunner: async (id, xDaytonaOrganizationID, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('deleteRunner', 'id', id);
            const localVarPath = `/runners/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'DELETE', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            if (xDaytonaOrganizationID != null) {
                localVarHeaderParameter['X-Daytona-Organization-ID'] = String(xDaytonaOrganizationID);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Get info for authenticated runner
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getInfoForAuthenticatedRunner: async (options = {}) => {
            const localVarPath = `/runners/me`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Get runner by ID
         * @param {string} id Runner ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getRunnerById: async (id, xDaytonaOrganizationID, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('getRunnerById', 'id', id);
            const localVarPath = `/runners/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            if (xDaytonaOrganizationID != null) {
                localVarHeaderParameter['X-Daytona-Organization-ID'] = String(xDaytonaOrganizationID);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Get runner by sandbox ID
         * @param {string} sandboxId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getRunnerBySandboxId: async (sandboxId, options = {}) => {
            // verify required parameter 'sandboxId' is not null or undefined
            (0, common_1.assertParamExists)('getRunnerBySandboxId', 'sandboxId', sandboxId);
            const localVarPath = `/runners/by-sandbox/{sandboxId}`
                .replace(`{${"sandboxId"}}`, encodeURIComponent(String(sandboxId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Get runner by ID
         * @param {string} id Runner ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getRunnerFullById: async (id, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('getRunnerFullById', 'id', id);
            const localVarPath = `/runners/{id}/full`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Get runners by snapshot ref
         * @param {string} ref Snapshot ref
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getRunnersBySnapshotRef: async (ref, options = {}) => {
            // verify required parameter 'ref' is not null or undefined
            (0, common_1.assertParamExists)('getRunnersBySnapshotRef', 'ref', ref);
            const localVarPath = `/runners/by-snapshot-ref`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            if (ref !== undefined) {
                localVarQueryParameter['ref'] = ref;
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary List all runners
         * @param {string} [regionId] Filter runners by region ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listRunners: async (regionId, xDaytonaOrganizationID, options = {}) => {
            const localVarPath = `/runners`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'GET', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            if (regionId !== undefined) {
                localVarQueryParameter['regionId'] = regionId;
            }
            localVarHeaderParameter['Accept'] = 'application/json';
            if (xDaytonaOrganizationID != null) {
                localVarHeaderParameter['X-Daytona-Organization-ID'] = String(xDaytonaOrganizationID);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Endpoint for version 2 runners to send healthcheck and metrics. Updates lastChecked timestamp and runner metrics.
         * @summary Runner healthcheck
         * @param {RunnerHealthcheck} runnerHealthcheck
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        runnerHealthcheck: async (runnerHealthcheck, options = {}) => {
            // verify required parameter 'runnerHealthcheck' is not null or undefined
            (0, common_1.assertParamExists)('runnerHealthcheck', 'runnerHealthcheck', runnerHealthcheck);
            const localVarPath = `/runners/healthcheck`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'POST', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(runnerHealthcheck, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Update runner draining status
         * @param {string} id Runner ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateRunnerDraining: async (id, xDaytonaOrganizationID, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('updateRunnerDraining', 'id', id);
            const localVarPath = `/runners/{id}/draining`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'PATCH', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            if (xDaytonaOrganizationID != null) {
                localVarHeaderParameter['X-Daytona-Organization-ID'] = String(xDaytonaOrganizationID);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         *
         * @summary Update runner scheduling status
         * @param {string} id Runner ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateRunnerScheduling: async (id, xDaytonaOrganizationID, options = {}) => {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('updateRunnerScheduling', 'id', id);
            const localVarPath = `/runners/{id}/scheduling`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = { method: 'PATCH', ...baseOptions, ...options };
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            await (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication oauth2 required
            localVarHeaderParameter['Accept'] = 'application/json';
            if (xDaytonaOrganizationID != null) {
                localVarHeaderParameter['X-Daytona-Organization-ID'] = String(xDaytonaOrganizationID);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = { ...localVarHeaderParameter, ...headersFromBaseOptions, ...options.headers };
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.RunnersApiAxiosParamCreator = RunnersApiAxiosParamCreator;
/**
 * RunnersApi - functional programming interface
 */
const RunnersApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.RunnersApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @summary Create runner
         * @param {CreateRunner} createRunner
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async createRunner(createRunner, xDaytonaOrganizationID, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.createRunner(createRunner, xDaytonaOrganizationID, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['RunnersApi.createRunner']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Delete runner
         * @param {string} id Runner ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async deleteRunner(id, xDaytonaOrganizationID, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.deleteRunner(id, xDaytonaOrganizationID, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['RunnersApi.deleteRunner']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Get info for authenticated runner
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getInfoForAuthenticatedRunner(options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getInfoForAuthenticatedRunner(options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['RunnersApi.getInfoForAuthenticatedRunner']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Get runner by ID
         * @param {string} id Runner ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getRunnerById(id, xDaytonaOrganizationID, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getRunnerById(id, xDaytonaOrganizationID, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['RunnersApi.getRunnerById']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Get runner by sandbox ID
         * @param {string} sandboxId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getRunnerBySandboxId(sandboxId, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getRunnerBySandboxId(sandboxId, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['RunnersApi.getRunnerBySandboxId']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Get runner by ID
         * @param {string} id Runner ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getRunnerFullById(id, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getRunnerFullById(id, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['RunnersApi.getRunnerFullById']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Get runners by snapshot ref
         * @param {string} ref Snapshot ref
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async getRunnersBySnapshotRef(ref, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.getRunnersBySnapshotRef(ref, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['RunnersApi.getRunnersBySnapshotRef']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary List all runners
         * @param {string} [regionId] Filter runners by region ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async listRunners(regionId, xDaytonaOrganizationID, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.listRunners(regionId, xDaytonaOrganizationID, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['RunnersApi.listRunners']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         * Endpoint for version 2 runners to send healthcheck and metrics. Updates lastChecked timestamp and runner metrics.
         * @summary Runner healthcheck
         * @param {RunnerHealthcheck} runnerHealthcheck
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async runnerHealthcheck(runnerHealthcheck, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.runnerHealthcheck(runnerHealthcheck, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['RunnersApi.runnerHealthcheck']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Update runner draining status
         * @param {string} id Runner ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async updateRunnerDraining(id, xDaytonaOrganizationID, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.updateRunnerDraining(id, xDaytonaOrganizationID, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['RunnersApi.updateRunnerDraining']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
        /**
         *
         * @summary Update runner scheduling status
         * @param {string} id Runner ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async updateRunnerScheduling(id, xDaytonaOrganizationID, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.updateRunnerScheduling(id, xDaytonaOrganizationID, options);
            const localVarOperationServerIndex = configuration?.serverIndex ?? 0;
            const localVarOperationServerBasePath = base_1.operationServerMap['RunnersApi.updateRunnerScheduling']?.[localVarOperationServerIndex]?.url;
            return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
        },
    };
};
exports.RunnersApiFp = RunnersApiFp;
/**
 * RunnersApi - factory interface
 */
const RunnersApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.RunnersApiFp)(configuration);
    return {
        /**
         *
         * @summary Create runner
         * @param {CreateRunner} createRunner
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createRunner(createRunner, xDaytonaOrganizationID, options) {
            return localVarFp.createRunner(createRunner, xDaytonaOrganizationID, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Delete runner
         * @param {string} id Runner ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteRunner(id, xDaytonaOrganizationID, options) {
            return localVarFp.deleteRunner(id, xDaytonaOrganizationID, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get info for authenticated runner
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getInfoForAuthenticatedRunner(options) {
            return localVarFp.getInfoForAuthenticatedRunner(options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get runner by ID
         * @param {string} id Runner ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getRunnerById(id, xDaytonaOrganizationID, options) {
            return localVarFp.getRunnerById(id, xDaytonaOrganizationID, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get runner by sandbox ID
         * @param {string} sandboxId
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getRunnerBySandboxId(sandboxId, options) {
            return localVarFp.getRunnerBySandboxId(sandboxId, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get runner by ID
         * @param {string} id Runner ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getRunnerFullById(id, options) {
            return localVarFp.getRunnerFullById(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get runners by snapshot ref
         * @param {string} ref Snapshot ref
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getRunnersBySnapshotRef(ref, options) {
            return localVarFp.getRunnersBySnapshotRef(ref, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary List all runners
         * @param {string} [regionId] Filter runners by region ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listRunners(regionId, xDaytonaOrganizationID, options) {
            return localVarFp.listRunners(regionId, xDaytonaOrganizationID, options).then((request) => request(axios, basePath));
        },
        /**
         * Endpoint for version 2 runners to send healthcheck and metrics. Updates lastChecked timestamp and runner metrics.
         * @summary Runner healthcheck
         * @param {RunnerHealthcheck} runnerHealthcheck
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        runnerHealthcheck(runnerHealthcheck, options) {
            return localVarFp.runnerHealthcheck(runnerHealthcheck, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Update runner draining status
         * @param {string} id Runner ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateRunnerDraining(id, xDaytonaOrganizationID, options) {
            return localVarFp.updateRunnerDraining(id, xDaytonaOrganizationID, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Update runner scheduling status
         * @param {string} id Runner ID
         * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        updateRunnerScheduling(id, xDaytonaOrganizationID, options) {
            return localVarFp.updateRunnerScheduling(id, xDaytonaOrganizationID, options).then((request) => request(axios, basePath));
        },
    };
};
exports.RunnersApiFactory = RunnersApiFactory;
/**
 * RunnersApi - object-oriented interface
 */
class RunnersApi extends base_1.BaseAPI {
    /**
     *
     * @summary Create runner
     * @param {CreateRunner} createRunner
     * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    createRunner(createRunner, xDaytonaOrganizationID, options) {
        return (0, exports.RunnersApiFp)(this.configuration).createRunner(createRunner, xDaytonaOrganizationID, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Delete runner
     * @param {string} id Runner ID
     * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    deleteRunner(id, xDaytonaOrganizationID, options) {
        return (0, exports.RunnersApiFp)(this.configuration).deleteRunner(id, xDaytonaOrganizationID, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get info for authenticated runner
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getInfoForAuthenticatedRunner(options) {
        return (0, exports.RunnersApiFp)(this.configuration).getInfoForAuthenticatedRunner(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get runner by ID
     * @param {string} id Runner ID
     * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getRunnerById(id, xDaytonaOrganizationID, options) {
        return (0, exports.RunnersApiFp)(this.configuration).getRunnerById(id, xDaytonaOrganizationID, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get runner by sandbox ID
     * @param {string} sandboxId
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getRunnerBySandboxId(sandboxId, options) {
        return (0, exports.RunnersApiFp)(this.configuration).getRunnerBySandboxId(sandboxId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get runner by ID
     * @param {string} id Runner ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getRunnerFullById(id, options) {
        return (0, exports.RunnersApiFp)(this.configuration).getRunnerFullById(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get runners by snapshot ref
     * @param {string} ref Snapshot ref
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getRunnersBySnapshotRef(ref, options) {
        return (0, exports.RunnersApiFp)(this.configuration).getRunnersBySnapshotRef(ref, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary List all runners
     * @param {string} [regionId] Filter runners by region ID
     * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    listRunners(regionId, xDaytonaOrganizationID, options) {
        return (0, exports.RunnersApiFp)(this.configuration).listRunners(regionId, xDaytonaOrganizationID, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Endpoint for version 2 runners to send healthcheck and metrics. Updates lastChecked timestamp and runner metrics.
     * @summary Runner healthcheck
     * @param {RunnerHealthcheck} runnerHealthcheck
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    runnerHealthcheck(runnerHealthcheck, options) {
        return (0, exports.RunnersApiFp)(this.configuration).runnerHealthcheck(runnerHealthcheck, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Update runner draining status
     * @param {string} id Runner ID
     * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    updateRunnerDraining(id, xDaytonaOrganizationID, options) {
        return (0, exports.RunnersApiFp)(this.configuration).updateRunnerDraining(id, xDaytonaOrganizationID, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Update runner scheduling status
     * @param {string} id Runner ID
     * @param {string} [xDaytonaOrganizationID] Use with JWT to specify the organization ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    updateRunnerScheduling(id, xDaytonaOrganizationID, options) {
        return (0, exports.RunnersApiFp)(this.configuration).updateRunnerScheduling(id, xDaytonaOrganizationID, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.RunnersApi = RunnersApi;
