// @daytona/sdk/cjs/errors/DaytonaError.js
"use strict";
/*
 * Copyright Daytona Platforms Inc.
 * SPDX-License-Identifier: Apache-2.0
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DaytonaConnectionError = exports.DaytonaTimeoutError = exports.DaytonaValidationError = exports.DaytonaConflictError = exports.DaytonaAuthorizationError = exports.DaytonaAuthenticationError = exports.DaytonaRateLimitError = exports.DaytonaNotFoundError = exports.DaytonaError = void 0;
exports.errorClassFromStatusCode = errorClassFromStatusCode;
exports.createDaytonaError = createDaytonaError;
exports.createAxiosDaytonaError = createAxiosDaytonaError;
/**
 * Base error for Daytona SDK.
 *
 * @example
 * ```ts
 * try {
 *   await daytona.get('missing-sandbox')
 * } catch (error) {
 *   if (error instanceof DaytonaError) {
 *     console.log(error.statusCode)
 *     console.log(error.errorCode)
 *     console.log(error.message)
 *   }
 * }
 * ```
 */
class DaytonaError extends Error {
    /** HTTP status code if available */
    statusCode;
    /** Machine-readable error code if available */
    errorCode;
    /** Response headers if available */
    headers;
    constructor(message, statusCode, headers, errorCode) {
        super(message);
        this.name = new.target.name;
        this.statusCode = statusCode;
        this.headers = headers;
        this.errorCode = errorCode;
    }
}
exports.DaytonaError = DaytonaError;
/**
 * Error thrown when a resource is not found (HTTP 404).
 *
 * @example
 * ```ts
 * try {
 *   await sandbox.fs.downloadFile('/workspace/missing.txt')
 * } catch (error) {
 *   if (error instanceof DaytonaNotFoundError) {
 *     console.log(error.statusCode)
 *   }
 * }
 * ```
 */
class DaytonaNotFoundError extends DaytonaError {
}
exports.DaytonaNotFoundError = DaytonaNotFoundError;
/**
 * Error thrown when rate limit is exceeded.
 *
 * @example
 * ```ts
 * try {
 *   for await (const sandbox of daytona.list()) {
 *     console.log(sandbox.id)
 *   }
 * } catch (error) {
 *   if (error instanceof DaytonaRateLimitError) {
 *     console.log(error.errorCode)
 *   }
 * }
 * ```
 */
class DaytonaRateLimitError extends DaytonaError {
}
exports.DaytonaRateLimitError = DaytonaRateLimitError;
/**
 * Error thrown when authentication fails (HTTP 401).
 *
 * @example
 * ```ts
 * try {
 *   for await (const sandbox of daytona.list()) {
 *     console.log(sandbox.id)
 *   }
 * } catch (error) {
 *   if (error instanceof DaytonaAuthenticationError) {
 *     console.log(error.statusCode)
 *   }
 * }
 * ```
 */
class DaytonaAuthenticationError extends DaytonaError {
}
exports.DaytonaAuthenticationError = DaytonaAuthenticationError;
/**
 * Error thrown when the request is forbidden (HTTP 403).
 *
 * @example
 * ```ts
 * try {
 *   await daytona.get('sandbox-without-access')
 * } catch (error) {
 *   if (error instanceof DaytonaAuthorizationError) {
 *     console.log(error.message)
 *   }
 * }
 * ```
 */
class DaytonaAuthorizationError extends DaytonaError {
}
exports.DaytonaAuthorizationError = DaytonaAuthorizationError;
/**
 * Error thrown when a resource conflict occurs (HTTP 409).
 *
 * @example
 * ```ts
 * try {
 *   await daytona.create({ name: 'existing-sandbox' })
 * } catch (error) {
 *   if (error instanceof DaytonaConflictError) {
 *     console.log(error.errorCode)
 *   }
 * }
 * ```
 */
class DaytonaConflictError extends DaytonaError {
}
exports.DaytonaConflictError = DaytonaConflictError;
/**
 * Error thrown when input validation fails (HTTP 400 or client-side validation).
 *
 * @example
 * ```ts
 * try {
 *   Image.debianSlim('3.8' as never)
 * } catch (error) {
 *   if (error instanceof DaytonaValidationError) {
 *     console.log(error.message)
 *   }
 * }
 * ```
 */
class DaytonaValidationError extends DaytonaError {
}
exports.DaytonaValidationError = DaytonaValidationError;
/**
 * Error thrown when a timeout occurs.
 *
 * @example
 * ```ts
 * try {
 *   await sandbox.waitUntilStarted(1)
 * } catch (error) {
 *   if (error instanceof DaytonaTimeoutError) {
 *     console.log(error.message)
 *   }
 * }
 * ```
 */
class DaytonaTimeoutError extends DaytonaError {
}
exports.DaytonaTimeoutError = DaytonaTimeoutError;
/**
 * Error thrown when a network connection fails.
 *
 * @example
 * ```ts
 * try {
 *   await ptyHandle.waitForConnection()
 * } catch (error) {
 *   if (error instanceof DaytonaConnectionError) {
 *     console.log(error.message)
 *   }
 * }
 * ```
 */
class DaytonaConnectionError extends DaytonaError {
}
exports.DaytonaConnectionError = DaytonaConnectionError;
const STATUS_CODE_TO_ERROR = {
    400: DaytonaValidationError,
    401: DaytonaAuthenticationError,
    403: DaytonaAuthorizationError,
    404: DaytonaNotFoundError,
    409: DaytonaConflictError,
    429: DaytonaRateLimitError,
};
/**
 * Maps an HTTP status code to the corresponding Daytona error class.
 */
function errorClassFromStatusCode(statusCode) {
    if (statusCode === undefined) {
        return DaytonaError;
    }
    return STATUS_CODE_TO_ERROR[statusCode] || DaytonaError;
}
/**
 * Creates the appropriate Daytona error subclass from structured error metadata.
 */
function createDaytonaError(message, statusCode, headers, errorCode) {
    const ErrorClass = errorClassFromStatusCode(statusCode);
    return new ErrorClass(message, statusCode, headers, errorCode);
}
function isAxiosTimeoutError(error) {
    return error.code === 'ECONNABORTED' || error.code === 'ETIMEDOUT' || error.message.includes('timeout of');
}
function getAxiosResponseDataObject(error) {
    if (!error.response?.data || typeof error.response.data !== 'object') {
        return undefined;
    }
    return error.response.data;
}
function extractAxiosErrorCode(responseData) {
    if (typeof responseData?.code === 'string') {
        return responseData.code;
    }
    if (typeof responseData?.error_code === 'string') {
        return responseData.error_code;
    }
    if (typeof responseData?.error === 'string') {
        return responseData.error;
    }
    return undefined;
}
function extractAxiosErrorMessage(error) {
    if (isAxiosTimeoutError(error)) {
        return 'Operation timed out';
    }
    const responseData = getAxiosResponseDataObject(error);
    const responseMessage = responseData?.message || error.response?.data;
    const message = responseMessage || error.message || String(error);
    if (typeof message === 'object') {
        try {
            return JSON.stringify(message);
        }
        catch {
            return String(message);
        }
    }
    return String(message);
}
/**
 * Creates the appropriate Daytona error subclass from an Axios error.
 */
function createAxiosDaytonaError(error) {
    const message = extractAxiosErrorMessage(error);
    const statusCode = error.response?.status;
    const headers = error.response?.headers;
    const responseData = getAxiosResponseDataObject(error);
    const errorCode = extractAxiosErrorCode(responseData);
    if (isAxiosTimeoutError(error)) {
        return new DaytonaTimeoutError(message, statusCode, headers, errorCode);
    }
    if (!error.response && (error.request || error.code)) {
        return new DaytonaConnectionError(message, statusCode, headers, errorCode);
    }
    return createDaytonaError(message, statusCode, headers, errorCode);
}
