// @daytona/sdk/esm/ObjectStorage.js
/*
 * Copyright Daytona Platforms Inc.
 * SPDX-License-Identifier: Apache-2.0
 */
import { __esDecorate, __runInitializers } from "tslib";
import { ListObjectsV2Command, S3Client } from '@aws-sdk/client-s3';
import { Upload } from '@aws-sdk/lib-storage';
import * as crypto from 'crypto';
import * as pathe from 'pathe';
import { DaytonaNotFoundError } from './errors/DaytonaError.js';
import { dynamicImport } from './utils/Import.js';
import { WithInstrumentation } from './utils/otel.decorator.js';
/**
 * ObjectStorage class for interacting with object storage services.
 *
 * @class
 * @param {ObjectStorageConfig} config - The configuration for the object storage service.
 */
let ObjectStorage = (() => {
    let _instanceExtraInitializers = [];
    let _upload_decorators;
    return class ObjectStorage {
        static {
            const _metadata = typeof Symbol === "function" && Symbol.metadata ? Object.create(null) : void 0;
            _upload_decorators = [WithInstrumentation()];
            __esDecorate(this, null, _upload_decorators, { kind: "method", name: "upload", static: false, private: false, access: { has: obj => "upload" in obj, get: obj => obj.upload }, metadata: _metadata }, null, _instanceExtraInitializers);
            if (_metadata) Object.defineProperty(this, Symbol.metadata, { enumerable: true, configurable: true, writable: true, value: _metadata });
        }
        bucketName = __runInitializers(this, _instanceExtraInitializers);
        s3Client;
        constructor(config) {
            this.bucketName = config.bucketName || 'daytona-volume-builds';
            this.s3Client = new S3Client({
                region: this.extractAwsRegion(config.endpointUrl) || 'us-east-1',
                endpoint: config.endpointUrl,
                credentials: {
                    accessKeyId: config.accessKeyId,
                    secretAccessKey: config.secretAccessKey,
                    sessionToken: config.sessionToken,
                },
                forcePathStyle: true,
            });
        }
        /**
         * Upload a file or directory to object storage.
         *
         * @param {string} path - The path to the file or directory to upload.
         * @param {string} organizationId - The organization ID to use for the upload.
         * @param {string} archiveBasePath - The base path to use for the archive.
         * @returns {Promise<string>} The hash of the uploaded file or directory.
         */
        async upload(path, organizationId, archiveBasePath) {
            const fs = await dynamicImport('fs', '"upload" is not supported: ');
            if (!fs.existsSync(path)) {
                const errMsg = `Path does not exist: ${path}`;
                throw new DaytonaNotFoundError(errMsg);
            }
            // Compute hash for the path
            const pathHash = await this.computeHashForPathMd5(path, archiveBasePath);
            // Define the S3 prefix
            const prefix = `${organizationId}/${pathHash}/`;
            const s3Key = `${prefix}context.tar`;
            // Check if it already exists in S3
            if (await this.folderExistsInS3(prefix)) {
                return pathHash;
            }
            // Upload to S3
            await this.uploadAsTar(s3Key, path, archiveBasePath);
            return pathHash;
        }
        /**
         * Compute a hash for a file or directory.
         *
         * @param {string} pathStr - The path to the file or directory to hash.
         * @param {string} archiveBasePath - The base path to use for the archive.
         * @returns {Promise<string>} The hash of the file or directory.
         */
        async computeHashForPathMd5(pathStr, archiveBasePath) {
            const fs = await dynamicImport('fs', '"computeHashForPathMd5" is not supported: ');
            const md5Hasher = crypto.createHash('md5');
            const absPathStr = pathe.resolve(pathStr);
            md5Hasher.update(archiveBasePath);
            if (fs.statSync(absPathStr).isFile()) {
                // For files, hash the content
                await this.hashFile(absPathStr, md5Hasher);
            }
            else {
                // For directories, recursively hash all files and their paths
                await this.hashDirectory(absPathStr, pathStr, md5Hasher);
            }
            return md5Hasher.digest('hex');
        }
        /**
         * Recursively hash a directory and its contents.
         *
         * @param {string} dirPath - The path to the directory to hash.
         * @param {string} basePath - The base path to use for the hash.
         * @param {crypto.Hash} hasher - The hasher to use for the hash.
         * @returns {Promise<void>} A promise that resolves when the directory has been hashed.
         */
        async hashDirectory(dirPath, basePath, hasher) {
            const fs = await dynamicImport('fs', '"hashDirectory" is not supported: ');
            const entries = fs.readdirSync(dirPath, { withFileTypes: true });
            const hasSubdirs = entries.some((e) => e.isDirectory());
            const hasFiles = entries.some((e) => e.isFile());
            if (!hasSubdirs && !hasFiles) {
                // Empty directory
                const relDir = pathe.relative(basePath, dirPath);
                hasher.update(relDir);
            }
            for (const entry of entries) {
                const fullPath = pathe.join(dirPath, entry.name);
                if (entry.isDirectory()) {
                    await this.hashDirectory(fullPath, basePath, hasher);
                }
                else if (entry.isFile()) {
                    const relPath = pathe.relative(basePath, fullPath);
                    hasher.update(relPath);
                    await this.hashFile(fullPath, hasher);
                }
            }
        }
        /**
         * Hash a file.
         *
         * @param {string} filePath - The path to the file to hash.
         * @param {crypto.Hash} hasher - The hasher to use for the hash.
         * @returns {Promise<void>} A promise that resolves when the file has been hashed.
         */
        async hashFile(filePath, hasher) {
            const fs = await dynamicImport('fs', '"hashFile" is not supported: ');
            await new Promise((resolve, reject) => {
                const stream = fs.createReadStream(filePath, { highWaterMark: 8192 });
                stream.on('data', (chunk) => hasher.update(chunk));
                stream.on('end', resolve);
                stream.on('error', reject);
            });
        }
        /**
         * Check if a prefix (folder) exists in S3.
         *
         * @param {string} prefix - The prefix to check.
         * @returns {Promise<boolean>} True if the prefix exists, false otherwise.
         */
        async folderExistsInS3(prefix) {
            const response = await this.s3Client.send(new ListObjectsV2Command({
                Bucket: this.bucketName,
                Prefix: prefix,
                MaxKeys: 1,
            }));
            return !!response.Contents && response.Contents.length > 0;
        }
        /**
         * Create a tar archive of the specified path and upload it to S3.
         *
         * @param {string} s3Key - The key to use for the uploaded file.
         * @param {string} sourcePath - The path to the file or directory to upload.
         * @param {string} archiveBasePath - The base path to use for the archive.
         */
        async uploadAsTar(s3Key, sourcePath, archiveBasePath) {
            const importErrorPrefix = '"uploadAsTar" is not supported: ';
            const tar = await dynamicImport('tar', importErrorPrefix);
            const stream = await dynamicImport('stream', importErrorPrefix);
            sourcePath = pathe.resolve(sourcePath);
            const normalizedSourcePath = pathe.normalize(sourcePath);
            const normalizedArchiveBasePath = pathe.normalize(archiveBasePath);
            let basePrefix;
            if (normalizedArchiveBasePath === '.') {
                // When archiveBasePath is empty (normalized to '.'), use the normalizedSourcePath as cwd and the '.' as target
                basePrefix = normalizedSourcePath;
            }
            else {
                // Normal case: extract the base prefix by removing archiveBasePath from the end
                basePrefix = normalizedSourcePath.slice(0, normalizedSourcePath.length - normalizedArchiveBasePath.length);
            }
            const tarStream = tar.create({
                cwd: basePrefix,
                portable: true,
                gzip: false,
            }, [normalizedArchiveBasePath]);
            const pass = new stream.PassThrough();
            tarStream.pipe(pass);
            const uploader = new Upload({
                client: this.s3Client,
                params: {
                    Bucket: this.bucketName,
                    Key: s3Key,
                    Body: pass,
                },
            });
            await uploader.done();
        }
        extractAwsRegion(endpoint) {
            const match = endpoint.match(/s3[.-]([a-z0-9-]+)\.amazonaws\.com/);
            return match?.[1];
        }
    };
})();
export { ObjectStorage };
