// @daytona/sdk/esm/Secret.js
/*
 * Copyright Daytona Platforms Inc.
 * SPDX-License-Identifier: Apache-2.0
 */
import { __esDecorate, __runInitializers } from "tslib";
import { WithInstrumentation } from './utils/otel.decorator.js';
/**
 * Service for managing organization-scoped Daytona Secrets.
 *
 * This service provides methods to create, list, get, update, and delete Secrets. Secrets can be
 * mounted into Sandboxes as environment variables by referencing them via the `secrets` field on
 * the create-sandbox parameters. The Sandbox only ever sees the Secret's opaque placeholder; the
 * real value is substituted at the network egress layer for the Secret's allowed hosts.
 *
 * @class
 */
let SecretService = (() => {
    let _instanceExtraInitializers = [];
    let _list_decorators;
    let _get_decorators;
    let _create_decorators;
    let _update_decorators;
    let _delete_decorators;
    return class SecretService {
        static {
            const _metadata = typeof Symbol === "function" && Symbol.metadata ? Object.create(null) : void 0;
            _list_decorators = [WithInstrumentation()];
            _get_decorators = [WithInstrumentation()];
            _create_decorators = [WithInstrumentation()];
            _update_decorators = [WithInstrumentation()];
            _delete_decorators = [WithInstrumentation()];
            __esDecorate(this, null, _list_decorators, { kind: "method", name: "list", static: false, private: false, access: { has: obj => "list" in obj, get: obj => obj.list }, metadata: _metadata }, null, _instanceExtraInitializers);
            __esDecorate(this, null, _get_decorators, { kind: "method", name: "get", static: false, private: false, access: { has: obj => "get" in obj, get: obj => obj.get }, metadata: _metadata }, null, _instanceExtraInitializers);
            __esDecorate(this, null, _create_decorators, { kind: "method", name: "create", static: false, private: false, access: { has: obj => "create" in obj, get: obj => obj.create }, metadata: _metadata }, null, _instanceExtraInitializers);
            __esDecorate(this, null, _update_decorators, { kind: "method", name: "update", static: false, private: false, access: { has: obj => "update" in obj, get: obj => obj.update }, metadata: _metadata }, null, _instanceExtraInitializers);
            __esDecorate(this, null, _delete_decorators, { kind: "method", name: "delete", static: false, private: false, access: { has: obj => "delete" in obj, get: obj => obj.delete }, metadata: _metadata }, null, _instanceExtraInitializers);
            if (_metadata) Object.defineProperty(this, Symbol.metadata, { enumerable: true, configurable: true, writable: true, value: _metadata });
        }
        secretApi = __runInitializers(this, _instanceExtraInitializers);
        constructor(secretApi) {
            this.secretApi = secretApi;
        }
        /**
         * Lists all Secrets in the organization.
         *
         * @returns {Promise<Secret[]>} List of all Secrets in the organization
         *
         * @example
         * const daytona = new Daytona();
         * const secrets = await daytona.secret.list();
         * console.log(`Found ${secrets.length} secrets`);
         * secrets.forEach(secret => console.log(`${secret.name} (${secret.id})`));
         */
        async list() {
            const response = await this.secretApi.listSecrets();
            return response.data;
        }
        /**
         * Gets a Secret by its ID.
         *
         * @param {string} secretId - ID of the Secret to retrieve
         * @returns {Promise<Secret>} The requested Secret
         * @throws {DaytonaNotFoundError} If the Secret does not exist
         *
         * @example
         * const daytona = new Daytona();
         * const secret = await daytona.secret.get("secret-id");
         * console.log(`Secret ${secret.name} can be used on ${secret.hosts.join(', ')}`);
         */
        async get(secretId) {
            const response = await this.secretApi.getSecret(secretId);
            return response.data;
        }
        /**
         * Creates a new Secret.
         *
         * @param {CreateSecretParams} params - Parameters for the new Secret
         * @returns {Promise<Secret>} The newly created Secret (without the plaintext `value`)
         * @throws {DaytonaConflictError} If a Secret with the same name already exists in the organization
         *
         * @example
         * const daytona = new Daytona();
         * const secret = await daytona.secret.create({
         *   name: "anthropic-prod",
         *   value: "sk-ant-...",
         *   hosts: ["api.anthropic.com"],
         * });
         * console.log(`Created secret ${secret.name} with placeholder ${secret.placeholder}`);
         */
        async create(params) {
            const response = await this.secretApi.createSecret({
                name: params.name,
                value: params.value,
                description: params.description,
                hosts: params.hosts,
            });
            return response.data;
        }
        /**
         * Updates an existing Secret. Omitted fields are left unchanged.
         *
         * @param {string} secretId - ID of the Secret to update
         * @param {UpdateSecretParams} params - Fields to update
         * @returns {Promise<Secret>} The updated Secret
         * @throws {DaytonaNotFoundError} If the Secret does not exist
         *
         * @example
         * const daytona = new Daytona();
         * const secret = await daytona.secret.update("secret-id", {
         *   value: "sk-ant-new-value",
         *   hosts: ["api.anthropic.com", "*.anthropic.com"],
         * });
         */
        async update(secretId, params) {
            const response = await this.secretApi.updateSecret(secretId, {
                value: params.value,
                description: params.description,
                hosts: params.hosts,
            });
            return response.data;
        }
        /**
         * Deletes a Secret.
         *
         * @param {string} secretId - ID of the Secret to delete
         * @returns {Promise<void>}
         * @throws {DaytonaNotFoundError} If the Secret does not exist
         *
         * @example
         * const daytona = new Daytona();
         * await daytona.secret.delete("secret-id");
         * console.log("Secret deleted successfully");
         */
        async delete(secretId) {
            await this.secretApi.deleteSecret(secretId);
        }
    };
})();
export { SecretService };
