// @daytona/sdk/esm/Snapshot.js
/*
 * Copyright Daytona Platforms Inc.
 * SPDX-License-Identifier: Apache-2.0
 */
import { __esDecorate, __runInitializers } from "tslib";
import { SnapshotState } from '@daytona/api-client';
import { DaytonaError } from './errors/DaytonaError.js';
import { processStreamingResponse } from './utils/Stream.js';
import { dynamicImport } from './utils/Import.js';
import { WithInstrumentation } from './utils/otel.decorator.js';
/**
 * Service for managing Daytona Snapshots. Can be used to list, get, create and delete Snapshots.
 *
 * @class
 */
let SnapshotService = (() => {
    let _staticExtraInitializers = [];
    let _instanceExtraInitializers = [];
    let _static_processImageContext_decorators;
    let _list_decorators;
    let _get_decorators;
    let _delete_decorators;
    let _create_decorators;
    let _activate_decorators;
    return class SnapshotService {
        static {
            const _metadata = typeof Symbol === "function" && Symbol.metadata ? Object.create(null) : void 0;
            _list_decorators = [WithInstrumentation()];
            _get_decorators = [WithInstrumentation()];
            _delete_decorators = [WithInstrumentation()];
            _create_decorators = [WithInstrumentation()];
            _activate_decorators = [WithInstrumentation()];
            _static_processImageContext_decorators = [WithInstrumentation()];
            __esDecorate(this, null, _static_processImageContext_decorators, { kind: "method", name: "processImageContext", static: true, private: false, access: { has: obj => "processImageContext" in obj, get: obj => obj.processImageContext }, metadata: _metadata }, null, _staticExtraInitializers);
            __esDecorate(this, null, _list_decorators, { kind: "method", name: "list", static: false, private: false, access: { has: obj => "list" in obj, get: obj => obj.list }, metadata: _metadata }, null, _instanceExtraInitializers);
            __esDecorate(this, null, _get_decorators, { kind: "method", name: "get", static: false, private: false, access: { has: obj => "get" in obj, get: obj => obj.get }, metadata: _metadata }, null, _instanceExtraInitializers);
            __esDecorate(this, null, _delete_decorators, { kind: "method", name: "delete", static: false, private: false, access: { has: obj => "delete" in obj, get: obj => obj.delete }, metadata: _metadata }, null, _instanceExtraInitializers);
            __esDecorate(this, null, _create_decorators, { kind: "method", name: "create", static: false, private: false, access: { has: obj => "create" in obj, get: obj => obj.create }, metadata: _metadata }, null, _instanceExtraInitializers);
            __esDecorate(this, null, _activate_decorators, { kind: "method", name: "activate", static: false, private: false, access: { has: obj => "activate" in obj, get: obj => obj.activate }, metadata: _metadata }, null, _instanceExtraInitializers);
            if (_metadata) Object.defineProperty(this, Symbol.metadata, { enumerable: true, configurable: true, writable: true, value: _metadata });
            __runInitializers(this, _staticExtraInitializers);
        }
        clientConfig = __runInitializers(this, _instanceExtraInitializers);
        snapshotsApi;
        objectStorageApi;
        defaultRegionId;
        constructor(clientConfig, snapshotsApi, objectStorageApi, defaultRegionId) {
            this.clientConfig = clientConfig;
            this.snapshotsApi = snapshotsApi;
            this.objectStorageApi = objectStorageApi;
            this.defaultRegionId = defaultRegionId;
        }
        /**
         * List paginated list of Snapshots.
         *
         * @param {number} [page] - Page number for pagination (starting from 1)
         * @param {number} [limit] - Maximum number of items per page
         * @returns {Promise<PaginatedSnapshots>} Paginated list of Snapshots
         *
         * @example
         * const daytona = new Daytona();
         * const result = await daytona.snapshot.list(2, 10);
         * console.log(`Found ${result.total} snapshots`);
         * result.items.forEach(snapshot => console.log(`${snapshot.name} (${snapshot.imageName})`));
         */
        async list(page, limit) {
            const response = await this.snapshotsApi.getAllSnapshots(undefined, page, limit);
            return {
                items: response.data.items.map((snapshot) => snapshot),
                total: response.data.total,
                page: response.data.page,
                totalPages: response.data.totalPages,
            };
        }
        /**
         * Gets a Snapshot by its name.
         *
         * @param {string} name - Name of the Snapshot to retrieve
         * @returns {Promise<Snapshot>} The requested Snapshot
         * @throws {Error} If the Snapshot does not exist or cannot be accessed
         *
         * @example
         * const daytona = new Daytona();
         * const snapshot = await daytona.snapshot.get("snapshot-name");
         * console.log(`Snapshot ${snapshot.name} is in state ${snapshot.state}`);
         */
        async get(name) {
            const response = await this.snapshotsApi.getSnapshot(name);
            return response.data;
        }
        /**
         * Deletes a Snapshot.
         *
         * @param {Snapshot} snapshot - Snapshot to delete
         * @returns {Promise<void>}
         * @throws {Error} If the Snapshot does not exist or cannot be deleted
         *
         * @example
         * const daytona = new Daytona();
         * const snapshot = await daytona.snapshot.get("snapshot-name");
         * await daytona.snapshot.delete(snapshot);
         * console.log("Snapshot deleted successfully");
         */
        async delete(snapshot) {
            await this.snapshotsApi.removeSnapshot(snapshot.id);
        }
        /**
         * Creates and registers a new snapshot from the given Image definition.
         *
         * @param {CreateSnapshotParams} params - Parameters for snapshot creation.
         * @param {object} options - Options for the create operation.
         * @param {boolean} options.onLogs - This callback function handles snapshot creation logs.
         * @param {number} options.timeout - Default is no timeout. Timeout in seconds (0 means no timeout).
         * @returns {Promise<void>}
         *
         * @example
         * const image = Image.debianSlim('3.12').pipInstall('numpy');
         * await daytona.snapshot.create({ name: 'my-snapshot', image: image }, { onLogs: console.log });
         */
        async create(params, options = {}) {
            const createSnapshotReq = {
                name: params.name,
            };
            if (typeof params.image === 'string') {
                createSnapshotReq.imageName = params.image;
                createSnapshotReq.entrypoint = params.entrypoint;
            }
            else {
                const contextHashes = await SnapshotService.processImageContext(this.objectStorageApi, params.image);
                createSnapshotReq.buildInfo = {
                    contextHashes,
                    dockerfileContent: params.entrypoint
                        ? params.image.entrypoint(params.entrypoint).dockerfile
                        : params.image.dockerfile,
                };
            }
            if (params.resources) {
                createSnapshotReq.cpu = params.resources.cpu;
                createSnapshotReq.gpu = params.resources.gpu;
                if (params.resources.gpuType !== undefined) {
                    createSnapshotReq.gpuType = Array.isArray(params.resources.gpuType)
                        ? params.resources.gpuType
                        : [params.resources.gpuType];
                }
                createSnapshotReq.memory = params.resources.memory;
                createSnapshotReq.disk = params.resources.disk;
            }
            createSnapshotReq.regionId = params.regionId || this.defaultRegionId;
            createSnapshotReq.sandboxClass = params.sandboxClass;
            let createdSnapshot = (await this.snapshotsApi.createSnapshot(createSnapshotReq, undefined, {
                timeout: (options.timeout || 0) * 1000,
            })).data;
            if (!createdSnapshot) {
                throw new DaytonaError("Failed to create snapshot. Didn't receive a snapshot from the server API.");
            }
            const terminalStates = [SnapshotState.ACTIVE, SnapshotState.ERROR, SnapshotState.BUILD_FAILED];
            const snapshotRef = { createdSnapshot: createdSnapshot };
            let streamPromise;
            // eslint-disable-next-line @typescript-eslint/no-empty-function
            const startLogStreaming = async (onChunk = () => { }) => {
                if (!streamPromise) {
                    const response = await this.snapshotsApi.getSnapshotBuildLogsUrl(createdSnapshot.id);
                    const url = `${response.data.url}?follow=true`;
                    streamPromise = processStreamingResponse(() => fetch(url, { method: 'GET', headers: this.clientConfig.baseOptions.headers }), (chunk) => onChunk(chunk.trimEnd()), async () => terminalStates.includes(snapshotRef.createdSnapshot.state));
                }
            };
            if (options.onLogs) {
                options.onLogs(`Creating snapshot ${createdSnapshot.name} (${createdSnapshot.state})`);
                if (createSnapshotReq.buildInfo &&
                    createdSnapshot.state !== SnapshotState.PENDING &&
                    !terminalStates.includes(createdSnapshot.state)) {
                    await startLogStreaming(options.onLogs);
                }
            }
            let previousState = createdSnapshot.state;
            while (!terminalStates.includes(createdSnapshot.state)) {
                if (options.onLogs && previousState !== createdSnapshot.state) {
                    if (createSnapshotReq.buildInfo && createdSnapshot.state !== SnapshotState.PENDING && !streamPromise) {
                        await startLogStreaming(options.onLogs);
                    }
                    options.onLogs(`Creating snapshot ${createdSnapshot.name} (${createdSnapshot.state})`);
                    previousState = createdSnapshot.state;
                }
                await new Promise((resolve) => setTimeout(resolve, 1000));
                createdSnapshot = await this.get(createdSnapshot.name);
                snapshotRef.createdSnapshot = createdSnapshot;
            }
            if (options.onLogs) {
                if (streamPromise) {
                    await streamPromise;
                }
                if (createdSnapshot.state === SnapshotState.ACTIVE) {
                    options.onLogs(`Created snapshot ${createdSnapshot.name} (${createdSnapshot.state})`);
                }
            }
            if (createdSnapshot.state === SnapshotState.ERROR || createdSnapshot.state === SnapshotState.BUILD_FAILED) {
                const errMsg = `Failed to create snapshot. Name: ${createdSnapshot.name} Reason: ${createdSnapshot.errorReason}`;
                throw new DaytonaError(errMsg);
            }
            return createdSnapshot;
        }
        /**
         * Activates a snapshot.
         *
         * @param {Snapshot} snapshot - Snapshot to activate
         * @returns {Promise<Snapshot>} The activated Snapshot instance
         */
        async activate(snapshot) {
            return (await this.snapshotsApi.activateSnapshot(snapshot.id)).data;
        }
        /**
         * Processes the image contexts by uploading them to object storage
         *
         * @private
         * @param {Image} image - The Image instance.
         * @returns {Promise<string[]>} The list of context hashes stored in object storage.
         */
        static async processImageContext(objectStorageApi, image) {
            if (!image.contextList || !image.contextList.length) {
                return [];
            }
            const ObjectStorageModule = await dynamicImport('ObjectStorage', '"processImageContext" is not supported: ');
            const pushAccessCreds = (await objectStorageApi.getPushAccess()).data;
            const objectStorage = new ObjectStorageModule.ObjectStorage({
                endpointUrl: pushAccessCreds.storageUrl,
                accessKeyId: pushAccessCreds.accessKey,
                secretAccessKey: pushAccessCreds.secret,
                sessionToken: pushAccessCreds.sessionToken,
                bucketName: pushAccessCreds.bucket,
            });
            const contextHashes = [];
            for (const context of image.contextList) {
                const contextHash = await objectStorage.upload(context.sourcePath, pushAccessCreds.organizationId, context.archivePath);
                contextHashes.push(contextHash);
            }
            return contextHashes;
        }
    };
})();
export { SnapshotService };
